{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "52277277",
   "metadata": {},
   "source": [
    "### Data Ingestion\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e2956bb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "### document data structure\n",
    "from langchain_core.documents import Document"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "09483734",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Document(metadata={'source': 'example.txt', 'pages': 1, 'author': 'Harshiddhi Pathak', 'date_created': '2025-11-25'}, page_content='this is the main text content I am using to  create RAG')"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doc=Document(\n",
    "    page_content= \"this is the main text content I am using to  create RAG\",\n",
    "    metadata={\n",
    "        \"source\" : \"example.txt\",\n",
    "        \"pages\": 1,\n",
    "        \"author\" : \"Harshiddhi Pathak\",\n",
    "        \"date_created\": \"2025-11-25\"\n",
    "    }\n",
    ")\n",
    "doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "ca74a60e",
   "metadata": {},
   "outputs": [],
   "source": [
    "##create a simple txt file\n",
    "import os\n",
    "os.makedirs(\"../data/text_files\",exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "20cec632",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample txt file created!\n"
     ]
    }
   ],
   "source": [
    "sample_texts={\n",
    "    \"../data/text_files/python_intro.txt\":\"\"\"Python Programming Introduction\n",
    "    Python is a high-level, interpreted programming language known for its simplicity and readability.\n",
    "Created by Guido van Rossum and first released in 1991, Python has become one of the most popular\n",
    "programming languages in the world.\n",
    "\n",
    "Key Features:\n",
    "- Easy to learn and use\n",
    "- Extensive standard library\n",
    "- Cross-platform compatibility\n",
    "- Strong community support\n",
    "\n",
    "Python is widely used in web development, data science, artificial intelligence, and automation.\"\"\",\n",
    "    \n",
    "    \"../data/text_files/machine_learning.txt\": \"\"\"Machine Learning Basics\n",
    "\n",
    "Machine learning is a subset of artificial intelligence that enables systems to learn and improve\n",
    "from experience without being explicitly programmed. It focuses on developing computer programs\n",
    "that can access data and use it to learn for themselves.\n",
    "\n",
    "Types of Machine Learning:\n",
    "1. Supervised Learning: Learning with labeled data\n",
    "2. Unsupervised Learning: Finding patterns in unlabeled data\n",
    "3. Reinforcement Learning: Learning through rewards and penalties\n",
    "\n",
    "Applications include image recognition, speech processing, and recommendation systems\n",
    "\n",
    "\n",
    "\"\"\"\n",
    "}\n",
    "\n",
    "for filepath,content in sample_texts.items():\n",
    "    with open(filepath, 'w', encoding=\"utf-8\") as f:\n",
    "        f.write(content)\n",
    "\n",
    "print(\"sample txt file created!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "f56f22af",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Document(metadata={'source': '../data/text_files/python_intro.txt'}, page_content='Python Programming Introduction\\n    Python is a high-level, interpreted programming language known for its simplicity and readability.\\nCreated by Guido van Rossum and first released in 1991, Python has become one of the most popular\\nprogramming languages in the world.\\n\\nKey Features:\\n- Easy to learn and use\\n- Extensive standard library\\n- Cross-platform compatibility\\n- Strong community support\\n\\nPython is widely used in web development, data science, artificial intelligence, and automation.')]\n"
     ]
    }
   ],
   "source": [
    "### TextLoader\n",
    "from langchain_community.document_loaders import TextLoader\n",
    "loader =TextLoader(\"../data/text_files/python_intro.txt\",encoding=\"utf-8\")\n",
    "document = loader.load()\n",
    "print(document)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "c7fd6882",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'source': '..\\\\data\\\\text_files\\\\machine_learning.txt'}, page_content='Machine Learning Basics\\n\\nMachine learning is a subset of artificial intelligence that enables systems to learn and improve\\nfrom experience without being explicitly programmed. It focuses on developing computer programs\\nthat can access data and use it to learn for themselves.\\n\\nTypes of Machine Learning:\\n1. Supervised Learning: Learning with labeled data\\n2. Unsupervised Learning: Finding patterns in unlabeled data\\n3. Reinforcement Learning: Learning through rewards and penalties\\n\\nApplications include image recognition, speech processing, and recommendation systems\\n\\n\\n'),\n",
       " Document(metadata={'source': '..\\\\data\\\\text_files\\\\python_intro.txt'}, page_content='Python Programming Introduction\\n    Python is a high-level, interpreted programming language known for its simplicity and readability.\\nCreated by Guido van Rossum and first released in 1991, Python has become one of the most popular\\nprogramming languages in the world.\\n\\nKey Features:\\n- Easy to learn and use\\n- Extensive standard library\\n- Cross-platform compatibility\\n- Strong community support\\n\\nPython is widely used in web development, data science, artificial intelligence, and automation.')]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### directory loader\n",
    "\n",
    "from langchain_community.document_loaders import DirectoryLoader\n",
    "\n",
    "## load all the text files from the directory\n",
    "dir_loader=DirectoryLoader(\n",
    "    \"../data/text_files\",\n",
    "    glob=\"**/*.txt\", ## Pattern to match files  \n",
    "    loader_cls= TextLoader, ##loader class to use\n",
    "    loader_kwargs={'encoding': 'utf-8'},\n",
    "    show_progress=False\n",
    "\n",
    ")\n",
    "\n",
    "documents=dir_loader.load()\n",
    "documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "be525e9c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content=\"2007 IEEE/ICME International Conference on Complex Medical En2ineerin2\\nA Study of Combination fMRI and DTI in A\\nCognitive Task\\nTingting Wang', Zifan Cai1, Ni Tang1, Jie Lu3, Li Yao1 2 Xiaojie Zhaol2*\\n1 School of Information Science and Technology, Beijing Normal University\\n2 State Key Laboratory of Cognitive Neuroscience and Learning, Beijing Normal University\\n3 Department of Radiology, Xuanwu Hospital of Capital University of Medical Sciences\\n*zhaoxj 86ghotmail.com\\nAbstract-Magnetic\\nresonance\\ndiffusion\\ntensor\\nimaging\\n(MR-DTI)\\nas\\na noninvasive technique\\ncan\\ninvestigate\\nthe\\ncerebral white matter architecture based on the directional\\ninformation of diffusion that is related to fiber tract. With the\\ndevelopment of DTI technique in recent years, more and more\\nstudies of combining fMRI with DTI are used to explore the\\nrelationships between functional connectivity and its structural\\nproperties in vivo, especially in pathological researches. In our\\nwork, we\\nfocus\\non whether\\nfunctional\\nconnectivity\\ntakes\\nstructural connectivity as its basis in a specific cognitive task in\\nnormal\\nbrains.\\nTemporal\\ncorrelation method was used\\nto\\nacquire the functional connectivity network based on fMRI in a\\npicture-naming task. Fractional anisotropy (FA) that denoted\\nthe degree of the diffusion anisotropy was calculated using\\ndiffusion tensor matrix based on DTI. Then the principle\\nvector of tensor matrix was mapped into X, Y, and Z axes with\\nRGB color and weighting the color brightness by FA. We\\nnormalized the two data results before combining them so as to\\nmatch them better. Then functional connectivity maps were\\noverlaid upon DTI directional color maps that indicated the\\nwhite matter tract orientation based on the principle vector of\\ntensor\\nmatrix.\\nThe\\nresults\\nshowed\\nthat\\nthe\\nfunctional\\nconnected regions of brain in this cognitive task had white\\nmatter\\nconnectivity.\\nIt\\nmay\\nbe\\nsuggested\\nfunctional\\nconnectivity\\ndid\\ntake\\nstructure\\nas\\nits\\nbasis,\\nand\\nthe\\ncombination of fMRI and DTI could explore effectively the\\ncorrelation of structure and function\\nin both healthy and\\ndisease cerebra.\\nI. INTRODUCTION\\nInvestigation of the relationships between the cerebra\\nfunctional\\nconnectivity\\nof the\\nbrain\\nand\\nits\\nstructure\\nproperties has become a popular approach for neuroscience\\nresearch. Function-structure correlations indicate how white\\nmatter structural connections relate to regional gray matter\\nactivity connections [1] [2]. Functional connected network\\nare often obtained by using functional MRI (fMRI) with\\ntemporal correlation or independent component analysis\\n(ICA),\\netc.[3][4].\\nIn vivo Magnetic resonance diffusion\\ntensor imaging (MR-DTI) as a noninvasive technique has\\nbecome a popular tool for investigation of white matter\\narchitecture [5][6].\\nDTI detect fiber pathways by measuring the degree of\\nwater\\ndiffusion\\nanisotropic\\n(directionally\\ndependent)\\nin\\nwhite matter. Axonal membranes and myelin sheaths present\\nbarriers to the motion of water molecules in directions not\\nparallel to their own orientation. The direction of maximum\\ndiffusivity has been shown to coincide with white matter\\nfiber tract orientation [7]. This information is contained in\\nthe diffusion tensor which is a 3*3 diagonally symmetric\\n(Dij= Dji) matrix with six degrees of freedom. Based on the\\ntensor\\nmatrix,\\none\\ncould\\nobtain\\ndiffusion\\nanisotropy\\ninformation\\n(e.g.\\nFractional\\nanisotropy,\\nFA)\\nor\\nfiber\\ntractography etc. These results derived from DTI have been\\nwidely\\nused\\nin\\nclinic\\nand\\nvisualization\\nof anatomic\\nconnections between different parts of brain.\\nWith the quickly development of DTI, more and more\\nstudies have combined fMRI\\nand DTI techniques\\nfor\\nresearch on correlation of structure and function. During the\\ncombination, fiber tractography, fast-marching tractography\\n(FMT),\\nProbabilistic\\nIndex\\nof\\nConnectivity\\n(PICo)\\ntractography and DTI directional eigenvector color maps are\\noften used\\nto denote white matter connectivity.\\nThese\\nstudies show that this combination can give additional\\ninformation about brain disease and organization of brain\\nfunctions for doctors and researchers [8][9]. However, few\\nstudies\\nfocus on\\na given cognitive task\\nto\\ninvestigate\\nwhether functional connectivity in this task takes structural\\nconnectivity as its basis in normal brain.\\nIn this work, a picture-naming cognitive task was chosen\\nto\\ninvestigate\\nfunction-structure\\nrelationships.\\nThis task\\nwould activate the cerebral visual\\ncortex and semantic\\nprocessing regions. Brain activity regions are processed by\\ntemporal\\ncorrelation method that\\nis\\na commonly used\\napproach\\nfor\\nidentifying\\nfunctional\\nnetworks\\nof brain\\nregions. Then we obtained directional eigenvector color\\nmaps which\\nis\\nrelated\\nto\\nthe\\nprinciple\\nvector of the\\ndiagonally tensor matrix and weighted the color brightness\\nby FA. These directional color maps depict the main white\\nmatter\\nfiber\\npathways.\\nIn\\nfurther\\nresearches,\\nfiber\\ntractography could be used to find more specific fiber\\nconnectivity information between activated brain regions.\\nAfter that, combination of the results from fMRI and DTI\\ndata\\nwere\\nimplemented\\nby\\noverlaying\\nfunctional\\nconnectivity\\nmaps\\non\\ndirectional\\ncolor\\nmaps\\nwith\\nnormalization [1]. It has shown that functional connectivity\\ntook structural connectivity as its basis in this cognitive task.\\n1-4244-1078-9/07/$25.00 c2007 IEEE.\\n896\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content='2007 IEEE/ICME International Conference on Complex Medical Engineering\\nII. METHODS\\nA. Data acquisition\\nFifteen\\nhealthy\\nright-handed\\nsubjects\\n(8\\nmales)\\nparticipated in this task after giving informed consent in\\naccordance with the Edinburgh Handedness Inventory. Their\\nages range from 18 to 25 years, with a mean age of 2 1.1.\\nAll subjects executed picture naming task, implemented\\nin\\na\\nstandard block\\ndesign.\\nThe\\npicture naming\\ntask\\nconsisted of five experimental interleaved with six control\\nepochs. Each experimental and control epoch lasted 24s.\\nDuring\\nthe\\nexperimental\\nperiod,\\nthe\\nparticipant\\nwas\\ninstructed to view the screen, on which exposed black-white\\npictures, and then named the pictures bleatingly. During the\\ncontrol period, crosshair (+) presented at the screen center\\nwas used\\nas an experimental resting baseline and the\\nparticipant just passively viewed the screen. The session\\nlasted 4 minutes and 24 seconds.\\nThe fMRI scanning was performed with a Siemens 1.5\\nTesla scanner at Xuan Wu Hospital in Beijing. A gradient\\necho EPI sequence was used during functional scans (TR=\\n2000ms,\\nTE=40ms,\\nmatrix\\nsize\\n64x64,\\nfield\\nof view\\n220x220mm2, 6mm thickness with 1.2mm gap, and 20\\nslices over whole brain).\\n132 volumes of images were\\nacquired.\\nThe\\nDTI\\nacquisition\\nsequence\\nwas\\na\\nsingle-shot\\nspine-echo\\nplanar\\nimaging\\n(SE/EPI)\\n(TR=61OOms,\\nTE=106ms, 128X128 matrix size (256X256 reconstructed),\\n230x230mm2 field of view, 4mm thickness with 4mm gap,\\n30\\nslices\\nover\\nwhole\\nbrain,\\na maximum\\nb-value\\nof\\n1000mm2/s in 6 noncollinear directions ). The reconstructed\\nvoxel size was 0.9x0.9x4.Omm3.\\nB. Functional MRI dataprocessing\\nData\\nwere\\npreprocessed\\nusing\\nSPM2\\nsoftware\\n(www.fil.ion.ucl.ac.uk/spm). The first 5 time points were\\ndiscarded for scanner calibration and for subjects to get used\\nto the circumstance. Redundant 127 time series were kept\\nfor further analysis. Images were corrected for movement by\\nusing\\nleast-square\\nminimization\\nwithout\\nhigher-order\\ncorrections for spin history and normalized to stereotaxic\\ncoordinates of Talairach and Tournoux. Images were then\\nresampled into 3x3x4mm3 and smoothed with an 8-mm\\nGaussian kernel to decrease spatial noise.\\nStatistical\\nanalysis was performed on individual and\\ngroup data by using the general linear model and Gaussian\\nrandom\\nfields theory\\nas implemented\\nin SPM2. Group\\nanalyses were performed using a random-effects model that\\nestimates the error variance across subjects, and therefore\\nprovided a stronger generalization to the population from\\nwhich data are acquired. Statistical maps were superimposed\\non normalized high-resolution Ti -weighted images and their\\nlocations\\ninterpreted\\nusing\\nknown\\nneuroanatomical\\nlandmarks.\\nA commonly used approach for identifying networks of\\nfunctionally connected brain regions is temporal correlation\\nmethod involving defining a cluster of activated voxels in a\\nregion of interest (ROI) and using this \"seed\" to determine\\nwhich voxels throughout the entire brain are functionally\\ninteracting with the seed. The seed region\\nis generally\\nchosen based on its known involvement in the behavioral\\ntask and the\\nresearcher\\'s\\ninterest\\nin\\ncharacterizing\\nits\\ninteractions with other regions of brain. In this case, the\\nmiddle occipital gyrus in the left cerebrum was used as a\\nseed because the experiment was designed to test a specific\\nhypothesis\\nconcerning\\ninterhemispheric\\ninteractions\\nbetween visual cortices and between visual cortex and\\nsemantic processing cortex. Seen from the group activation\\nmap, in the middle occipital gyrus, the voxel at [-36 -90 0]\\n(Talairach coordinates) had the highest peak t score and was\\ntaken as a center of a sphere with a radius of 8mm, which\\nwas selected as the ROI (seed). Then, we applied the ROI to\\nthe subjects to analyze the functional connectivity.\\nBefore\\nthe\\nfMRI\\nanalysis,\\ntemporal\\nsmoothing\\nand\\nscaling steps were performed across all brain voxels as\\nfollows. Time series of all brain voxels was smoothed\\nthrough\\na\\n3-point\\nhamming\\nwindow\\nto\\nreduce\\nthe\\nhigh-frequency noise and the effect of low-frequency drift.\\nAfter this, to minimize the effect of global drift, voxel\\nintensities were scaled by dividing each time point\\'s value\\nby the mean value of the whole-brain image at that time\\npoint. To perform the fMRI analysis, time series from the\\nscan were extracted for the seed by weighted-averaging the\\ntime series of all voxels in the seed. The weight for every\\nvoxel in the seed was the proportion of its t score to the sum\\nof\\nall\\nthe\\nt\\nscores\\nin\\nthe\\nseed.\\nThe\\nresulting\\nweighted-averaged time series served as a reference model\\nfor cross-correlation analysis with the whole brain voxels,\\nyielding a spatial zero-lag cross-correlation map.\\nC. DTI data processing\\nDiffusion weighted images\\nS\\napplied with\\ndiffusion\\nsensitizing field gradient G were acquired based on the\\nStejeskal-Tanner spin echo Scheme and signal value So\\nwithout the G:\\n_S\\n2 G 2\\n2 ( A --) D\\nSe-\\nb\\n0O\\n(1)\\nWhere b is the diffusion sensitization. In this case, the\\nnonlinear directions ofthe gradient were six:\\ngl = 1 /2(1,0, 1-)\\ng2 = 1/ V/2(1, -1,0)\\ng3 = 1/\\n2(0,-1,1)\\ng4 = 1/\\n2(1,0,1)\\ng5 = I /\\n/2(1, 1,0)\\ng6 = I/+2(0,1,l) ,\\nFig. 1. Several slices of brain activated regions. Colors denote the strong\\ndegree of activity.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 2}, page_content=\"2007 IEEE/ICME International Conference on Complex Medical Engineering\\n1/\\n...\\ni.\\n:M\\nFig. 2. Location of ROI (white area). It's in the middle occipital gyrus at\\n[-36 -90 0] (r=8mm, Talairach coordinates)\\nThe images of the S and SO\\nis presented as follows\\n(Fig.3):\\n4;/\\nFig. 4. the diffusion ellipsoid\\nFig. 3. images at six nonlinear gradient directions and without gradient\\nThus, tensor matrix is derived from (1):\\nD\\n-ln\\nb\\nS0\\n(2)\\nWhere D is a 3*3 symmetric matrix with six freedom\\ndegrees:\\n(3)\\nThen the diffusion tensor matrix D was diagonalized for\\nobtaining three eigenvalues (XQl>2>23) and corresponding\\neigenvectors (el>e2>e3) that represented three axes of the\\ndiffusion ellipsoid (Fig.4).\\nWe can calculate some diffusion index from the three\\neigenvalues to scale the degree of the diffusion. The most\\ncommon index\\nis the\\nfractional anisotropy (FA) which\\ndenoted\\nthe\\ndegree\\nof\\nthe\\ndiffusion\\nanisotropy\\nwas\\ncalculated\\nfrom\\nthe\\nthree\\neigenvalues\\nranging\\nfrom\\n0\\n(isotropy) to\\n1 (maximum anisotropy) [10]:\\nFA = 13[(A1- A\\n+ (A2 - A )2 + (A3 - A )2]\\n_ aA12\\n+ A\\n2\\n+\\n)\\n(4)\\nWhere X denotes the mean of the three eigenvalues, which\\nis equal to the directionally averaged diffusivity.\\nThe major eigenvector\\ncorresponding\\nto\\nthe maximum\\neigenvalue denotes the local fiber orientation. The vector\\norientation maps were generated by mapping the major\\neigenvector directional components into RGB color along x,\\ny,\\nand\\nz\\nand weighting\\nthe\\ncolor\\nbrightness by\\nFA.\\nConventionally the directional RGB color mapping is red\\nfor\\nleft-right,\\ngreen\\nfor\\nanteroposterior,\\nand\\nblue\\nfor\\nsuperior-inferior [17].\\nD. combination ofthefMRI and DTI\\nThe fMRI and DTI data were both normalized to a same\\nEPI-template to acquire some normalized parameters before\\nthe individual processing of the two kinds of data. These\\nparameters were preserved until the combination. After\\nobtaining\\nthe\\nfunctional\\nconnectivity\\nmaps\\nand\\nthe\\ndirectional\\neigenvector\\ncolor\\nmaps,\\nthese\\nnormalized\\nparameters were applied to them for normalization. Finally,\\nwe overlaid the normalized functional connectivity maps\\nupon the directional color maps.\\nIII. RESULTS AND CONCLUSIONS\\nAll subjects' fMRI data and DTI data were processed and\\ncombined correspondingly. We chose the representative\\nresult of 12th subject to show the correlation of the structure\\nand function.\\nSignificantly activated brain regions that covary with one\\nanother of PN task are interpreted as functional connected.\\nFrom the functional connectivity maps, we could observe\\nthat there was evident strong inter-hemispheric correlation\\nbetween occipital lobe, which was responsible for visual\\n|\\nl\\n1\\n......................\\ncorr\\nM0.5\\nFig. 5. Correlation maps showing the interregional temporal correlation\\nbetween ROI (white area) and all brain voxels. The colored regions are\\nconsidered in a same functional connectivity network.\\nP.\\ni.u\\nD-,(-.'\\nD=, .DOW\\ni.u\\nZ.z\\nD.z\\nD!7\\nD=\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content='2007 IEEE/ICME International Conference on Complex Medical Engineering\\nFig. 6. Two slices of directional color maps overlaid by functional\\nconnectivity network. The color indicates direction as follows: red means\\nleft-right, green means antero-posterior, and blue means superior-inferior.\\nBrightness is proportional to FA. Functional connected regions were\\nprofiled in white color.\\nprocessing,\\nand\\nthere\\nwas\\nalso\\nsignificant\\ncorrelation\\nbetween middle occipital gyrus and middle frontal gyrus in\\nleft\\ncerebrum,\\nwhich\\nwas\\nresponsible\\nfor\\nsemantic\\nprocessing (Fig. 5).\\nThe\\ndirectional\\ncolor\\nmaps\\noverlaid\\nby\\nfunctional\\nconnectivity networks obviously depicted the fiber pathways\\nbetween the functional connected regions. From the maps,\\nthe green color compound with blue indicated the structural\\nconnectivity between the occipital gyrus and the middle\\nfrontal\\ngyrus\\nboth\\nin\\nleft\\ncerebrum\\nalong\\nthe\\nantero-posterior direction (Fig. 6 left). The green compound\\nwith red color indicated that the optic radiant fiber derived\\nfrom\\nthe\\nbilateral\\nvisual\\ncortex\\ntrack\\nalong\\nthe\\nantero-posterior direction a short distance and then joint at\\nthe lateral geniculate body along the left-right direction (Fig.\\n6 right).\\nIV. DISCUSSION\\nThe combination of fMRI and DTI is related to many\\naspects. First, data acquisition is important for combination\\nof fMRI and DTI. The ideal data are acquired by employing\\nclosely matched EPI sequences for both datasets. The most\\ninfluential parameters are the voxel size and the number of\\nwhole brain\\nslices. Though there\\nare some differences\\nbetween the two scan technologies, we suggest the data\\nparameters be as matched as possible. If this condition can\\'t\\nbe satisfied, we should do normalization before combination.\\nAll the original images of fMRI and DTI were normalized to\\nthe EPI template and preserved the parameters. After the\\nindividual processing of the two kinds of data, we used the\\nparameters\\nobtained\\nbefore\\nto\\nnormalize\\nthem.\\nThis\\nprocedure will bring some distortion to the images. In\\nfurther researches, the methods of matching the two data\\nwill be also the main aspects of study on combination.\\nFunctional connectivity networks obtained in our work\\nare temporal correlation method. The chosen of ROI and the\\nweighted values of the time series in ROI is the key point of\\nthis method and can be optimized in further researches. Also\\nwe could use ICA or other models to acquire the functional\\nconnectivity networks.\\nStructure\\nconnections\\ncan\\nbe\\nobtained\\nby\\nfiber\\ntractography, FMT algorithm, PICo tractography algorithm\\nor directional eigenvector color maps as mentioned before.\\nFMT\\nand\\nPICo\\ntractography\\nalgorithm\\nonly\\nprovide\\nconnectivity probability between brain regions, so they are\\nless visual than the other two. In this work, we chose\\ndirectional color maps to denote the fiber orientation. This\\nmethod provide a concise approach to typically visualize the\\nfiber\\nmajor\\ntract\\nwithout\\nthe\\nmore\\nadvanced\\nand\\nsophisticated tractography techniques which will take longer\\nto reach practice.\\nSince\\nit\\nalso can indicate the\\nfiber\\norientation, we often use it when we need rapid analysis of\\nsome cognitive functional connectivity or some clinical\\ncases. Furthermore, basing on the results of this method,\\nresearchers\\ncan\\ndetermine\\nweather\\nthey\\nneed\\nfiber\\ntractography to provide more specific information of white\\nmatter structure.\\nPicture Naming (PN) cognitive task was chosen in our\\nstudy\\nfor\\nits\\nsimple\\nand\\nacknowledged\\nfunctional\\nconnectivity network [19]. The results of combination are\\nconsistent\\nwith\\nour\\nhypothesis\\nthat\\ncerebra\\nfunctional\\nconnectivity\\nhas\\nstructure\\nconnectivity\\nas\\nits\\nbasis\\nin\\nspecific cognitive task.\\nThis study may provide a basis for future studies to reveal\\nthe underlying neural networks associated with function and\\nstructure.\\nACKNOWLEDGMENT\\nThis\\nwork\\nis\\nsupported\\nby\\nNatural\\nScience\\nKey\\nFoundation of Beijing (4061004), National Natural Science\\nFoundation of China (60628101), and National High-tech\\nR&D Program (863 Program, 2006AAOlZ132).\\nREFERENCES\\n[1] D.J. Werring, C.A. Clark, and G.J.M. Parker, \"A Direct Demonstration\\nof both Structure and Function in the Visual System: Combining\\nDiffusion\\nTensor\\nImaging\\nwith\\nFunctional\\nMagnetic\\nResonance\\nImaging\", NeuroImage, Vol. 9, pp.352-361, 1999\\n[2]\\nM.\\nGuye,\\nand\\nG.J.M.\\nParker,\\n\"Combined\\nfunctional MRI\\nand\\ntractography to demonstrate the connectivity of the human primary\\nmotor cortex in vivo\", NeuroImage, Vol. 19, pp.1349-1360, 2003\\n[3] K.J. Friston, \"Functional and effective connectivity in neuroimaging: a\\nsynthesis\", Hum. Brain Mapp., Vol. 2, pp. 56-78, 1994\\n[4] M. Hampson, B.S. Peterson, P. Skudlarski, J.C. Gatenby, and J.C, Gore,\\n\" Detection of functional connectivity using temporal correlations in\\nMR Images\", Hum. Brain Mapp., Vol. 15, pp.247-262, 2002\\n[5] C.F. Westin, S.E. Maier, H. Mamata, \"Processing and visualization for\\ndiffusion tensor MRI\", Mledical Image Analysis, Vol. 6, pp.93-108,\\n2002\\n[6] E.R. Melhem, S. Mori, and G. Mukundan, \"Diffusion Tensor MR\\nImaging of the Brain and White Matter Tractography\", AJR , ppI78,\\n2002\\n[7]\\nP.J.\\nBasser,\\nJ.\\nMatiello,\\nand D.L.\\nBihan, \"MR diffusion tensor\\nspectroscopyand imaging\", Biophys, Vol. 66, pp. 259-267,1994\\n[8] S.L. Heller, and L.A. Heiera, \"Evidence of cerebral reorganization\\nfollowing\\nperinatal\\nstroke\\ndemonstrated\\nwith\\nfMRI\\nand\\nDTI\\ntractography\", Clin. Imaging, Vol. 29, pp.283-287, 2005\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 4}, page_content='2007 IEEE/ICME International Conference on Complex Medical Engineering\\n[9] T. Schonberg, P. Pianka, and T. Hendler, \"Characterization of displaced\\nwhite\\nmatter by\\nbrain tumors\\nusing combined DTI\\nand fMRI\",\\nNeuroImage, Vol. 30, pp. 1100-1111, 2006\\n[10]\\nP.\\nBasser,\\nand C.\\nPierpaoli,\\n\"Microstructural\\nand physiological\\nfeaturesof tissues elucidated by quantitative-diffusion-tensor\", MRI[J].J\\nMagn Reson B, Vol. 111, pp.209-219, 1996\\n[11] Y. Masutani, and S. Aoki, \"MR diffusion tensor imaging: recent\\nadvance and new techniques for diffusion tensor visualization\", Eur. J.\\nRadio., Vol. 46, pp.53-66, 2003\\n[12] D. Le Bihan, J.F. Mangin, and C. Poupon, \"Diffusion Tensor Imaging:\\nConcepts and Applications\", J. magn. reson. imaging, Vol.\\n13, pp.\\n534-546, 2001\\n[13]\\nA.T.\\nToosy,\\n0.\\nCiccarelli,\\nand\\nG.J.M.\\nParker,\\n\"Characterizing\\nfunction-structure\\nrelationships\\nin the human\\nvisual\\nsystem with\\nfunctional MRI and diffusion tensor imaging\", NeuroImage, Vol. 21,\\npp. 1452- 1463, 2004\\n[14] M.L. Seghier, F. Lazeyras, and S. Zimine, \"Combination of\\nevent-related fMRI and diffusion tensor imaging in an infant with perinatal\\nstroke\\', NeuroImage, Vol. 21, pp.463-472, 2004\\n[15]\\nP.J.\\nBasser,\\nand\\nD.K.\\nJones,\\n\"Diffusion-tensor\\nMRI:\\ntheory,\\nexperimental\\ndesign and\\ndata\\nanalysis-a technical\\nreview\", NMR\\nBiomed., Vol. 15, pp.456-467, 2002\\n[16] C.R. Tench, P.S. Morgan, M. Wilson, and L.D. Blumhardt, \"White\\nMatter Mapping Using Diffusion Tensor MRI\", Magn. Reson. Med.,\\nVol. 47, pp.967-972, 2002\\n[17] B.J. Jellison, A.S. Field, J. Medow, and M. Lazar, \"Diffusion Tensor\\nImaging of Cerebral White Matter: A Pictorial Review of Physics, Fiber\\nTract Anatomy, and Tumor Imaging Patterns\", AJNR Am JNeuroradiol,\\nVol. 25, pp. 356-369, 2004\\n[18]\\nS.\\nMori,\\nand van\\nZijl\\nP.C.M.,\\n\"Fiber tracking:\\nprinciples\\nand\\nstrategies\\na technical review\", NMR Biomed., Vol. 15, pp. 468-480,\\n2002\\n[19] C. Whatmough, and H. Chertkow, \"Dissociable brain regions process\\nobject\\nmeaning\\nand\\nobject\\nstructure\\nduring\\npicture\\nnaming\",\\nNeuropsychologia, Vol. 40, pp.174-186, 2002\\nw90\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='Hybrid CNN-RNN Models for Multimodal \\nAnalysis of Autism Spectrum Disorder \\nNeuroimaging \\nRajat Kishor Varshney \\nSchool of Computing Science & \\nEngineering, Research Scholar, Galgotias \\nUniversity, Greater Noida ,203201, India \\nrajatvarshney1207@gmail.com \\n \\n \\nAlok Katiyar \\nSchool of Computing Science & \\nEngineering, Professor, Galgotias \\nUniversity, Greater Noida ,203201, India \\nalok_katiyar@hotmail.com \\n \\n \\nPrashant Johri \\nSchool of Computer Application & \\nTechnology, Professor, Galgotias \\nUniversity, Greater Noida ,203201, India \\njohri.prashant@gmail.com \\n \\nAbstract. ASD is a developmental disorder in the \\nneurological category and has numerous and diverse symptoms. \\nDiagnosis \\nis \\nstill \\nproblematic \\nbecause \\nsymptoms \\nof \\nparkinsonism are diverse, and the dimensionality of the \\nneuroimaging data is high. This paper presents a combination \\nof CNN and RNN for classifying the ASD by analyzing \\nmultimodal neuroimaging data such as fMRI data, MRI data, \\nand DTI data. The CNN component analyzes spatial \\ncharacteristics of the neuroimaging data, and the RNN learns \\ntemporal patterns in fMRI series. Enhancements include a novel \\npreprocessing step called 3D Brain Region Alignment (3D-BRA) \\nas well as the 3D version of Adaptive Temporal, minimizing \\ndimensionality through an innovative feature selection \\ntechnique employing EN-CC. Predictive analysis with the \\nhybrid model utilizes standard benchmark datasets such as \\nABIDE I, ABIDE II, HCP, and SFARI and records classification \\nperformances \\nover \\n95 \\npercentage. \\nTo \\nenhance \\nthe \\ninterpretability of the model, the Grad-CAM algorithm is used, \\nwhich marks specific regions of the brain important in the \\ncontext of ASD. The proposed framework seems to possess \\nconsiderable feasibility for enhancing the diagnosis of ASD and \\nimproving the understanding of its neural substrates, making \\nthe developed method highly advantageous for applications in \\nneuroinformatics and clinical utility. \\nKeywords: Autism Spectrum Disorder, Neuroimaging, fMRI, \\nHybrid CNN-RNN, Deep Learning, Brain Connectivity, Feature \\nSelection, Multimodal Analysis. \\nI. \\nINTRODUCTION \\nASD is a developmental disorder with notable deficits in \\nsocial skills acts, verbal and nonverbal communication and \\nrepetitive and stereotyped behavior. ASD identification has \\nhistorically been made based on behavioral approaches; \\nnevertheless, such procedures are limited as a result of diverse \\nsymptomatology. \\nBrain \\nimaging \\nmethods, \\nincluding \\nfunctional Magnetic Resonance Imaging (fMRI) and \\nStructural MRI, are also beneficial to the organization in \\nstudying ASD, since they help to establish a different brain \\nconnectivity and structure. These advancements have created \\na new avenue and strategy in which neuroimaging data can be \\napplied in computational models to aid in ASD diagnosis and \\nits categorization[1]. \\nUnlike conventional machine learning techniques that \\nattempt to develop hypothesis for single layer inputs, deep \\nlearning has made it possible to extract high level abstractions \\nfrom high dimensional medical imaging data. Earlier, \\nConvolutional Neural Networks (CNNs) have performed well \\nin identifying spatial patterns of neuroimaging data, and \\nRecurrent Neural Networks (RNNs) are quite effective in \\ncapturing temporal dynamic[2]. Although both of these \\narchitectures have their weakness when using neuroimaging \\ndata for ASD classification, integrating CNN and RNN into a \\nCNN-RNN hybrid model advantageously, thus making it a \\nsuitable framework for the classification of ASD. Working \\nwith neuroimaging data is very difficult because such data is \\nhigh-dimensional, noisy, and can differ greatly between \\nindividuals. Moreover, while fMRI data and MRI data are \\ncollected with temporal resolution in terms of time, and spatial \\ndistribution in terms of space, the complexity of these two \\ndimensions in the data highlights the need for adequately \\ncapturing both temporal and spatial components of the data. \\nThese then pose challenge to standard approaches hence the \\npoor results performances in categorizing ASD. These \\nchallenges can be tackle by a hybrid CNN-RNN framework \\nthat can capture the spatial and temporal features of an image \\nand thus performs a comprehensive feature extraction.  \\nMulti-modal neuroimaging combines fMRI, structural \\nMRI, and DTI, which shows superior informative content in \\ncomparison with only fMRI or sMRI. All these modalities \\nbring in differing information; fMRI reveals functional \\nconnection \\nwhile \\nMRI \\nand \\nDTI \\nreveal \\nstructural \\ncharacteristics [3]. These should be integrated models where \\nadditional data can enrich the result by adding complementary \\ndata from the other modalities. Convolutional Neural \\nNetworks are applied successfully for analyzing the spatial \\ncharacteristics of the neuroimaging data. To localize the ASD \\nsymptoms, CNNs used contextual convolutional filters and \\nsuccessfully identified the active regions in the brain as well \\nas the structural changes. The architectures including ResNet-\\n50 and Efficient Net have been tested on large-scale datasets, \\ntherefore can extract hierarchical features for spatial feature \\nextraction in this hybrid way [4]. \\nDue to the time dependency of the data, especially in fMRI \\nrecordings, mod- els that can handle time series data are \\nrequired. Recurrent Neural Networks, especially Long Short \\nTerm Memory (LSTM) networks, are ideal in observing \\ntemporal behaviors and variations in the neural activity. \\nIncorporation of the RNNs into the hybrid model affords \\nidentification of dynamic connectivity pat- terns important in \\nelucidating ASD. The combination of CNN and RNN in the \\nproposed architecture takes advantage of spatial and temporal \\ncharacteristics of the neuroimaging data. CNNs capture spatial \\nrelations which are then provided to RNNs to take temporal \\nrelations. Such integration guarantees that the hybrid model \\ncaptures temporal dynamics of the interactions between \\ncortical regions and underlying neural activities, which will \\nimprove the accuracy of classifying ASD [5]. \\nPreprocessing remains integrated for sealing for \\nneuroimaging hybrid model outcomes. To overcome such \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n155\\n979-8-3315-4237-5/25/$31.00 ©2025 IEEE\\n2025 International Conference on Automation and Computation (AUTOCOM) | 979-8-3315-4237-5/25/$31.00 ©2025 IEEE | DOI: 10.1109/AUTOCOM64127.2025.10956945\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='issues, brain region alignment, motion correction, and \\ntemporal filtering are performed to reduce noise as well as \\ninter-subject variations. Moreover, other techniques such as \\nElastic Net Correlation Coefficient (EN-CC) used to feature \\nselection, allow elimination of features with low correlation \\nand many more, thus making the hybrid model the best [6]. \\nHaving reliable information for medical imaging is a \\nsignificant advantage when applying deep learning, though \\none drawback is the interpretability. To overcome these \\nchallenges, Hybrid CNN-RNN models are designed equipped \\nwith application of Explainable AI techniques including \\nLayer-wise Relevance Propagation (LRP) and Grad-CAM. \\nThese methods bring the focus to propose the most distinct \\nand crucially connected brain parts necessary for the model’s \\nprediction, which can be effectively explained to clinicians \\nabout the neural basis of ASD and help in clinical decision \\nmaking [7]. The purpose of this work is therefore to design a \\nCNN-RNN model for simultaneous analysis of neuroimaging \\ndata for ASD. Thus, the proposed framework aims at high \\nclassification accuracy and at the same time strives to generate \\nclinically relevant information by incorporating the advanced \\npreprocessing, spatial-temporal modeling, and explainable AI. \\nThis work contributes to the development of Advanced AI \\ntechniques dealing with neuroimaging and may have \\nimplications for the actual diagnosis of ASD and advancing \\nthe knowledge of its neurological basis [8]. \\nII. \\nRELATED WORKS \\nEarlier ASD detection based on neuroimaging data \\nemployed conventional ma- chine learning paradigms like \\nSupport Vector Machines (SVM) and Random Forests. They \\ninvolved significant feature extraction and dimensionality \\nreduction methods such as Principal Component Analysis \\n(PCA) and Independent Component Analysis (ICA), as most \\nneuroimaging data are high dimensional. Although useful up \\nto some extent, these kinds of approaches did not allow the \\nauthors to decompose the relationships within the data fields \\nbased on their spatial and temporal components [9]. Deep \\nlearning emerged and made analyzing neuroimaging data \\neven easier because it allowed for the automation of feature \\nextraction. CNNs have been used more often for the \\nclassification of ASD since the method is good at finding \\nspatial features of the brain activity and structure. For \\ninstance, there are ResNet , VGG models, which have large \\naccuracy rates for classifying ASD from MRI and fMRI data. \\nNevertheless, such approaches do not consider the time nature \\nof most neuroimaging studies [10]. \\nWhile CNNs are very efficient in extracting spatial \\nfeatures, one disadvantage is that spatial CNNs do not \\nincorporate a mechanism to encode temporal dynamics \\ninherent to most functional neuroimaging data including \\nfMRI. Temporal dimension turns out to be a critical factor in \\nstudying brain connectivity and neural activation. Therefore, \\nmore research is being done on hybrid solutions that will act \\nas temporal modeling capabilities [11]. \\n RNNs, especially the LSTM, has been used to process \\ntemporal data in neuroimaging research. LSTMs exploit \\ntemporal dependence structure in fMRI data, making it ideal \\nfor analysis in time-series neural activity. Research \\nincorporating RNNs in diagnosing ASD has provided positive \\noutcomes, and incorporating the CNN for spatial feature \\nextraction can improve their performance [12]. CNN- RNN \\nhybrid models have become the standard approach for \\ncombining spatial and temporal analysis models. These \\nmodels, which integrate the advantages of CNNs and RNNs, \\nafford extensive feature analysis over neuroimaging data. The \\nexperiments have demonstrated that the proposed hybrid \\narchitectures enhance classification accuracy and reveal the \\nspatial and temporal patterns of brain activity. For example, \\nthe combination of CNN and LSTM has been revealed to \\nafford better performance in ASD classification compared to \\nsingle models of CNN or RNN [13]. There is much interest in \\ncombining fMRI, sMRI, and DTI, which are imaging \\ntechniques yielding data from a patient’s brain. Continuous \\nintegration of functional, structural, and connectivity \\ninformation improves the diagnostic yield of multimodal \\nmethods. The hybrid CNN-RNN structures are more relevant \\nfor solving problems with multimedia data since it is possible \\nto use the benefits of each modality in parallel [14]. \\nThe neuroimaging datasets of high dimension typically \\nsuffer from noisy or less relevant features, so the process of \\nfeature selection becomes mandatory. There is the use of \\nfeatures like Elastic Net Correlation Coefficient (EN-CC) \\ntogether with Recursive Feature Elimination (RFE) to \\nimprove the model. The latest heterogeneous CNN-RNN \\nmodel-based investigations integrate feature se- lection \\ntechniques in an attempt to avoid high dimensionality, thus \\nfocusing the model towards the important connectivity \\nfeatures of the brain [15]. Data preprocessing is another \\nimportant step in filtering and enhancing neuroimaging data \\nso that they can be easily interpreted. Standardization and re- \\nduction of noise can be performed through the utilization of \\nmethods such as 3D Brain Region Alignment (3D-BRA) and \\nmotion correction. A number of temporal filtering methods \\nare then employed to remove artifacts from fMRI data while \\npreserving important connectivity patterns within the \\nbrain[16]. These have laid a strong background for hybrid \\nCNN-RNN models to reduce robustness analysis on. \\nWhen deep learning is applied to medical experiences and \\napplications, one major concern is how to explain the model’s \\noutput. Addressing this problem, some novel approaches, \\ncalled Explainable AI methods, including Grad-CAM and \\nLayer-wise Relevance Propagation (LRP), have been \\ndeveloped. Such techniques serve to pinpoint areas and \\nconnections most consistent with the model predictor, and \\nmay be useful to specialists[17-19]. The CNN-RNN models \\nare entirely accurate for ASD diagnosis yet their clinical \\ninterpretability \\nremains \\nhigh \\nwhen \\nintegrated \\nwith \\nexplainability tools. However, there are several issues with \\ndeep learning and neuroimaging even today. Some of the \\nchallenges they were posed stay relevant, including limited \\navailability of data, inter-subject variability, and the call for \\nmore interpretable models. Multimodal analysis with CNN–\\nRNN systems and with more involved preprocessing is \\nsuggested as the promising direction for further research \\nMoreover, it is feasible to enhance the quantity and variability \\nof data through such a possibility as using larger sized data \\nsets and from other sources, e.g., ABIDE or SFARI. \\nTherefore, the future studies need to aim at various directions \\nfor the improvement of the hybrid architectures and the \\nintegration of XAI in order to establish connections between \\nnew and innovative algorithms’ solutions and feasible clinical \\nuse. \\nIII. \\nPROPOSED METHODOLOGY \\nTo this end, a novel multimodal brain image analysis \\nframework using both the CNNs and RNNs has been proposed \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n156\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 2}, page_content='to classify patients with ASD. CNN is used for finding spatial \\nfeatures and RNN for temporal dependencies of the desired \\nobject. Such a model guarantees the significant feature \\ncoverage and uses the distinctive advantages of both networks. \\nThis is developed in Python with TensorFlow/ PyTorch and \\nwith MATLAB for data preprocessing and visualization in \\nTableau. \\n \\nFig. 1. Proposed Flowchart for Multimodal Analysis of Autism Spectrum \\nDisorder \\nA.  Data Preprocessing and Background Subtraction \\nWith preprocessing, neuroimaging data is well prepared \\nfor analysis prior to any other procedure being undertaken. \\nSubstantial noise and extraneous information in fMRI images \\nare abstracted using 3D Brain Region Alignment (3D-BRA) \\nand motion correction They include What was done during the \\npreprocessing steps: Towards this goal, a temporal mean filter \\nis used to principally filter out noise as a precursor to \\nextracting significant signals. \\nB (x, y ,t)=f (x, y ,t)− T/1 ∑்\\n௧ୀଵ\\nf(x,y,t)                               (1) \\nwhere f (x, y, t) represents the pixel intensity at time t, and \\nT is the total number of frames.  \\nMotion correction and spatial normalization align all scans \\nto a common anatomical template.. \\nB. Feature Extraction with CNN \\nThe CNN module is the layer that helps identify the spatial \\nfeatures from the processed neuroimaging raw data. ResNet-\\n50 is used here because it is able to perform residual learning \\neffectively in a comparatively low number of layers. The \\nconvolutional layers analyze small areas of brain activation. \\n                             y = ReLU (W · x + b)    \\n (2) \\nwhere x is the input image, W are the weights of the \\nconvolutional filter, and b is the bias. Subsequent to \\nconvolution to help decrease the spatial dimensions while \\npreserving significant features, there is max pooling. \\nC. Temporal Analysis with RNN \\nThe extracted spatial features are flattened and fed into the \\nRNN module, specif- ically a Long Short-Term Memory \\n(LSTM) network. LSTM is ideal for capturing temporal \\ndependencies in fMRI data. The LSTM cell operations are \\ndefined as follows [?]: \\nft = σ(Wf · [ht−1, xt] + bf )  \\n (3) \\nit = σ(Wi · [ht−1, xt] + bi)  \\n (4) \\nCt = ft ∗ Ct−1 + it ∗ tanh(WC · \\n[ht−1, xt] + bC)  \\n  \\n    (5) \\not = σ(Wo · [ht−1, xt] + bo)  \\n (6) \\nht = ot ∗ tanh(Ct)  \\n \\n (7) \\nHere   f_t, i_t, and o_t represent the forget, input, and \\noutput gates, respectively, while Ct is the cell state.  \\nD.  Multimodal Integration \\nIn this case, the fMRI, MRI, and DTI data are integrated \\nby combining the features of each modality together in a \\ncascading manner and then fed into the RNN. This integration \\ntakes advantage of common information between two \\nmodalities to increase the classification rate. To ensure each \\nmodality is equally represented, a weighting mechanism is \\nadopted: \\nF = αFfMRI + βFMRI + γFDT I  \\n \\n(8) \\nWhere α, β, γ are the weights assigned to each modality’s \\nfeatures. \\nE. Model Training and Evaluation \\nThe hybrid model is trained using the Adam optimizer to \\nminimize the cross- entropy loss: \\nL=−\\nଵ\\nே(+ܽ) 1 ∑ N  [y i  log ( y ^  i  )+(1−y i  )log(1− y \\n^  i  )]       (9) \\nwhere ݕ ݅  is the true label, and ݕ ^ ݅  is the predicted \\nprobability. The model undergoes Stratified Group Cross-\\nValidation (SGCV) to ensure reliable performance evaluation \\nacross subject groups. Metrics such as accuracy, precision, \\nrecall, and F1-score are computed. \\nF. Software and Implementation Details \\nx Programming Environment: TensorFlow and PyTorch \\nbased on Python for model development. \\nx Preprocessing Tools: MATLAB for motion correction \\nand background subtraction. \\nx Visualization: Tableau for presenting results and \\ninsights. \\nx Hardware: The datasets for training include NVIDIA \\nGPUs for accelerated training. \\nx Libraries: For data handling and preprocessing, Python \\nlibraries including NumPy, Pandas, OpenCV, and \\nScikit-learn are used. \\n \\nG. Dataset \\nThe ASD datasets employed in the present paper are \\nbenchmark datasets such as ABIDE I, ABIDE II, HCP, and \\nSFARI, which contain significant ASD data for neuroimaging \\ninvestigation. In ABIDE I and ABIDE II, data includes \\nresting- state fMRI and structural MRI from individuals with \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n157\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 3}, page_content='ASD and from typically developing controls, arriving from \\nmore than a thousand participants across several sites. The \\nreleased HCP dataset provides RS-fMRI, task-fMRI, \\nStructural MRI, and DTI data from more than 1200 healthy \\nadults. SFARI has a vast resource of gene, clinical, and \\nneuroimaging \\ndata \\nestablishing \\nthe \\nneuroanatomical \\nfoundation of ASD as well as about thousands of participants. \\nThe availability of these datasets provides a rich database for \\nunderstanding BCT profiles, structural alterations, and \\ngenomic architecture underlying ASD, as well as for training \\nand testing complex machine-learning algorithms for ASD \\nidentification. \\nSo as to allow for reproducing the results, the specifics of \\nthe proposed technique have been described in detail. This \\ninclude TensorFlow and PyTorch for model compiling, \\nMATLAB for preprocessing data and Tableau for visualizing \\nthe data. The preprocessing pipeline, containing such tools as \\n3D BRA and Elastic Net Correlation Coefficient for feature \\nselection, \\nis \\nexplained in detail. \\nIn \\naddition, \\nthe \\nhyperparameters used in training and evaluation are described \\nnext. In subsequent versions of this paper, there will be a \\nsecondary repository with the code, pretrained models, and \\nlinks to the datasets for research purposes. \\nIV. \\nRESULT AND DISCUSSION \\nThe hybrid CNN-RNN model was evaluated on four well-\\nestablished neuroimaging datasets: ABIDE I, ABIDE II, \\nHuman Connectome Project (HCP), and the Simons \\nFoundation Autism Research Initiative (SFARI). Data \\ncollection was obtained from online surveys. This section \\nprovides the outcome of the method suggested in this paper \\nand the analyses of the findings, complemented by tables to \\nenhance readability. \\nTABLE I.  \\nCLASSIFICATION ACCURACY OF THE HYBRID CNN-RNN \\nMODEL \\nDataset \\nAccuracy (%) \\nPrecision (%) \\nRecall \\n(%) \\nF1-\\nScore \\n(%) \\nABIDE I \\n96.5 \\n95.8 \\n96.2 \\n96.0 \\nABIDE II \\n97.3 \\n97.0 \\n96.9 \\n97.0 \\nHCP \\n98.1 \\n98.0 \\n97.8 \\n97.9 \\nSFARI \\n99.0 \\n98.9 \\n99.1 \\n99.0 \\n \\nx ABIDE I: The Autism Brain Imaging Data Exchange \\nI. \\nx ABIDE II: The Autism Brain Imaging Data Exchange \\nII. \\nx HCP: Human Connectome Project. \\nx SFARI: Autism Research Initiative of the Simons \\nFoundation. \\nThe proposed hybrid CNN-RNN yielded marvelous \\nclassification accuracy ranging from 96.5% to 99.0% across \\nall datasets. Among the various datasets used in the study, the \\nhighest accuracy was observed in the SFARI dataset. This can \\nbe attributed to the larger sample size and diverse \\nrepresentation of data, which demonstrates the scalability of \\nthe proposed model. High precision and recall values justify \\nthe referability of the model to accurately classify ASD \\ninstances and to reduce the incidence of Type I and II errors. \\n            \\n \\nFig. 2. Bar graph showing the classification accuracy of the hybrid \\nCNNRNN   model on different datasets \\nThe proposed hybrid CNN-RNN model yields robust \\nclassification of ASD. Across multiple datasets, with \\nespecially remarkable accuracy on the SFARI dataset. \\nTABLE II.  \\nIMPACT OF FEATURE SELECTION ON MODEL \\nPERFORMANCE \\nModel \\n \\nAccuracy Without \\nFeature Selection \\n(%) \\nAccuracy With \\nFeature Selection \\n(%) \\nResNet-50 (CNN) \\n85.2 \\n96.5 \\nVGG-16 (CNN) \\n87.1 \\n97.3 \\nInception-v3 (CNN) \\n88.3 \\n98.1 \\nHybrid \\nCNN-RNN \\n(Proposed) \\n89.5 \\n99.0 \\nApplication of Elastic Net Correlation Coefficient (EN-\\nCC) for feature selection enhanced the accuracy by about 10-\\n15% in all the investigated architectures. This result indicates \\nthe crucial need for dimensionality reduction and feature \\nselection, especially when dealing with multifaceted \\nneuroimaging datasets \\n \\n \\nFig. 3. When considering feature selection, we obtain the following bar \\ngraph of the model performance: All bars on the left side of the figure \\nindicate that the presented results have been obtained without feature \\nselection, while all bars on the right side have been obtained with feature \\nselection. \\n \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n158\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='To think that feature selection is still a big deal in \\nenhancing the performance of a model, and the application of \\nEN-CC significantly improved the classification rate of the \\nCNN-RNN hybrid model. \\nTABLE III.  \\nCOMPARISON WITH STATE-OF-THE-ART MODELS \\nModel \\nAccuracy \\n(%) \\nPrecision \\n(%) \\nRecall \\n(%) \\nF1-\\nScore \\n(%) \\nSVM + PCA \\n81.2 \\n79.5 \\n80.1 \\n79.8 \\nCNN (ResNet-\\n50) \\n96.5 \\n95.8 \\n96.2 \\n96.0 \\nCNN + RNN \\n(LSTM) \\n97.8 \\n97.5 \\n97.6 \\n97.5 \\nHybrid CNN-\\nRNN \\n(Proposed) \\n99.0 \\n98.9 \\n99.1 \\n99.0 \\n The proposed hybrid CNN-RNN model yielded better \\nresults than earlier methods, particularly in accuracy and F1-\\nscores. By combining CNNs to ana- lyze spatial features and \\nRNNs to determine temporal features, the model was able to \\nidentify intricate patterns in the neuroimaging datasets. The \\ncomparison with traditional models such as SVM+PCA  \\n \\nFig. 4. The bar graphical representation showing a comparison between the \\nmodel-M3 and the model-M4 on the basis of Accuracy, precision, recall, and \\nF1 score. \\nTABLE IV.  \\nEXPLAINABILITY ANALYSIS USING GRAD-CAM \\nDataset \\nKey Brain Regions \\nIdentified \\nRelevance to ASD \\nABIDE I \\nPrefrontal Cortex, \\nTemporal Lobe \\nAssociated with social and \\ncommunication skills. \\nABIDE II \\nAmygdala, \\nHippocampus \\nKey roles in emotion and \\nmemory processing. \\nHCP \\nDefault Mode Network \\n(DMN) \\nCritical for self-referential \\nthought. \\nSFARI \\nSocial Cognition \\nCircuitry, Insula \\nRelevant to empathy and \\nemotional regulation \\nFurthermore, the Grad-CAM analysis offered insights into \\nthe brain regions from which the model draws its decisions. \\nFor instance, the Default Mode Net- work (DMN) detected in \\nthe HCP dataset corresponds to regions frequently implicated \\nin ASD individuals, particularly in terms of self-related \\nprocessing. The prefrontal cortex and temporal lobe were \\nmentioned most often, as expected, due to social and \\ncommunicative dysfunction in the disorder \\nThe hybrid CNN-RNN used for sound analysis is coupled \\nwith Grad-CAM for better explain- ability and it identified key \\nregions of the brain related to ASD including the prefrontal \\ncortex, the amygdala, and the Default Mode Network (DMN). \\nThis kind of interpretability is fundamental to reconcile AI-\\ndriven predictions with clinical practice. The approach \\nguarantees the effective utilisation of the multitask \\npreprocessing which focuses on machines to apply different \\npreprocessing techniques and also ensures merging of \\nmultimodal neuroimaging data; In other words, the \\npreprocessing stage plays a crucial role of ensuring effective \\nfeature extraction throughout the diagnostic procedure as a \\nresult of advanced spatial–temporal analyses. The study \\nstrengths also relate to the clinical applicability for \\ninterpersonal psychotherapy and the research utility of the \\nproposed framework. \\nV. \\nCONCLUSION \\nThis paper proposes a new CNN–RNN system for the \\nmultimodal investigation of ASD based on neuroimaging \\ndata. An important facet of the proposed method- ology \\nprovides compatibility with spatial and temporal feature \\nextraction while overcoming the problems introduced by \\nfMRI, structural MRI, and DTI streams given the high \\ndimensionality of the data. Program enhancements like 3D \\nBrain Region Alignment (3D-BRA) and Adaptive Temporal \\nFiltering pre-processing essentially improve the data quality \\nof this model. Overall, the proposed hybrid CNN-RNN model \\nexhibited high classification accuracy of above 95 percentage \\nin different data sets, that are ABIDE I, ABIDE II, HCP and \\nSFARI. The inclusion and exclusion step based on Elastic Net \\nCorrelation Coefficient (EN-CC) increased the model \\naccuracy by shrinking the model down to only 10, 15 or 20 \\nimportant features out of thousands of gene features that are \\nrelevant for separating cancerous tissues from normal tissues. \\nAdding methods like Grad-CAM as a part of explainable AI, \\nthe authors were able to determine that the model focused on \\nthe most critical regions of the brain related to ASD and that \\nthe results will be beneficial for clinicians to make decisions \\nbased on the AI model’s predictions. The findings show that \\nthe application of the combined deep learning architecture can \\nenhance the significance of ASD diagnosis and boosted clf by \\nmaking it more accurate and easier to understand. Through \\nintegration \\nof \\nmultimodal \\nneuroimaging \\ndata \\nand \\ndevelopment of highly efficient computational methods, this \\napproach provides a possibility to improve the clinical \\npractice and reveal the mechanisms of ASD’s development. \\nKalecki-type equations and nonlinear least squares estimation \\nused in this paper can be subjected to more extensive \\nvariations and use a much broader range of data from different \\npopulations in future studies. \\nThe study also recognizes that the Crowdbase was \\nsomewhat limited by the kind of datasets available and the \\nlack of much greater diversity and representation. Despite \\nusing benchmark datasets including ABIDE I and II, HCP, \\nand SFARI with their distinct neuroimaging and behavioral \\ncontributors, more optimization is required. In future work, we \\nplan to add more datasets from other demographic and clinical \\npoint of view. The extension of this study will increase the \\nstability and applicability of the enshrouded model to other \\ndifferent groups and diseases. \\nREFERENCES \\n[1] \\nAlhussen, A., Vinoth, N., Jenis, M.R.A., Surendran, S., Ganesh, V.D., \\nThangaraj, S.J.J.: Development of weighted ensemble deep learning \\nnetwork for surface rough- ness prediction and flank wear \\nmeasurement. \\nJ. \\nMater. \\nEng. \\nPerform. \\n(2024). \\nhttps://doi.org/10.1007/s11665-024-09726-7 \\n[2] \\nDeepa, S.N., Devi, B.A.: A survey on artificial intelligence approaches \\nfor medical image classification. Indian J. Sci. Technol. 4(11), 1583–\\n1595 (2011). https://doi.org/10.17485/ijst/2011/v4i11/30291 \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n159\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 5}, page_content='[3] \\nDhamale, T.D., Bhandari, S.U., Harpale, V.K., Nandan, D.: Autism \\nspec- trum disorder detection using parallel deep convolution neural \\nnetwork and generative adversarial networks. Trait. du Signal 41(2), \\n643–652 (2024). https://doi.org/10.18280/ts.410208 \\n[4] \\nEslami, T., Saeed, F.: Auto-asd-network: A technique based on deep \\nlearning and support vector machines for diagnosing autism spectrum \\ndisorder using fmri data. In: ACM-BCB 2019 - Proc. 10th ACM Int. \\nConf. Bioinformatics, Comput. Biol. Heal. Informatics. pp. 646–651 \\n(2019). https://doi.org/10.1145/3307339.3343482 \\n[5] \\nGanesh, D.V., Bommi, R.M.: Neural network based predictive analysis \\nof sur- face roughness using bayesian regularization in turning of \\nmonel k500. In: 6th Int. Conf. Inven. Comput. Technol. ICICT 2023 - \\nProc. \\npp. \\n448–452 \\n(2023). \\nhttps://doi.org/10.1109/ICICT57646.2023.10134220 \\n[6] \\nGanesh, V.D., Bommi, R.: Flankpix: An image segmentation algorithm \\nfor flank wear analysis in monel k500 turning. Trait. du Signal 41(3), \\n1625–1631 (2024). https://doi.org/10.18280/ts.410349 \\n[7] \\nJung, Y.J., Han, S.H., Choi, H.J.: Explaining cnn and rnn using selec- \\ntive layer-wise relevance propagation. IEEE Access 9, 18670–18681 \\n(2021). https://doi.org/10.1109/ACCESS.2021.3051171 \\n[8] \\nKe, F., Choi, S., Kang, Y.H., Cheon, K.A., Lee, S.W.: Exploring the \\nstructural and strategic bases of autism spectrum disorders with deep \\nlearning. \\nIEEE \\nAccess \\n8, \\n153341–153352 \\n(2020). \\nhttps://doi.org/10.1109/ACCESS.2020.3016734 \\n[9] \\nNandhini, T.J., Thinakaran, K.: An improved crime scene detection \\nsystem based on convolutional neural networks and video surveillance. \\nIn: 2023 5th Int. Conf. Electr. Comput. Commun. Technol. ICECCT \\n2023. \\npp. \\n1–6 \\n(2023). \\nhttps://doi.org/10.1109/ICECCT56650.2023.10179852 \\n[10] Saleh,  A.Y.,  Chern,  L.H.:  Autism  spectrum  disorder  classification  \\nus- ing deep learning. Int. J. online Biomed. Eng. 17(8), 103–114 \\n(2021). https://doi.org/10.3991/ijoe.v17i08.24603 \\n[11] Shah, S.K., Kumar, N.S., Gurumoorthi, E., Shnain, A.H., Kumar, A., \\nNandhini, T.J.: A detailed analysis of speculative based algorithm \\nimplementation for modish house. In: 2024 4th Int. Conf. Adv. \\nComput. \\nInnov. \\nTechnol. \\nEng. \\npp. \\n984–987 \\n(2024). \\nhttps://doi.org/10.1109/ICACITE60783.2024.10616747 \\n[12] Shao, L., Fu, C., You, Y., Fu, D.: Classification of asd based on fmri \\ndata with deep learning. Cogn. Neurodyn. 15(6), 961–974 (2021). \\nhttps://doi.org/10.1007/s11571-021-09683-0 \\n[13] Sherkatghanad, Z., et al.: Automated detection of autism spectrum \\ndisorder us- ing a convolutional neural network. Front. Neurosci. \\n13(January), 1–12 (2020). https://doi.org/10.3389/fnins.2019.01325 \\n[14] Tang, M., Kumar, P., Chen, H., Shrivastava, A.: Deep multimodal \\nlearn- ing for the diagnosis of autism spectrum disorder. J. Imaging \\n6(6) (2020). https://doi.org/10.3390/jimaging6060047 \\n[15] Yang, X., Sarraf, S., Zhang, N.: Deep learning-based framework for \\nautism func- tional mri image classification. Journal of the Arkansas \\nAcademy of Science 72 (2018) \\n[16] Jahani, A., Jahani, I., Khadem, A. et al. Twinned neuroimaging \\nanalysis contributes to improving the classification of young people \\nwith autism spectrum disorder. Sci Rep 14, 20120 (2024). \\nhttps://doi.org/10.1038/s41598-024-71174-z \\n[17] Jainy Sachdeva, Riyaansh Mittal, Jiya Mehta, Riya Jain, Anmol \\nRanjan, Resolving autism spectrum disorder (ASD) through brain \\ntopologies using fMRI dataset with multi-layer perceptron (MLP), \\nPsychiatry Research: Neuroimaging, Volume 343, 2024, 111858, ISSN \\n0925-4927, https://doi.org/10.1016/j.pscychresns.2024.111858. \\n[18] A. K. Singh, I. R. Khan, S. Khan, K. Pant, S. Debnath, and S. Miah, \\n“Multichannel CNN Model for Biomedical Entity Reorganization,” \\nBioMed \\nResearch \\nInternational, \\nvol. \\n2022, \\n2022, \\ndoi: \\n10.1155/2022/5765629. \\n[19] K. Maithili et al., “Optimized CNN Model for Diabetic Retinopathy \\nDetection and Classification,” International Journal of Intelligent \\nSystems and Applications in Engineering, vol. 11, no. 7s, pp. 317–331, \\n2023, [Online]. Available: Scopus Link. \\n \\n \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n160\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='Preprocessing FMRI Data In SPM12\\nFatima Ez-zahraa Bazay\\nLRIT - CNRST URAC 29\\nRabat IT Center, FS\\nMohammed V University in Rabat\\nMorocco\\nFatima Zahra Benabdallah\\nLRIT - CNRST URAC 29\\nRabat IT Center, FS\\nMohammed V University in Rabat\\nMorocco\\nAhmed Drissi El Maliani\\nLRIT - CNRST URAC 29\\nRabat IT Center, FS\\nMohammed V University in Rabat\\nMorocco\\nAbstract—\\nFunctional magnetic resonance imaging (fMRI) has ended\\nup the most famous technique for imaging brain functions.\\nCurrently, there is a giant range of software packages for the\\nanalysis of fMRI data, such as AFNI, SPM, FreeSurfer, FSL,\\nfMRIPrep and Workbench.In this work, we propose an fMRI\\npreprocessing on SPM applied to ABIDE II database and we\\nevaluate this processing by applying an MLG.\\nIndex\\nTerms—Functional\\nmagnetic\\nresonance\\nimaging\\n(fMRI),\\nPreprocessing,\\nSegmentation,Slice\\ntiming,coregistration,normalization,smoothing,Realignment\\n,ABIDE II,NIFTI.\\nI. INTRODUCTION\\nFunctional\\nMagnetic\\nresonance\\nimaging\\n(fMRI),\\na\\ncontemporary imaging technique is an effective non-invasive,\\nsecure device used to study the brain’s features primarily\\nbased on a measure of the brain neural activation. [1]\\nThe preprocessing of raw fMRI records is a crucial step\\nearlier than performing further analyses for the following\\nreasons. First, fMRI measures spontaneous fluctuations of\\nblood-oxygen-level dependent (BOLD) signals associated with\\nneuronal activities. However, BOLD indicators comprise non-\\nneuronal contributions, such as head motion, physiological\\ncontributions, tissues outside the scope of interest, and\\nMRI-induced artifacts, as nicely as neuronal signals.( [2], [3],\\n[4])The non-neuronal aspects of BOLD signals complicate\\nthe interpretation of fMRI signals. Secondly, the remarkable\\nfMRI facts largely rely upon the image acquisition parameters.\\nDepending on the acquisition parameters, different MRI data\\nmight have a distinctive variety of intensity values, matrix\\nsizes, and orientations. So preprocessing steps for fMRI data\\nare required to care for these problems.\\nIn\\nprevious\\nstudies,\\nresearchers\\nhave\\ndeveloped\\nfreely\\naccessible\\nopen-source\\nneuroimaging\\ndata\\npreprocessing\\ntools,\\nsuch\\nas\\nstatistical\\nparametric\\nmapping\\n(SPM)1,\\nfMRIPrep [5],Workbench [6] ,FreeSurfer [7],FMRIB software\\nlibrary (FSL) [8], and analysis of functional neuroimages\\n(AFNI) [9] . These are broadly used software program\\ntools, but everyone employs a specific data preprocessing\\nstrategy. FreeSurfer is appropriate for surface data and gives\\na fully automated command line-based pipeline.SPM and\\nIdentify applicable funding agency here. If none, delete this.\\n1http://www.fil.ion.ucl.ac.uk/spm/\\nFSL grant fully intuitive graphical user interface (GUI)-based\\npreprocessing pipelines and are suitable for volume data.\\nAFNI and Workbench system each volume and surface data.\\nThere are many strides in a given preprocessing pipeline,\\nincluding\\ndiscipline\\ninhomogeneity\\ncorrection,\\nmotion\\ncorrection, registration, and segmentation steps. This pre-\\nprocessing of FMRI images is intended to be used in several\\nworks, such as the detection of Autism Spectrum Disorders.\\n[10], [11], [12]\\nHere, we are interested in the different stages of FMRI\\ndata preprocessing in SPM. This preprocessing is applied to\\nthe ABIDE II database and has not yet been done on this\\ndatabase. And we evaluate our processing using a general\\nlinear model (GLM) [13], which is a tool for statistical\\nanalysis of fMRI data, which allows us to evaluate the\\nactivated brain areas by defining and estimating contrasts that\\nenable us to determine which are the significantly activated\\nregions detected by the model.\\nII. METHODS\\nIn this paper, we will apply the FMRI data preprocessing\\non SPM(Statistical Parametric Mapping).\\nStatistical Parametric Mapping is a statistical technique to\\ndecide the difference in recorded brain activity by construction\\nof spatially prolonged statistical processes used to check\\nhypotheses during the practical neuroimaging experiments\\nand the usage of neuroimaging applied sciences [14]. These\\nthoughts have been blanketed in software that is known as\\nSPM. Brain imaging information sequences are applied to\\nthe SPM software program for analysis. The sequences can\\nbe a series of images from different subjects or time series\\nfrom identical subjects. SPM is made freely on hand to\\nthe neuroimaging community throughout laboratories. The\\nsoftware is an entire evaluation package deal that represents\\nthe implementation of the theoretical concept of Statistical\\nParametric Mapping.\\nThe SPM software is a suite of MATLAB (MathWorks,\\nInc) functions and subroutines with some externally compiled\\nC routines. The current version of SPM is SPM12.\\nFMRI data processing in SPM includes two types of\\nprocessing; Temporal Processing, which includes slice timing\\noperation, and also has Spatial Processing, which provides\\nfor Realign, Co-registration, Segmentation, Normalization,\\n978-1-6654-5276-2/22/$31.00 © 2022 IEEE\\n2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM) | 978-1-6654-5276-2/22/$31.00 ©2022 IEEE | DOI: 10.1109/WINCOM55661.2022.9966463\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='and Smoothing.\\nThe ABIDE database consists of the first (ABIDE I) and\\nsecond (ABIDE II) data releases provided by the University of\\nUtah. Seventeen sites provided 1112 individuals in the ABIDE\\nI version, including 539 autism patients and 573 controls.\\nThe ABIDE II data was compiled from 19 sites, a total\\ndataset of 1114 individuals, including 521 individuals with\\nautism and 593 controls. Site-specific details on diagnostic\\ncriteria and more detailed characterization For clinical and\\nbehavioral phenotypes, see [15]. The current study is based\\non the ABIDE II database only.\\nA. Slice Timing Correction\\nA total volume is acquired in a TR ranging, in general,\\nfrom a few milliseconds to several seconds. A time lag\\nbetween slices is, therefore, present, depending on the\\nposition of the slice within the volume and the number\\nof cuts. As a result, this problem can cause a time lag in\\nthe measured hemodynamic response between the different\\nslices, compromising the reliability of the statistical results in\\ndetecting activations.\\nThe impact of this time lag is all the more important for a\\nlong TR and non-interlaced acquisitions. To compensate for\\nthis time lag, one solution consists of a time interpolation to\\nbring all the slices back to a common time, which is the time\\nof acquisition of one of the slices of the volume (reference\\nslice). The reference cut is generally the central cut in the\\nvolume to reduce the interpolation to a TR/2. Subsequently,\\nwe could consider that all the slices of a volume were\\nacquired at the same time [16].\\nAnother time lag model was proposed to avoid modifying\\nthe data before statistical analysis. This involves including\\na regressor in the linear model GLM, whose principle is\\nbased on the time derivatives of the HRF [17]. Generally,\\nthis method is used for a long TR (greater than 2 seconds) to\\neliminate any artefactual variance components.\\nB. Realignment\\nThe subject’s movement during the experiment is one of\\nthe significant artifacts and inevitable, given an acquisition\\ntime of several minutes. A rigid readjustment allows partial\\ncorrection of motion artifacts. It consists first of an estimate\\nof the six translation parameters (”right,” ”forward,” ”up,”\\nin mm) and rotation (”pitch,” ”roll,” ”yaw,” in radians) of\\nthe volumes acquired for the subject relative to a reference\\nvolume (usually the first volume). Then, an interpolation\\nconcerning the estimate makes it possible to readjust the\\nvolumes on the reference volume.\\nNote that there is an interdependence between the motion\\ncorrection and slice-timing steps, and the order in which\\nthese two pre-processing operations are performed impacts\\nthe sensitivity of the results. Usually, slice timing is applied\\nbefore motion correction when slices are acquired in an\\ninterleaved way so as not to resample voxels from one\\nslice to an adjacent slice acquired at a distance in time.\\nConversely, if the acquisition was performed in sequential\\nmode, the motion correction is applied first, not temporarily,\\nto interpolate signals from distant brain regions.\\nC. Coregistration\\nIn fMRI, this step is usually performed to align the func-\\ntional images with each other on the anatomical image to\\nallow the final individual activations to be represented on a\\nhighly resolved image. The anatomy is often readjusted to the\\naverage functional volumes when one wishes to remain in the\\nnative space without disturbing the functional data. As the\\nmodalities are different, the similarity measurement method is\\nusually based on mutual information [18], as anatomical and\\nfunctional images are often weighted differently.\\nD. Segmentation\\nThe segmentation step creates, from a single image, a white\\nmatter image, a grey matter image, and a cerebrospinal fluid\\n(CSF) image. These images can display a 3D rendering of the\\nactivations overlaid on the anatomy. The segmentation in SPM\\nis based on a modified mixed Gaussian model. The Bayesian\\nrule assigns the probability for each voxel belonging to a tissue\\nclass by combining the probability of belonging to that tissue\\nclass and the prior probability derived from probability maps\\nfrom many subjects.\\nE. Normalization\\nThis operation is crucial for group analysis. Indeed, two\\ninter-individual variability sources must be considered when\\nconducting a group study: anatomical variability (shape, size\\nof the brain) and functional variability (localization of regions\\nactivated during stimulation). Spatial normalization allows all\\nimages to be placed in the same reference space. (reference\\nor template) to compare the functional fMRI results obtained\\nfor different subjects.\\nSeveral stereotactic references are used in neuroimaging\\nstudies: the one based on the Talairach and Tournoux atlas\\n[19], and that of the Montreal Neurological Institute, of\\nwhich several versions have been developed [20] and the\\nAnatomical Automatic Labeling atlas (AAL) [21].\\nThese two first atlases differ radically in their conception, the\\nfirst being based on the post-mortem study of the brain of a\\nsingle individual, the second being constructed by averaging\\nT1-weighted MRI images of dozens of brains (305 in the first\\nversion [22], then 152 in the currently most used version).\\nOnce spatial normalization has been achieved, the activations\\ncan be localized in the common space and compared between\\ndifferent subjects. It is then assumed that voxel to voxel\\ncorrespondence exists for all subjects, which allows the\\ncalculation of group of group statistics.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 2}, page_content='F. Smoothing\\nAn inherent problem with fMRI acquisition is that adjacent\\nvoxels do not have independent signals from each other. As\\na result, the signals have spatial correlations that are difficult\\nto estimate. The functional images undergo spatial smoothing\\nto impose known and controlled correlation characteristics.\\nThis smoothing consists of convolution with an isotropic\\nGaussian filter with a Full Width at Halh Maximum (FWHM),\\ngenerally fixed at 2 or 4 times a voxel size. However, this\\nchoice of Gaussian kernel must be adapted to the study being\\nconducted [23].\\nThis\\nstep\\nmakes\\nit\\npossible\\nto\\nminimize\\nthe\\nresidual\\nrealignment and normalization faults and to increase the\\nsignal-to-noise ratio. On the other hand, the accuracy of the\\nlocalization of activations may be reduced, especially by\\nextension.\\nG. General linear model (GLM)\\nThe GLM is a fundamental tool for the statistical analysis\\nof functional data. The linear approach has appeared in\\nthe literature through the parametric statistical map method\\n[24], [25], [26]. This method has become a reference in\\nthe univariate analysis of functional data. It consists in\\nmodeling the observed time series for each voxel by a linear\\ncombination of explanatory functions with a residual error\\nterm.\\nMore precisely, this model is a multiple linear regression in\\nwhich the BOLD signal yj,t is acquired at voxel j at time t.\\nIt is defined through three types of signals:\\n•The useful signal, which is assumed to be the sum of the\\neffects induced by the different conditions of the experimental\\nparadigm M;\\n•Low-frequency drifts due to physical (thermal noise, etc.),\\nphysiological (folding of respiratory and cardiac frequencies),\\nand effects induced by the interaction between movements\\nand the static magnetic field;\\n•The noise expressing the modeling error, which is generally\\nassumed to be Gaussian and dependent on voxel j.\\nyj,t = Xβj + ej =\\nR\\nX\\nr=1\\nβr\\nj + ej,t\\n(1)\\nwhere yj,t is a vector of dimension N (the number of\\nscans) representing the preprocessed BOLD signal at voxel j\\nmeasured at time t,with j = n∗TR, n ∈[1, N].\\nThe wanted signal, as well as the low-frequency drift, is\\nrepresented by X∗βj,where X is a matrix of dimension N x\\nR called the drawing matrix, and βj are the linear regression\\ncoefficients (vector of dimension R). Finally ej represents the\\nresidual error.\\nThe estimation of the parameters βj is done in the classical\\nway by least squares adjustment. This minimizes the squared\\nerror between the signal yj,t measured at time t and the\\nestimated one X∗bβj. The estimation of the parameters βj is\\ntherefore performed independently for each voxel of the brain\\n(massively univariate estimation). Let bβj be the value of βj\\nadjusted by the least squares method. The expression for bβj\\nis then written :\\nbβj = (XT X)\\n−1XT yj,t\\n(2)\\nThe estimated parameter bβj thus follows a normal distribution,\\ncentered on βj and with matrix of covariance Var[bβj]=\\nσ2\\nj (XT X)\\n−1. The estimator of the noise variance σ2\\nj at voxel\\nj is estimated by :\\nbσ2\\nj = (yj −X bβj)T (yj −X bβj)\\nN −R\\n(3)\\nAfter estimating the linear regression coefficients, statistical\\ntests should be chosen according to the nature and distribution\\nof the variables studied. A test is used to test whether a con-\\ntrast (i.e., a linear combination) of parameters is significantly\\ndifferent from a valued, under the following two assumptions:\\nH0 : cT βj = d\\n(4)\\nH0 : cT βj ̸= d\\n(5)\\nWhere cT = [cr; r = 1 : R] is a row vector determining\\nthe coefficients of the desired linear combination with\\ncr{−1, 0, 1}. It is therefore important to know the distribution\\nof βj. As we have previously predicted that the estimator of\\neach parameter βj follows a normal distribution, the linear\\ncombination of several parameters cT βj also follows a normal\\ndistribution:cT bβjN(cT βj, cT σ2\\nj (XT X)\\n−1c\\nNote also that\\n(N−R)σ2\\nj\\nσ2\\nj\\nis thus expressed as the sum of\\nthe squares of N random variables following a normal\\ndistribution, hence :\\nbσ2\\nj\\ncT\\nx2\\nN−R\\n(N −R)\\n(6)\\n(Chi-square law with N - R degrees of freedom). As a\\nresult, we can use the following statistical test which follows\\nStudent’s law with N - R degrees of freedom:\\nTj =\\ncT bβj −cT βj\\nq\\ncT σ2\\nj (XT X)−1c\\n(7)\\nIII. RESULTS\\nPreprocessing is applied on the the ABIDE II database\\non the OHSU site, on 276 functional images, and the pre-\\nprocessing steps will be in the following order: Slice tim-\\ning correction, motion correction (realignment), anatomical-\\nfunctional realignment (Coregistration), segmentation, spatial\\nnormalization (Normalize), and smoothing.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 3}, page_content='FIGURE 1. SPM Preprocessing Pipeline of fMRI data.\\nTable I shows the preprocessing steps with input and output\\ndata for each level, assuming that the name of the fMRI data\\ninput is rest.\\nmethod Name\\nInput Data\\nOutput Data\\nSlice Timing\\nrest.nii\\narest.nii\\nRealignment\\narest.nii\\narest.nii\\nmeanarest.nii\\nrarest.nii\\nrp rest.txt\\nCoregistration\\nReference\\nimage\\n(meanarest.nii)\\nSource\\nimage\\nis\\nanatomical\\nimage\\n(anat.nii)\\nModified\\nanatomical\\nimage (ranat.nii)\\nSegmentation\\nModified anatomical im-\\nage (ranat.nii)\\nc1*.nii:image of the\\ngrey matter\\nc2*.nii:image of the\\nwhite matter\\nc3*.nii\\n:image\\nof\\ncerebrospinal fluid\\nc4*.nii\\n:image\\nof\\nbone tissue\\nc5*.nii :image of soft\\ntissue image\\nranat* seg8.mat\\n:Transformation\\nmatrix for MNI space\\ny ranat*.nii\\n:Field\\ndeformation image:\\nNormalization\\nDeformation\\nimage(y ranat*.nii)\\nimage to write(arrest*.nii)\\nwarest.nii normalized\\nSmoothing\\nwarest.nii normalized\\nswarest.nii smoothed\\nimages\\nTABLE I\\nINPUT AND OUTPUT DATA FOR PROCESSING.\\nTo evaluate the preprocessing a statistical analysis is\\napplied to the basis of the GLM. p the threshold value equal\\n0.01.\\nset-level\\ncluster-level\\nmm mm mm\\np\\nc\\npF W E−cor\\nqF DR−cor\\nkE\\npuncor\\n0.715\\n7\\n0.741\\n0.996\\n0.995\\n0.997\\n0.999\\n0.999\\n1.000\\n0.970\\n0.970\\n0.970\\n0.970\\n0.970\\n0.970\\n0.970\\n457\\n46\\n54\\n35\\n17\\n8\\n1\\n0.164\\n0.676\\n0.647\\n0.722\\n0.818\\n0.885\\n0.970\\n0\\n-2\\n12\\n-2\\n-4\\n-36\\n-44\\n8\\n-2\\n30\\n46\\n52\\n-2\\n46\\n12\\n-2\\n50\\n-18\\n24\\n12\\n16\\n10\\n32\\n14\\n32\\n30\\n14\\nTABLE II\\nSTATISTICAL REPORT 1.\\npeak-level\\nmm mm mm\\npF W E−corr\\nqF DR−corr\\nT\\n(Ze)\\npuncorr\\n0.449\\n0.633\\n0.993\\n0.838\\n0.995\\n0.997\\n0.999\\n0.999\\n1.000\\n0.731\\n0.731\\n0.910\\n0.885\\n0.910\\n0.910\\n0.910\\n0.910\\n0.977\\n3.53\\n3.34\\n2.64\\n3.10\\n2.60\\n2.56\\n2.48\\n2.45\\n2.36\\n3.48\\n3.30\\n2.62\\n3.07\\n2.58\\n2.54\\n2.46\\n2.44\\n2.34\\n0.000\\n0.000\\n0.004\\n0.001\\n0.005\\n0.006\\n0.007\\n0.007\\n0.010\\n0\\n-2\\n12\\n-2\\n-4\\n-36\\n-44\\n8\\n-2\\n30\\n46\\n52\\n-2\\n46\\n12\\n-2\\n50\\n-18\\n24\\n12\\n16\\n10\\n32\\n14\\n32\\n30\\n14\\nTABLE III\\nSTATISTICAL REPORT 2.\\nThe statistical results give the coordinates of the activated\\nareas in the MNI space. The information on each activated\\naggregate is the voxels of the statistical values exceeding the\\nthreshold specified during contrast estimation. Set-level: c is\\nthe number of activated aggregates above the threshold. The\\np-value is the probability of obtaining c or more clusters in\\nthe volume of interest. Cluster-level: this item gives statistical\\ninference results at the level of a cluster. kE gives the number\\nof voxels of each cluster or the selected cluster on the glass\\nbrain. In case the test performed is a T-test. p corrected is\\nthe probability of obtaining kE voxels or more in the volume\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='of interest. p uncorrected is the probability of getting an\\naggregate of kE voxels or more.\\nIV. CONCLUSION\\nThere has been an explosive interest in the use of fMRI\\nin recent years. Working with FMRI data is complex, and\\npre-processing is essential before any analysis. The choice of\\npre-processing requires an application of several pipelines to\\nevaluate each part to arrive at the most efficient processing\\nin the end. In the course of this article, we have worked\\non FMRI data processing in SPM includes two types of\\nprocessing; Temporal Processing, which includes slice timing\\noperation, and also has Spatial Processing, which provides\\nfor Realign, Co-registration, Segmentation, Normalization,\\nand Smoothing. and we evaluate this processing by applying\\nan MLG.\\nREFERENCES\\n[1] Farah M., J. Emerging ethical issues in neuroscience.(2002). Nature\\nNeuroscience, 5, 1123-1129.\\n[2] Murphy, K., Birn, R. M., and Bandettini, P. A. (2013). Resting-\\nstate fMRI confounds and cleanup. Neuroimage 80, 349–359. doi:\\n10.1016/j.neuroimage.2013.04.001.\\n[3] Bright,\\nM.\\nG.,\\nand\\nMurphy,\\nK.\\n(2015).\\nIs\\nfMRI\\n“noise”\\nreally\\nnoise?\\nResting\\nstate\\nnuisance\\nregressors\\nremove\\nvariance\\nwith\\nnetwork\\nstructure.\\nNeuroimage\\n114,\\n158–169.\\ndoi:\\n10.1016/j.neuroimage.2015.03.070.\\n[4] Caballero-Gaudes, C., and Reynolds, R. C. (2017). Methods for\\ncleaning the BOLD fMRI signal. Neuroimage 154, 128–149. doi:\\n10.1016/j.neuroimage.2016.12.018.\\n[5] Esteban, O., Markiewicz, C. J., Blair, R. W., Moodie, C. A., Isik, A. I.,\\nErramuzpe, A., et al. (2019). fMRIPrep: a robust preprocessing pipeline\\nfor functional MRI. Nat. Methods 16, 111–116. doi: 10.1038/s41592-\\n018-0235-4.\\n[6] Marcus, D. S., Harms, M. P., Snyder, A. Z., Jenkinson, M., Wilson, J. A.,\\nGlasser, M. F., et al. (2013). Human Connectome Project informatics:\\nquality control, database services, and data visualization. Neuroimage\\n80, 202–219. doi: 10.1016/j.neuroimage.2013.05.077.\\n[7] Fischl,\\nB.\\n(2012).\\nFreeSurfer.\\nNeuroimage\\n62,\\n774–781.\\ndoi:\\n10.1016/j.neuroimage.2012.01.021.\\n[8] Jenkinson, M., Beckmann, C. F., Behrens, T. E. J., Woolrich, M.\\nW., and Smith, S. M. (2012). Fsl. Neuroimage 62, 782–790. doi:\\n10.1016/j.neuroimage.2011.09.015.\\n[9] Cox, R. W. (1996). AFNI: software for analysis and visualization of\\nfunctional magnetic resonance neuroimages. Comput. Biomed. Res. 29,\\n162–173. doi: 10.1006/cbmr.1996.0014.\\n[10] FZ Benabdallah, A Drissi El Maliani, D Lotfi, R Jennane, M El\\nHassouni.An autism spectrum disorder adaptive identification based on\\nthe Elimination of brain connections: a proof of long-range undercon-\\nnectivity,Soft Computing 26 (10), 4701-4711,26,2022.\\n[11] FZ Benabdallah, E Maliani, A Drissi, M El Hassouni:A CNN Based\\n3D Connectivity Matrices Features for Autism Detection: Application\\non ABIDE I,International Symposium on Ubiquitous Networking, 293-\\n302,2021.\\n[12] FZ Benabdallah, AD El Maliani, D Lotfi, M El Hassouni:Analysis of the\\nover-connectivity in autistic brains using the maximum spanning tree:\\napplication on the multi-site and heterogeneous ABIDE dataset,2020 8th\\nInternational Conference on Wireless Networks and Mobile,2020.\\n[13] K. J. Friston, A. P. Holmes, K. J. Worsley, J.-P. Poline, C. D. Frith, and\\nR. S. Frackowiak. Statistical parametric maps in functional imaging : a\\ngeneral linear approach. Human brain mapping, 2(4) :189–210, 1994a.\\n[14] Penny W. D., Friston K. J., Ashburner J. T., Kiebel S. J., Nichols T.\\nE.(2006). Statistical Mapping: the analysis of functional brain imaging\\n(1st edition). Academic Press.\\n[15] Autism Brain Imaging Data Exchange, [http://fcon 1000.projects.nitrc.\\norg/indi/abide/].\\n[16] V. Calhoun, X. Golay, and G. Pearlson. Improved fmri slice timing\\ncorrection : interpolation errors and wrap around e!ects. In Proceedings,\\nISMRM, 9th annual meeting, Denver, page 810, 2000.\\n[17] R. Sladky, K. J. Friston, J. Tr¨ostl, R. Cunnington, E. Moser, and C.\\nWindischberger. Slice-timing effects and their correction in functional\\nmri. Neuroimage, 58(2) :588–594, 2011.\\n[18] F. Maes, A. Collignon, D. Vandermeulen, G. Marchal, and P. Suetens.\\nMultimodality image registration by maximization of mutual informa-\\ntion. IEEE transactions on medical imaging, 16(2) :187–198, 1997.\\n[19] J. Talairach and P. Tournoux. Co-planar stereotaxic atlas of the human\\nbrain. 3- dimensional proportional system : an approach to cerebral\\nimaging. 1988.\\n[20] A. Evans. 3-d multi-modality human brain mapping : Past, present and\\nfuture. ANNALS OF NUCLEAR MEDICINE, 7 :S13–S13, 1993.\\n[21] N. Tzourio-Mazoyer, B. Landeau, D. Papathanassiou, F. Crivello, O.\\nEtard, N. Delcroix, B. Mazoyer, and M. Joliot. Automated anatomical\\nlabeling of activations in spm using a macroscopic anatomical parcella-\\ntion of the mni mri single-subject brain. Neuroimage, 15(1) :273–289,\\n2002.\\n[22] A. Evans. 3-d multi-modality human brain mapping : Past, present and\\nfuture. ANNALS OF NUCLEAR MEDICINE, 7 :S13–S13, 1993.\\n[23] K. Friston, O. Josephs, E. Zarahn, A. Holmes, S. Rouquette, and J.-\\nB. Poline. To smooth or not to smooth ? : Bias and effciency in fmri\\ntime-series analysis. NeuroImage, 12 (2) :196–208, 2000.\\n[24] G. M. Boynton, S. A. Engel, G. H. Glover, and D. J. Heeger. Linear\\nsystems analysis of functional magnetic resonance imaging in human\\nv1. Journal of Neuroscience, 16(13) : 4207–4221, 1996.\\n[25] K. J. Friston, A. P. Holmes, K. J. Worsley, J.-P. Poline, C. D. Frith, and\\nR. S. Frackowiak. Statistical parametric maps in functional imaging : a\\ngeneral linear approach. Human brain mapping, 2(4) :189–210, 1994a.\\n[26] R. Goebel. fmri : From nuclear spins to brain functions. In Springer US,\\n2015.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='Slice-timing effects and their correction in functional MRI\\nRonald Sladky a,b, Karl J. Friston c, Jasmin Tröstl a,b, Ross Cunnington a,d,\\nEwald Moser a,b, Christian Windischberger a,b,⁎\\na MR Centre of Excellence, Medical University of Vienna, Lazarettgasse 14, 1090 Vienna, Austria\\nb Center for Medical Physics and Biomedical Engineering, Medical University of Vienna, Währinger Gürtel 18-20, 1090 Vienna, Austria\\nc Wellcome Trust Centre for Neuroimaging, Institute of Neurology, University College London, 12 Queen Square, London WC1N 3BG, UK\\nd School of Psychology & Queensland Brain Institute, University of Queensland, Brisbane, Australia\\na b s t r a c t\\na r t i c l e\\ni n f o\\nArticle history:\\nReceived 7 March 2011\\nRevised 23 June 2011\\nAccepted 24 June 2011\\nAvailable online 2 July 2011\\nKeywords:\\nFunctional MRI\\nPre-processing\\nAnalysis\\nSlice-timing correction\\nExact timing is essential for functional MRI data analysis. Datasets are commonly measured using repeated 2D\\nimaging methods, resulting in a temporal offset between slices. To compensate for this timing difference,\\nslice-timing correction (i.e. temporal data interpolation) has been used as an fMRI pre-processing step for\\nmore than ﬁfteen years. However, there has been an ongoing debate about the effectiveness and applicability\\nof this method. This paper presents the ﬁrst elaborated analysis of the impact of the slice-timing effect on\\nsimulated data for different fMRI paradigms and measurement parameters, taking into account data noise and\\nsmoothing effects. Here we show, depending on repetition time and paradigm design, slice-timing effects can\\nsigniﬁcantly impair fMRI results and slice-timing correction methods can successfully compensate for these\\neffects and therefore increase the robustness of the data analysis. In addition, our results from simulated data\\nwere supported by empirical in vivo datasets. Our ﬁndings suggest that slice-timing correction should be\\nincluded in the fMRI pre-processing pipeline.\\n©\\nIntroduction\\nFunctional magnetic resonance imaging (fMRI) data sets are\\ngenerally acquired using sequential 2D imaging techniques like\\nsingle-shot echo planar imaging sequences (Stehling et al., 1991;\\nTurner et al., 1998). Since fMRI data analysis is essentially a time\\ncourse analysis, exact timing with respect to the stimulus presenta-\\ntion paradigm is crucial.\\nSingle-shot EPI sequences allow for single-slice acquisition times\\nin the range of 50–150 ms. Whole-brain coverage is achieved by\\nsequentially repeated image acquisition for a stack of individual slices.\\nDepending on brain coverage and slice thickness, a whole volume can\\nbe acquired within typical repetition times (TRs) ranging from\\nhundreds of milliseconds to several seconds. This causes slice\\nacquisition delays between individual slices which may add up to\\nsigniﬁcant temporal shifts over the full 3D volume between the\\nexpected and actually measured hemodynamic response (Fig. 1). As a\\nconsequence, the reliability and power of time series analysis may be\\ncompromised, resulting in degraded sensitivity to detect activations.\\nTo compensate for slice acquisition delays slice-timing correction\\n(STC) has been proposed as a pre-processing step (Calhoun et al.,\\n2000; Henson et al., 1999) and is currently included in all major fMRI\\nsoftware packages (such as SPM, AFNI, BrainVoyager or FSL). In STC,\\nthe individual slice is temporally realigned to a reference slice based\\non its relative timing using an appropriate resampling method.\\nDifferent data interpolation methods have been proposed for STC\\nincluding linear, sinc and cubic spline interpolation (Goebel, 2010).\\nA more elegant way would be to model known nuisance factors\\nduring analysis rather than changing the data. Accordingly, it has also\\nbeen suggested to account for slice-timing differences via adjust-\\nments in the analysis procedure, in particular the model setup, by\\nusing (a) additional regressors based on the temporal derivatives of\\nthe expected hemodynamic response function in the general linear\\nmodel (GLM), or (b) slice-dependent regressors time-shifted accord-\\ning to their acquisition lag (Henson et al., 1999).\\nAdding temporal derivatives might be advisable to compensate for\\nnon-linear neural and vascular effects resulting in time-shifted and\\ntime-dispersed BOLD responses in different regions, a crucial issue\\nparticularly for event-related fMRI (Calhoun et al., 2004; Friston et al.,\\n1998; Worsley and Taylor, 2006). However, it has not been shown yet,\\nwhether including temporal derivatives into the GLM can fully\\ncompensate for slice acquisition delays.\\nThe second strategy to counter slice-timing effects is based on\\nconstructing slice-speciﬁc time-shifted regressors (Henson et al.,\\n1999). This strategy, however, is problematic with spatially smoothed\\nNeuroImage 58 (2011) 588–594\\n⁎ Corresponding author at: MR Centre of Excellence, Medical University of Vienna,\\nLazarettgasse 14, 1090 Vienna, Austria. Fax: +43 1 40400 7631.\\nE-mail addresses: ronald.sladky@meduniwien.ac.at (R. Sladky),\\nchristian.windischberger@meduniwien.ac.at (C. Windischberger).\\n1053-8119 © 2011 Elsevier Inc.\\ndoi:10.1016/j.neuroimage.2011.06.078\\nContents lists available at ScienceDirect\\nNeuroImage\\njournal homepage: www.elsevier.com/locate/ynimg\\n2011 Elsevier Inc. Open access under CC BY-NC-ND license.\\nOpen access under CC BY-NC-ND license.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='data sets. In many fMRI processing pipelines spatial smoothing is\\nconsidered a mandatory pre-processing step; it not only allows for\\nrandom-ﬁeld approximation based corrections for multiple compar-\\nisons, but also for group analyses in standard space (and also to\\nincrease image signal-to-noise ratios) (Hopﬁnger et al., 2000). Usually\\n6−10 mm3 isotropic Gaussian kernels are used for smoothing,\\nleading to considerable signal dispersion across adjacent slices.\\nWhereas this is not a serious concern for sequentially acquired\\nvolumes, it may cause signiﬁcant estimator bias for interleaved\\nacquisition, because neighboring slices are typically about a half TR\\napart in time.\\nSo far, no extensive study has been published on the effectiveness\\nand robustness of STC and potential unwanted effects on data quality.\\nHenson et al. (1999) showed a better detection rate of neural activity\\nfor slice-timing corrected data for a particular event-related design.\\nSince then, the applicability of STC, and whether to perform it or not,\\nhas been discussed vigorously (if informally) within the neuroimag-\\ning community. The main contention has been whether slice-timing\\ncorrection removes artefactual variance components when there is a\\nlong interval between scans (e.g. TRN2 s). This is the situation in\\nwhich STC would be most effective but it is also the situation, which\\npresents the greatest problems for STC interpolation schemes. This\\npaper is an attempt to treat slice-timing effects formally and assess\\nwhether their correction can improve fMRI data analyses.\\nHere we assess the inﬂuence of slice acquisition delays (slice-\\ntiming effects) using simulated data sets varying acquisition param-\\neters, paradigm designs and noise levels. Herein we compare the\\nresults from GLM analyses of uncorrected and slice-timing corrected\\ndata (as implemented in SPM8) in order to quantify the efﬁcacy of STC\\nmethods. To further assess the interaction between STC and the\\ninclusion of temporal derivatives in the GLM, we compared STC to no\\nSTC, with and without temporal derivatives. This resulted in a\\ncomparison of four schemes or methods corresponding to the cells\\nof a 2×2 factorial design. To validate the conclusion from simulated\\ndata, we also show the results of a study in human subjects using a\\nparadigm, which was speciﬁcally designed to investigate slice-timing\\neffects in real fMRI studies.\\nMethods\\nSlice-timing correction approaches\\nDifferent methods of temporal interpolation can be used for slice-\\ntiming correction. While linear interpolation (Eq. (1)) is easily\\nimplemented and fast, it may introduce undesired temporal smooth-\\ning and more sophisticated interpolation schemes are preferred, such\\nas sinc interpolation as used by default within SPM8. Assuming an\\nacquired time series y of slice number n at time point t(n), linear\\ninterpolation to the slice-time of a reference slice t(r) can formally be\\nexpressed as:\\ny rð Þ\\nn =\\nt rð Þ−t n−1\\nð\\nÞ\\nð\\nÞyn + t n\\nð Þ−t rð Þ\\nð\\nÞyn−1\\nt n\\nð Þ−t n−1\\nð\\nÞ\\nð1Þ\\nAn efﬁcient implementation of the sinc interpolation (Eq. (2))\\nentails applying a phase shift (i.e. adding a constant value) in the\\nfrequency domain of the signal, obtained by fast Fourier transforma-\\ntion. This method is used in SPM, using an implementation to\\ncompensate for interpolation and wrap-around effects (Calhoun et al.,\\n2000).\\ny rð Þ\\nn =\\n∑\\n∞\\ni=−∞\\nxisinc\\nπ\\nTR r−iTR\\nð\\nÞ\\n\\x01\\n\\x03\\nð2Þ\\nCubic spline interpolation is the default interpolation method in\\nBrainVoyager (Goebel, 2010).\\nThe effects of slice-timing correction were studied based on\\nextensive simulations covering a wide range of blocked and event-\\nrelated design paradigms, varying inter-stimulus intervals, contrast-\\nto-noise ratios (CNRs), slice acquisition orders and TRs. In addition,\\nfMRI data from a speciﬁcally designed group study were analyzed in\\norder to validate the ﬁndings in simulated data. Slice-timing\\ncorrections were performed using the spm_slice_timing.m function\\nas implemented in SPM8 (Wellcome Trust Centre for Neuroimaging,\\nUCL, UK) and results were analyzed using MathWorks MATLAB\\nR2009a (Natick, MA, USA).\\nSimulated data\\nA set of 4-dimensional matrices were created to simulate a series\\nof MRI scans with 20 slices, a resolution of 32×32 px2 and 300\\nseconds run duration. Voxel size was set to 1.5×1.5×3 mm3. The time\\ncourses of each set were based on different blocked and event-related\\ndesigns convolved with the canonical hemodynamic response\\nfunction included in SPM8.\\nThe length of on and off blocks in blocked designs was set to 10, 15,\\n20, 25, and 30 s. Event-related designs were single events with an\\naverage stimulus onset asynchrony (SOA, i.e. the time between the\\nonset of one stimulus and the consecutive stimulus) of 4, 6, 8, 10, and\\n12 s; SOAs were jittered according to a uniform distribution within\\n\\x02500=0 SOA. Although very short SOAs are known to furnish low\\ndetection sensitivity, they were included in our test set as an extreme\\nexample.\\nThis raw data with high temporal resolution (f=20Hz) were\\nsampled to temporal resolutions of TR=1, 2, 3, and 4 seconds,\\nincluding slice-dependent acquisition time differences by shifting the\\nresponse function by TR−\\nTR\\nnumber of slices between the ﬁrst and the\\nlast scanned slice. Sequential and interleaved scanning orders were\\nsimulated.\\nDifferent levels of white noise were added to time courses of each\\nvoxel yielding simulated contrast-to-noise ratios (CNR) of 1.7, 2.5, 5,\\nand inﬁnite (i.e. noise-free), respectively.\\n3 \\n4 \\n5 \\n6 \\n7 \\n8 \\n9 \\n10\\n11\\n12\\n1\\n2\\n3\\n4\\n5\\n1\\n2\\n3\\n4\\n5\\nTime [TRs]\\nSlices\\nFig. 1. Illustration of the slice-timing problem. The hemodynamic responses of the\\nindividual slices are acquired at different points in time (top), yielding an aberration in\\nthe scanned data (bottom). The observed time courses of a hemodynamic response\\nreach their maximum amplitude earlier in slices acquired later. Without adequate\\ncompensation this leads to biased estimators in fMRI analysis.\\n589\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 2}, page_content='590\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='After 3-dimensional spatial smoothing of 6 mm FWHM (isotropic\\nGaussian kernel), GLM analysis was performed with stimulus onsets\\naligned to the acquisition time of the middle slice. This caused a TR-\\ndependent slice-timing difference of approximately up to ±0.5,1,1.5,\\nand 2s, respectively, between slices. In separate analyses, the\\ntemporal derivative of the response function was also modeled in\\nthe GLM to allow some ﬂexibility in the signal onsets and to evaluate\\nits capabilities in compensating slice acquisition delays (Friston et al.,\\n1998).\\nFinally, four datasets were evaluated: uncorrected (original) and\\nSTC (STC) data, each analyzed with and without temporal derivatives\\n(original+TD, STC+TD) in the GLM.\\nHuman fMRI data\\nSubjects\\n13 healthy subjects (8 male, 5 female; age: 25.4±3.3 years)\\nparticipated in this study, which was approved by the ethics committee\\nof the Medical University of Vienna, Austria, and gave written informed\\nconsent prior to the experiment. Subjects were clinically assessed and\\nfree of current or past psychiatric or neurological conditions. Negative\\ndrug screening was mandatory. Nicotine or caffeine intake was not\\nallowed on the day of the measurement.\\nData acquisition and fMRI paradigm\\nThe participants were instructed to do bimanual ﬁnger tapping\\nduring the display of an animated checkerboard, changing black and\\nwhite tiles with a frequency of 8Hz, in a blocked paradigm, consisting\\nof six blocks of 10s resting and 10s ﬁnger-tapping periods (130s total).\\nThis design is known to robustly trigger neural activation in visual\\nareas (occipital lobe) and motor areas (precentral gyrus, supplemen-\\ntary motor area, basal ganglia). Finger movements were self-paced\\nand none of the subjects reported difﬁculties with this task.\\n75 volumes were acquired on a TIM TRIO 3T full-body MR scanner\\n(SIEMENS Medical, Germany) using single-shot 2D EPI with the\\nfollowing parameters: 20 axial slices of 3mm thickness (2.1mm slice\\ngap) aligned to the connecting line between anterior and posterior\\ncommissure with a matrix size of 128 × 128px2, TE = 40 ms,\\nTR=1800ms and a ﬁeld of view of 192×192mm2 (1.5×1.5×3mm3\\nvoxel size). Slices were acquired interleaved along the z-axis in feet-\\nto-head direction starting with even-numbered slices (i.e. 2, 4,…, 20,\\n1, 3,…, 19).\\nPre-processing and analysis\\nSlice-timing correction as implemented in SPM8 was applied to all\\n13 empirical datasets. All other standard pre-processing steps were\\nsimilarly performed on both the uncorrected original data and the STC\\ndata, including realignment to compensate for bulk head movements,\\nnormalization to MNI space and spatial smoothing with a Gaussian\\nkernel of 8×8×8mm3 FWHM. Each data set was masked with the\\nSPM8 brainmask template.\\nGLM analysis was performed using regressors generated by\\nconvolving the time course of the visual stimulation onsets and\\nduration with the SPM8 canonical hemodynamic response function.\\nMotion parameters obtained from realignment were included in the\\nGLM. All other analysis steps were identical to those described for\\nsimulated data, yielding four datasets of uncorrected and slice-timing\\ncorrected data, each with and without temporal derivatives. The\\nsingle-subject parameter estimate maps from the canonical regressors\\nwere used for second level group analyses.\\nWe assessed the effect of slice-timing correction and temporal\\nderivative modeling in two ways. First, we quantiﬁed the bias in\\nparameter estimation, associated with temporal correction and\\nmodeling, by examining the parameter estimates relative to the\\nestimates from noiseless data. To assess sensitivity, we analyzed all\\ncombinations of treated data to test for activations at the group level\\nin the empirical study, using statistical parametric maps based upon\\nT-statistics.\\nResults\\nSimulated data\\nOur results clearly showed a reduction of parameter estimates (β-\\nvalues) with increased estimator bias, depending on the measure-\\nment and paradigm parameters. In general, this bias was reduced by\\nthe STC procedure.\\nAs expected, the reduction of parameter estimates due to slice-\\ndelay effects were more pronounced for long TRs, event-related\\ndesigns and designs with shorter SOA. We found a maximum decrease\\nof parameter estimates of 63% (event-related design, average\\nSOA=4s, TR=4s). In blocked designs we observed a reduction of\\n12% (block length 10s, TR=4 s). Note that in our simulations the\\nparadigm regressors and slice-timing correction are temporally\\naligned to the middle slice of the 3D volume. Therefore, a TR of e.g.\\n4 s in this simulation is comparable to a dataset with a TR of 2s when\\nthe ﬁrst (or last) slice acquired are used as a reference slice. This can\\nbe a reasonable practice, if the region of interest and putative\\nactivations are located near the ﬁrst (or last) slice because it\\nsuppresses temporal interpolation effects in these areas.\\nCompared to uncorrected data an increase of parameter estimates\\nwas found for all tested paradigms, TRs and CNRs in slice-timing\\ncorrected data. Importantly, STC did not cause a decrease in average or\\nminimum parameter estimates in any of our datasets (Fig. 2B). One\\nsimulation set (event-related design, average SOA=4s, TR=4s;\\nFig. 2B, lower panel, 1st box, 4th column) showed a decrease by 4%\\nin maximum parameter estimates. This, however, is attributable to\\naliasing artifacts caused by noncritical sampling, which can occur\\nwhen the average SOA is shorter or equal to the TR.\\nOmitting STC and compensating for slice-timing effects by\\nincluding the temporal derivatives into the model yielded no increase\\nin mean parameter estimates (Fig. 2A).\\nWhile higher levels of Gaussian noise caused higher residual\\nvalues in the result of the GLM analysis, it had little impact on the\\nparameter estimates. We found no noise-related changes in the\\nperformance efﬁciency of slice-timing correction.\\nTo better assess the consequences of slice-timing correction we\\nfurther analyzed its effect on the parameter estimates in individual\\nslices. Fig. 3 shows a comparison of typical event-related designs. It\\ncan be seen that in all presented cases slice-timing correction led to an\\nincrease of parameter estimates and a reduction of residuals. We\\nfound this for all tested designs and image parameters, with the\\nexception of designs with subcritical sampling. It should be noted,\\nhowever, that this decrease in parameter estimates was not due to a\\nfailure of the slice-timing correction algorithm but due to aliasing\\nartifacts because of noncritical sampling of the data.\\nFig. 2. Effects of including temporal derivatives (TD) [A] or performing slice-timing correction (STC) [B] on simulated data. The ﬁgure shows a comparison of parameter estimates for\\nuncorrected (red) and corrected (green) data for blocked and event-related designs (upper and lower boxes respectively), varying TRs (columns) and CNRs (sub-columns within TR\\ncolumns). Values are shown relative to the maximum parameter estimate of the uncorrected, noise-free data. Bold lines indicate the mean across all slices. Shaded areas show the\\nrange from highest to lowest estimate. It can be seen that parameter estimates of the slice-timing corrected datasets [B] are, in general (higher mean and smaller range), higher than\\nthe uncorrected datasets and therefore closer to the unbiased estimates with optimal temporal alignment (i.e. 100%). Also for lower contrast-to-noise ratios (CNRs) higher parameter\\nestimates were consistently gained for slice-timing corrected data sets.\\n591\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 4}, page_content='Three-dimensional smoothing may conceal the effects of STC on\\nthe individual slices. Therefore we also analyzed simulated images\\nthat were only smoothed in-plane. For our tested datasets slice-timing\\ncorrection yielded robust and profound improvements of parameter\\nestimates for all individual slices. There was a signiﬁcant decrease of\\nparameter estimates and increase of residuals in the uncorrected\\ndatasets in slices with larger temporal offset (Fig. 3).\\nHuman fMRI data\\nAs expected, the paradigm stimulated brain activation changes\\nin the motor areas (M1, SMA, z=56mm [MNI]), the visual cortex (V1,\\nz=−4mm [MNI]) and the cerebellum (culmen, V1, z=−24mm\\n[MNI]) (Fig. 4, left table, original).\\nComparing slice-timing corrected and uncorrected data using\\npaired T-tests (Fig. 4, right table, orig. vs STC) revealed signiﬁcant\\n(pb0.001) activation differences. This particularly affected dorsal\\nregions, i.e. regions with higher temporal offset due to the acquisition\\ndirection, where the slice-timing effect was expected to be most\\ndominant. Importantly, robust SMA activation was detected in the STC\\ndata sets only. Differences were less prominent in ventral brain\\nregions, where slice-timing effects were expected to be lower.\\nFor uncorrected data, adding temporal derivative regressors in the\\ndesign matrix caused a reduction in sensitivity, particularly in the\\nmotor network (M1 and SMA) (Fig. 4, left table, TD, and right table,\\nxtitoriginal vs TD).\\nIncluding the temporal derivatives in the GLM analysis of the slice-\\ntiming corrected data yielded higher parameter estimates in the focal\\nFig. 3. Slice-dependent changes in a selection of typical event-related designs. The top part of each graph shows a comparison of parameter estimates from uncorrected (red) and\\nslice-timing corrected (green) data sets. Dashed lines represent the parameter estimates of models where temporal derivatives were included. Residuals of the models are plotted at\\nthe bottom part of the graphs. To visualize the impact of spatial smoothing on the temporal alignment the same data is presented 3D (panel A) and 2D (panel B) smoothed.\\n592\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content=\"point of M1 activation (Fig. 4, left table, STC+TD, and right table,\\noriginal vs STC+TD). However, we found no activation in the SMA.\\nDiscussion\\nOur results clearly show a beneﬁt of slice-timing correction for\\nparameter estimation on single-subject level, which, in turn, could\\nsigniﬁcantly improve sensitivity in group statistical analysis. This is\\nparticularly true for event-related designs and blocked designs with\\nshort block length. Analysis of in vivo fMRI images supported our\\nﬁndings from simulated data.\\nNo adverse effects of STC with respect to parameter estimate bias\\nwere found for realistic paradigm designs. Higher noise levels had no\\nimpact on estimates or the efﬁciency of the applied STC algorithm.\\nFurther, no adverse effects were found in in vivo datasets with non-\\nGaussian noise due to physiological and motion artifacts.\\nWhile beneﬁcial effects were found for STC, adding temporal\\nderivatives to the model did not suppress bias (shrinkage) in\\nparameter estimates to the same extent. Clearly, the shrinkage of\\nparameter estimates does not necessarily decrease sensitivity,\\nprovided the between subject variability in parameter estimates\\ndecreases in proportion. However, our empirical example suggests\\nthat slice-timing correction increased sensitivity to group effects in a\\nway that was subverted by including temporal derivatives. It has been\\nshown previously that using TD in whole-brain analyses can diminish\\npower depending on the response latency (Della-Maggiore et al.,\\n2002). This ﬁnding is consistent with our ﬁndings.\\nFurthermore, we know that temporal derivatives can explain\\nvariance due to temporal delays of about ±1s (Henson et al., 2002).\\nThis upper bound on the delays that temporal derivatives can handle\\nreﬂects the failure of the ﬁrst-order Taylor expansion implicit in their\\nuse, which is determined quantitatively by the characteristic time\\nconstants of the HRF. This bound means that, even under the\\nassumption of an optimal model (i.e. regressors time-locked to the\\nmiddle of one TR), derivatives will not model adequately slice-timing\\neffects caused by TRs greater than 2s. Our simulations corroborate this\\nfact and additionally showed no consistent improvements even for\\nshorter TRs. This is expected as the canonical HRF and its derivative\\ncan lose their orthogonality when convolved with a stimulus function.\\nTherefore, a part of the variance explained by the canonical HRF\\nregressor may be modeled by its derivative. In general, including the\\ntemporal derivatives will thus lead to a decrease in residuals, possibly\\nincreasing statistical signiﬁcance on the single-subject level, but it will\\nalso reduce the parameter estimates of the canonical regressor. This\\nmay reduce sensitivity, as only the latter usually enter ordinary least\\nsquares random-effects group analysis. In principle, the problem of\\nnon-orthogonality can be addressed by specifying (F) contrasts at the\\nsecond level over both the canonical and temporal derivative\\nparameter estimates. However, this depends upon their being a\\nconsistent group effect on both, which may not be guaranteed if slice-\\ntiming effects are artefactual (Calhoun et al., 2004; Henson et al.,\\n1999; Steffener et al., 2010).\\nThe question of whether to include temporal derivatives or not is\\nreally an issue of model selection. Generally speaking, their inclusion\\nwill explain away early or late hemodynamic responses, which could\\nreduce sensitivity. On the other hand, it is at least theoretically\\npossible that modeling real or artifactual latency differences will\\nincrease the precision of parameter estimates. The overall impact on\\nsensitivity to canonical hemodynamic responses is clearly difﬁcult to\\ngeneralize about. However, the procedure described in this paper\\nprovides a simple protocol for assessing the impact of temporal\\nderivatives, with or without slice-timing correction (see Fig. 4).\\nThe results in this paper are based on SPM8's implementation of\\nthe STC algorithm. Nevertheless, our ﬁndings should be transferable\\nto the implementations in other fMRI applications, as cubic spline\\ninterpolation and sinc interpolation have been shown to provide\\nalmost identical results (Goebel, 2010).\\nA sometimes-neglected issue is the interaction of the slice-timing\\neffect and motion. Subject movement along the z-axis or head tilts\\naround the x-axis will introduce not only spatial but also temporal\\nartifacts. This is a concern particularly for clinical applications where\\npatients might not be able to fully suppress their movements while in\\nFig. 4. Comparison of in vivo fMRI data. Group SPMs of a visual-motor task versus rest condition (left table) and paired t-test comparisons of compensation strategies for slice-timing\\neffects. Threshold was set to pb0.001. The positive effects of slice-timing correction on the parameter estimates (STC, left table, third column) are present particularly in the motor\\nnetwork (ﬁrst row). Due to our measurement parameters (feet-to-head acquisition order) slice-timing effects were expected to be most prominent in this dorsal section.\\nComparison of the uncorrected and STC data revealed signiﬁcant increase of t-values in group statistics.\\n593\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594\"),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 6}, page_content='the scanner. Given the nature of typical serial fMRI pre-processing\\npipelines one could consider performing slice-timing correction\\nbefore realignment if subject movement is moderate. In the presence\\nof pronounced inter-slice movements it might be preferable to realign\\nbefore STC. Clearly, serial correction approaches like these are\\nsuboptimal and cannot compensate for the interactions between\\ntime shifts and movement (Bannister et al., 2007), and a combined\\nmethod of motion and timing correction might thus be indicated\\n(Bannister et al., 2004). However, this is not yet implemented in the\\nmajor fMRI analysis packages.\\nA non-beneﬁcial side effect of all resampling methods is the\\npossible introduction of aliasing effects for signals at frequencies\\nabove the Nyquist sampling limit of f =\\n1\\n2 × TR. Given a typical TR of\\n2 s (f=0.25Hz) a minimal inter-stimulus interval (ISI) of more than\\n4s is recommended. When using the slice acquired in the middle of\\nthe acquisition period (TR) as a reference slice, the signal needs to be\\nshifted by TR / 2, therefore reducing the suggested minimal ISI to 2s.\\nWhen choosing the mid-slice as the target less temporal interpolation\\nis needed. This might decrease the risk of interpolation artifacts. Note\\nthat all regressors in the GLM also need to be adjusted for this shift in\\ntime by TR / 2.\\nFinally, Holmes et al. (1997) had shown that no physiologically\\ninduced signal change was present at frequencies over 0.25Hz. As a\\nconsequence fMRI data can be assumed band-limited, which means\\nthat potential aliasing effects by STC only affect signal components\\noriginating from noise.\\nIn conclusion, slice-timing correction has been shown to suppress\\nbias in quantitative parameter estimates at the single-subject level\\nand, empirically, to increase sensitivity in group-level statistical\\ninferences.\\nAcknowledgments\\nThe authors would like to thank Guillaume Flandin and Elmar\\nLaistler for helpful comments and the OeNB (Austrian National Bank,\\nP12982) and the FWF (Austrian Science Fund, P23021) for ﬁnancial\\nsupport.\\nReferences\\nBannister, P.R., Brady, J.M., Jenkinson, M., 2004. TIGER - A new model for spatio-\\ntemporal realignment of fMRI data. Lecture Notes in Computer Science 3117,\\n292–303.\\nBannister, P.R., Brady, J.M., Jenkinson, M., 2007. Integrating temporal information with a\\nnon-rigid method of motion correction for functional magnetic resonance images.\\nImage and Vision Computing 25, 311–320.\\nCalhoun, V., Golay, X., Pearlson, G., 2000. Improved fMRI slice timing correction:\\ninterpolation errors and wrap around effects. Proceedings, ISMRM, 9th Annual\\nMeeting, Denver, p. 810.\\nCalhoun, V., Stevens, M., Pearlson, G., Kiehl, K., 2004. fMRI analysis with the general\\nlinear model: removal of latency-induced amplitude bias by incorporation of\\nhemodynamic derivative terms. Neuroimage 22, 252–257.\\nDella-Maggiore, V., Wilkin, C., Peres-Neto, P.R., McIntosh, A.R., 2002. An empirical\\ncomparison of SPM preprocessing parameters to the analysis of fMRI data.\\nNeuroImage 17, 19–28.\\nFriston, K., Fletcher, P., Josephs, O., Holmes, A., Rugg, M., 1998. Event-related fMRI:\\ncharacterizing differential responses. Neuroimage 7, 77–83.\\nGoebel, R., 2010. Brain Voyager—Users Guide.\\nHenson, R., Buechel, C., Josephs, O., Friston, K., 1999. The slice-timing problem in event-\\nrelated fMRI. NeuroImage 9, 125-125.\\nHenson, R., Price, C., Rugg, M., Turner, R., Friston, K., 2002. Detecting latency differences\\nin event-related BOLD responses: application to words versus nonwords and initial\\nversus repeated face presentations. NeuroImage 15, 83–97.\\nHolmes, A.P., Josephs, O., Büchel, C., Friston, K.J., 1997. Statistical modelling of low-\\nfrequency confounds in fMRI. Proceedings, HBM, 3rd Annual Meeting, Copenhagen,\\np. 480.\\nHopﬁnger, J.B., Büchel, C., Holmes, A.P., Friston, K.J., 2000. A study of analysis\\nparameters that inﬂuence the sensitivity of event-related fMRI analyses. Neuro-\\nImage 11, 326–333.\\nSteffener, J., Tabert, M., Reuben, A., Stern, Y., 2010. Investigating hemodynamic\\nresponse variability at the group level using basis functions. NeuroImage 49,\\n2113–2122.\\nStehling, M., Turner, R., Mansﬁeld, P., 1991. Echo-planar imaging: magnetic resonance\\nimaging in a fraction of a second. Science 254, 43.\\nTurner, R., Howseman, A., Rees, G., Josephs, O., Friston, K., 1998. Functional magnetic\\nresonance imaging of the human brain: data acquisition and analysis. Experimental\\nBrain Research 123, 5–12.\\nWorsley, K., Taylor, J., 2006. Detecting fMRI activation allowing for unknown latency of\\nthe hemodynamic response. NeuroImage 29, 649–654.\\n594\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594')]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from langchain_community.document_loaders import PyPDFLoader, PyMuPDFLoader\n",
    "\n",
    "## load all the text files from the directory\n",
    "dir_loader=DirectoryLoader(\n",
    "    \"../data/literature\",\n",
    "    glob=\"**/*.pdf\", ## Pattern to match files  \n",
    "    loader_cls= PyMuPDFLoader, ##loader class to use\n",
    "    show_progress=False\n",
    "\n",
    ")\n",
    "\n",
    "pdf_documents=dir_loader.load()\n",
    "pdf_documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "9f63c57b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "langchain_core.documents.base.Document"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(pdf_documents[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "b5ebb397",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import numpy as np\n",
    "from sentence_transformers import SentenceTransformer\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "import uuid\n",
    "from typing import List, Dict, Any, Tuple\n",
    "from sklearn.metrics.pairwise import cosine_similarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "57b7b762",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading embedding model: all-MiniLM-L6-v2\n",
      "Model loaded successfully. Embedding dimension: 384\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<__main__.EmbeddingManager at 0x1df981bf230>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "#### Embedding and vectorStoreDB\n",
    "\n",
    "class EmbeddingManager:\n",
    "    \"\"\" Handles document embedding generation using SentenceTransformer\"\"\"\n",
    "\n",
    "    def __init__(self, model_name: str = \"all-MiniLM-L6-v2\"):\n",
    "        \"\"\"\n",
    "        Initialize the embedding manager\n",
    "        Args:\n",
    "            model_name: HuggingFace model name for sentence embeddings\n",
    "        \"\"\"\n",
    "\n",
    "        self.model_name = model_name\n",
    "        self.model = None\n",
    "        self._load_model()\n",
    "\n",
    "    def _load_model(self):\n",
    "        \"\"\"Load the SentenceTransformer model\"\"\"\n",
    "        try:\n",
    "            print(f\"Loading embedding model: {self.model_name}\")\n",
    "            self.model = SentenceTransformer(self.model_name)\n",
    "            print(f\"Model loaded successfully. Embedding dimension: {self.model.get_sentence_embedding_dimension()}\")\n",
    "        except Exception as e:\n",
    "            print(f\"Error loading model {self.model_name}: {e} \")\n",
    "            raise\n",
    "\n",
    "    def generate_embeddings(self, texts: List[str]) -> np.ndarray:\n",
    "        \"\"\"\n",
    "        Generate embeddings for a list of texts\n",
    "\n",
    "        Args:\n",
    "            texts: List of text strings to embed\n",
    "\n",
    "        Returns:\n",
    "            numpy array of embeddings with shape (len(texts), embedding_dim)\n",
    "          \n",
    "        \"\"\"\n",
    "        if not self.model:\n",
    "            raise ValueError(\"Model not loaded\")\n",
    "        \n",
    "        print(f\"Generate embeddings for {len(texts)} texts...\")\n",
    "        embeddings = self.model.encode(texts, show_progress_bar=True)\n",
    "        print(f\"Generate embeddings with shape: {embeddings.shape}\")\n",
    "        return embeddings\n",
    "    \n",
    "\n",
    "## initialize the embedding manager \n",
    "embedding_manager = EmbeddingManager()\n",
    "embedding_manager"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7aa5fc74",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "7e37d6c3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Vector store initialized. Collection: pdf_documents\n",
      "📄 Existing documents in collection: 146\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<__main__.VectorStore at 0x1df9a593620>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import uuid\n",
    "import chromadb\n",
    "import numpy as np\n",
    "from typing import List, Any\n",
    "\n",
    "\n",
    "class VectorStore:\n",
    "    \"\"\"Manages document embeddings in a ChromaDB vector store\"\"\"\n",
    "\n",
    "    def __init__(\n",
    "        self,\n",
    "        collection_name: str = \"pdf_documents\",\n",
    "        persist_directory: str = \"../data/vector_store\"\n",
    "    ):\n",
    "        \"\"\"\n",
    "        Initialize the vector store\n",
    "\n",
    "        Args:\n",
    "            collection_name: Name of the ChromaDB collection\n",
    "            persist_directory: Directory to persist the vector store\n",
    "        \"\"\"\n",
    "        self.collection_name = collection_name\n",
    "        self.persist_directory = persist_directory\n",
    "        self.client = None\n",
    "        self.collection = None\n",
    "        self._initialize_store()\n",
    "\n",
    "    def _initialize_store(self):\n",
    "        \"\"\"Initialize ChromaDB client and collection\"\"\"\n",
    "        try:\n",
    "            # Create persistent directory\n",
    "            os.makedirs(self.persist_directory, exist_ok=True)\n",
    "\n",
    "            # Create persistent ChromaDB client\n",
    "            self.client = chromadb.PersistentClient(path=self.persist_directory)\n",
    "\n",
    "            # Get or create collection\n",
    "            self.collection = self.client.get_or_create_collection(\n",
    "                name=self.collection_name,\n",
    "                metadata={\"description\": \"PDF document embeddings for RAG\"}\n",
    "            )\n",
    "\n",
    "            print(f\"✅ Vector store initialized. Collection: {self.collection_name}\")\n",
    "            print(f\"📄 Existing documents in collection: {self.collection.count()}\")\n",
    "\n",
    "        except Exception as e:\n",
    "            print(f\"❌ Error initializing vector store: {e}\")\n",
    "            raise\n",
    "\n",
    "    def add_documents(self, documents: List[Any], embeddings: np.ndarray):\n",
    "        \"\"\"\n",
    "        Add documents and embeddings to the vector store\n",
    "\n",
    "        Args:\n",
    "            documents: List of LangChain Document objects\n",
    "            embeddings: NumPy array of corresponding embeddings\n",
    "        \"\"\"\n",
    "\n",
    "        if len(documents) != len(embeddings):\n",
    "            raise ValueError(\"Number of documents must match number of embeddings\")\n",
    "\n",
    "        print(f\"Adding {len(documents)} documents to vector store...\")\n",
    "\n",
    "        ids = []\n",
    "        metadatas = []\n",
    "        documents_text = []\n",
    "        embeddings_list = []\n",
    "\n",
    "        for i, (doc, embedding) in enumerate(zip(documents, embeddings)):\n",
    "\n",
    "            # Unique document ID\n",
    "            doc_id = f\"doc_{uuid.uuid4().hex[:8]}_{i}\"\n",
    "            ids.append(doc_id)\n",
    "\n",
    "            # Metadata\n",
    "            metadata = dict(doc.metadata) if doc.metadata else {}\n",
    "            metadata[\"doc_index\"] = i\n",
    "            metadata[\"content_length\"] = len(doc.page_content)\n",
    "            metadatas.append(metadata)\n",
    "\n",
    "            # Document text\n",
    "            documents_text.append(doc.page_content)\n",
    "\n",
    "            # Convert embedding to list\n",
    "            embeddings_list.append(embedding.tolist())\n",
    "\n",
    "        try:\n",
    "            self.collection.add(\n",
    "                ids=ids,\n",
    "                embeddings=embeddings_list,\n",
    "                metadatas=metadatas,\n",
    "                documents=documents_text\n",
    "            )\n",
    "\n",
    "            print(f\"✅ Successfully added {len(documents)} documents\")\n",
    "            print(f\"📊 Total documents in collection: {self.collection.count()}\")\n",
    "\n",
    "        except Exception as e:\n",
    "            print(f\"❌ Error adding documents to vector store: {e}\")\n",
    "            raise\n",
    "\n",
    "vectorstore = VectorStore()\n",
    "vectorstore      \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "a69693ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Import working ✅\n"
     ]
    }
   ],
   "source": [
    "from langchain_text_splitters import RecursiveCharacterTextSplitter\n",
    "print(\"Import working ✅\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "63768cf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain_text_splitters import RecursiveCharacterTextSplitter\n",
    "\n",
    "def split_documents(documents, chunk_size=1000, chunk_overlap=200):\n",
    "    \"\"\"\n",
    "    Split documents into smaller chunks for better RAG performance\n",
    "    \"\"\"\n",
    "\n",
    "    text_splitter = RecursiveCharacterTextSplitter(\n",
    "        chunk_size=chunk_size,\n",
    "        chunk_overlap=chunk_overlap,\n",
    "        length_function=len,\n",
    "        separators=[\"\\n\\n\", \"\\n\", \" \", \"\"]\n",
    "    )\n",
    "\n",
    "    split_docs = text_splitter.split_documents(documents)\n",
    "\n",
    "    print(f\"✅ Split {len(documents)} documents into {len(split_docs)} chunks\")\n",
    "\n",
    "    # Show example of a chunk\n",
    "    if split_docs:\n",
    "        print(\"\\n📌 Example chunk:\")\n",
    "        print(f\"Metadata: {split_docs[0].metadata}\")\n",
    "        print(f\"Content: {split_docs[0].page_content[:200]}...\")\n",
    "\n",
    "    return split_docs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ba429395",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "✅ Split 23 documents into 146 chunks\n",
      "\n",
      "📌 Example chunk:\n",
      "Metadata: {'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}\n",
      "Content: 2007 IEEE/ICME International Conference on Complex Medical En2ineerin2\n",
      "A Study of Combination fMRI and DTI in A\n",
      "Cognitive Task\n",
      "Tingting Wang', Zifan Cai1, Ni Tang1, Jie Lu3, Li Yao1 2 Xiaojie Zhaol2*\n",
      "...\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content=\"2007 IEEE/ICME International Conference on Complex Medical En2ineerin2\\nA Study of Combination fMRI and DTI in A\\nCognitive Task\\nTingting Wang', Zifan Cai1, Ni Tang1, Jie Lu3, Li Yao1 2 Xiaojie Zhaol2*\\n1 School of Information Science and Technology, Beijing Normal University\\n2 State Key Laboratory of Cognitive Neuroscience and Learning, Beijing Normal University\\n3 Department of Radiology, Xuanwu Hospital of Capital University of Medical Sciences\\n*zhaoxj 86ghotmail.com\\nAbstract-Magnetic\\nresonance\\ndiffusion\\ntensor\\nimaging\\n(MR-DTI)\\nas\\na noninvasive technique\\ncan\\ninvestigate\\nthe\\ncerebral white matter architecture based on the directional\\ninformation of diffusion that is related to fiber tract. With the\\ndevelopment of DTI technique in recent years, more and more\\nstudies of combining fMRI with DTI are used to explore the\\nrelationships between functional connectivity and its structural\\nproperties in vivo, especially in pathological researches. In our\\nwork, we\\nfocus\\non whether\\nfunctional\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content='relationships between functional connectivity and its structural\\nproperties in vivo, especially in pathological researches. In our\\nwork, we\\nfocus\\non whether\\nfunctional\\nconnectivity\\ntakes\\nstructural connectivity as its basis in a specific cognitive task in\\nnormal\\nbrains.\\nTemporal\\ncorrelation method was used\\nto\\nacquire the functional connectivity network based on fMRI in a\\npicture-naming task. Fractional anisotropy (FA) that denoted\\nthe degree of the diffusion anisotropy was calculated using\\ndiffusion tensor matrix based on DTI. Then the principle\\nvector of tensor matrix was mapped into X, Y, and Z axes with\\nRGB color and weighting the color brightness by FA. We\\nnormalized the two data results before combining them so as to\\nmatch them better. Then functional connectivity maps were\\noverlaid upon DTI directional color maps that indicated the\\nwhite matter tract orientation based on the principle vector of\\ntensor\\nmatrix.\\nThe\\nresults\\nshowed\\nthat\\nthe\\nfunctional'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content='overlaid upon DTI directional color maps that indicated the\\nwhite matter tract orientation based on the principle vector of\\ntensor\\nmatrix.\\nThe\\nresults\\nshowed\\nthat\\nthe\\nfunctional\\nconnected regions of brain in this cognitive task had white\\nmatter\\nconnectivity.\\nIt\\nmay\\nbe\\nsuggested\\nfunctional\\nconnectivity\\ndid\\ntake\\nstructure\\nas\\nits\\nbasis,\\nand\\nthe\\ncombination of fMRI and DTI could explore effectively the\\ncorrelation of structure and function\\nin both healthy and\\ndisease cerebra.\\nI. INTRODUCTION\\nInvestigation of the relationships between the cerebra\\nfunctional\\nconnectivity\\nof the\\nbrain\\nand\\nits\\nstructure\\nproperties has become a popular approach for neuroscience\\nresearch. Function-structure correlations indicate how white\\nmatter structural connections relate to regional gray matter\\nactivity connections [1] [2]. Functional connected network\\nare often obtained by using functional MRI (fMRI) with\\ntemporal correlation or independent component analysis\\n(ICA),\\netc.[3][4].'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content='activity connections [1] [2]. Functional connected network\\nare often obtained by using functional MRI (fMRI) with\\ntemporal correlation or independent component analysis\\n(ICA),\\netc.[3][4].\\nIn vivo Magnetic resonance diffusion\\ntensor imaging (MR-DTI) as a noninvasive technique has\\nbecome a popular tool for investigation of white matter\\narchitecture [5][6].\\nDTI detect fiber pathways by measuring the degree of\\nwater\\ndiffusion\\nanisotropic\\n(directionally\\ndependent)\\nin\\nwhite matter. Axonal membranes and myelin sheaths present\\nbarriers to the motion of water molecules in directions not\\nparallel to their own orientation. The direction of maximum\\ndiffusivity has been shown to coincide with white matter\\nfiber tract orientation [7]. This information is contained in\\nthe diffusion tensor which is a 3*3 diagonally symmetric\\n(Dij= Dji) matrix with six degrees of freedom. Based on the\\ntensor\\nmatrix,\\none\\ncould\\nobtain\\ndiffusion\\nanisotropy\\ninformation\\n(e.g.\\nFractional\\nanisotropy,\\nFA)\\nor\\nfiber'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content='(Dij= Dji) matrix with six degrees of freedom. Based on the\\ntensor\\nmatrix,\\none\\ncould\\nobtain\\ndiffusion\\nanisotropy\\ninformation\\n(e.g.\\nFractional\\nanisotropy,\\nFA)\\nor\\nfiber\\ntractography etc. These results derived from DTI have been\\nwidely\\nused\\nin\\nclinic\\nand\\nvisualization\\nof anatomic\\nconnections between different parts of brain.\\nWith the quickly development of DTI, more and more\\nstudies have combined fMRI\\nand DTI techniques\\nfor\\nresearch on correlation of structure and function. During the\\ncombination, fiber tractography, fast-marching tractography\\n(FMT),\\nProbabilistic\\nIndex\\nof\\nConnectivity\\n(PICo)\\ntractography and DTI directional eigenvector color maps are\\noften used\\nto denote white matter connectivity.\\nThese\\nstudies show that this combination can give additional\\ninformation about brain disease and organization of brain\\nfunctions for doctors and researchers [8][9]. However, few\\nstudies\\nfocus on\\na given cognitive task\\nto\\ninvestigate\\nwhether functional connectivity in this task takes structural'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content='functions for doctors and researchers [8][9]. However, few\\nstudies\\nfocus on\\na given cognitive task\\nto\\ninvestigate\\nwhether functional connectivity in this task takes structural\\nconnectivity as its basis in normal brain.\\nIn this work, a picture-naming cognitive task was chosen\\nto\\ninvestigate\\nfunction-structure\\nrelationships.\\nThis task\\nwould activate the cerebral visual\\ncortex and semantic\\nprocessing regions. Brain activity regions are processed by\\ntemporal\\ncorrelation method that\\nis\\na commonly used\\napproach\\nfor\\nidentifying\\nfunctional\\nnetworks\\nof brain\\nregions. Then we obtained directional eigenvector color\\nmaps which\\nis\\nrelated\\nto\\nthe\\nprinciple\\nvector of the\\ndiagonally tensor matrix and weighted the color brightness\\nby FA. These directional color maps depict the main white\\nmatter\\nfiber\\npathways.\\nIn\\nfurther\\nresearches,\\nfiber\\ntractography could be used to find more specific fiber\\nconnectivity information between activated brain regions.'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 0}, page_content='matter\\nfiber\\npathways.\\nIn\\nfurther\\nresearches,\\nfiber\\ntractography could be used to find more specific fiber\\nconnectivity information between activated brain regions.\\nAfter that, combination of the results from fMRI and DTI\\ndata\\nwere\\nimplemented\\nby\\noverlaying\\nfunctional\\nconnectivity\\nmaps\\non\\ndirectional\\ncolor\\nmaps\\nwith\\nnormalization [1]. It has shown that functional connectivity\\ntook structural connectivity as its basis in this cognitive task.\\n1-4244-1078-9/07/$25.00 c2007 IEEE.\\n896\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content='2007 IEEE/ICME International Conference on Complex Medical Engineering\\nII. METHODS\\nA. Data acquisition\\nFifteen\\nhealthy\\nright-handed\\nsubjects\\n(8\\nmales)\\nparticipated in this task after giving informed consent in\\naccordance with the Edinburgh Handedness Inventory. Their\\nages range from 18 to 25 years, with a mean age of 2 1.1.\\nAll subjects executed picture naming task, implemented\\nin\\na\\nstandard block\\ndesign.\\nThe\\npicture naming\\ntask\\nconsisted of five experimental interleaved with six control\\nepochs. Each experimental and control epoch lasted 24s.\\nDuring\\nthe\\nexperimental\\nperiod,\\nthe\\nparticipant\\nwas\\ninstructed to view the screen, on which exposed black-white\\npictures, and then named the pictures bleatingly. During the\\ncontrol period, crosshair (+) presented at the screen center\\nwas used\\nas an experimental resting baseline and the\\nparticipant just passively viewed the screen. The session\\nlasted 4 minutes and 24 seconds.\\nThe fMRI scanning was performed with a Siemens 1.5'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content='was used\\nas an experimental resting baseline and the\\nparticipant just passively viewed the screen. The session\\nlasted 4 minutes and 24 seconds.\\nThe fMRI scanning was performed with a Siemens 1.5\\nTesla scanner at Xuan Wu Hospital in Beijing. A gradient\\necho EPI sequence was used during functional scans (TR=\\n2000ms,\\nTE=40ms,\\nmatrix\\nsize\\n64x64,\\nfield\\nof view\\n220x220mm2, 6mm thickness with 1.2mm gap, and 20\\nslices over whole brain).\\n132 volumes of images were\\nacquired.\\nThe\\nDTI\\nacquisition\\nsequence\\nwas\\na\\nsingle-shot\\nspine-echo\\nplanar\\nimaging\\n(SE/EPI)\\n(TR=61OOms,\\nTE=106ms, 128X128 matrix size (256X256 reconstructed),\\n230x230mm2 field of view, 4mm thickness with 4mm gap,\\n30\\nslices\\nover\\nwhole\\nbrain,\\na maximum\\nb-value\\nof\\n1000mm2/s in 6 noncollinear directions ). The reconstructed\\nvoxel size was 0.9x0.9x4.Omm3.\\nB. Functional MRI dataprocessing\\nData\\nwere\\npreprocessed\\nusing\\nSPM2\\nsoftware\\n(www.fil.ion.ucl.ac.uk/spm). The first 5 time points were'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content='voxel size was 0.9x0.9x4.Omm3.\\nB. Functional MRI dataprocessing\\nData\\nwere\\npreprocessed\\nusing\\nSPM2\\nsoftware\\n(www.fil.ion.ucl.ac.uk/spm). The first 5 time points were\\ndiscarded for scanner calibration and for subjects to get used\\nto the circumstance. Redundant 127 time series were kept\\nfor further analysis. Images were corrected for movement by\\nusing\\nleast-square\\nminimization\\nwithout\\nhigher-order\\ncorrections for spin history and normalized to stereotaxic\\ncoordinates of Talairach and Tournoux. Images were then\\nresampled into 3x3x4mm3 and smoothed with an 8-mm\\nGaussian kernel to decrease spatial noise.\\nStatistical\\nanalysis was performed on individual and\\ngroup data by using the general linear model and Gaussian\\nrandom\\nfields theory\\nas implemented\\nin SPM2. Group\\nanalyses were performed using a random-effects model that\\nestimates the error variance across subjects, and therefore\\nprovided a stronger generalization to the population from'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content='in SPM2. Group\\nanalyses were performed using a random-effects model that\\nestimates the error variance across subjects, and therefore\\nprovided a stronger generalization to the population from\\nwhich data are acquired. Statistical maps were superimposed\\non normalized high-resolution Ti -weighted images and their\\nlocations\\ninterpreted\\nusing\\nknown\\nneuroanatomical\\nlandmarks.\\nA commonly used approach for identifying networks of\\nfunctionally connected brain regions is temporal correlation\\nmethod involving defining a cluster of activated voxels in a\\nregion of interest (ROI) and using this \"seed\" to determine\\nwhich voxels throughout the entire brain are functionally\\ninteracting with the seed. The seed region\\nis generally\\nchosen based on its known involvement in the behavioral\\ntask and the\\nresearcher\\'s\\ninterest\\nin\\ncharacterizing\\nits\\ninteractions with other regions of brain. In this case, the\\nmiddle occipital gyrus in the left cerebrum was used as a'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content=\"task and the\\nresearcher's\\ninterest\\nin\\ncharacterizing\\nits\\ninteractions with other regions of brain. In this case, the\\nmiddle occipital gyrus in the left cerebrum was used as a\\nseed because the experiment was designed to test a specific\\nhypothesis\\nconcerning\\ninterhemispheric\\ninteractions\\nbetween visual cortices and between visual cortex and\\nsemantic processing cortex. Seen from the group activation\\nmap, in the middle occipital gyrus, the voxel at [-36 -90 0]\\n(Talairach coordinates) had the highest peak t score and was\\ntaken as a center of a sphere with a radius of 8mm, which\\nwas selected as the ROI (seed). Then, we applied the ROI to\\nthe subjects to analyze the functional connectivity.\\nBefore\\nthe\\nfMRI\\nanalysis,\\ntemporal\\nsmoothing\\nand\\nscaling steps were performed across all brain voxels as\\nfollows. Time series of all brain voxels was smoothed\\nthrough\\na\\n3-point\\nhamming\\nwindow\\nto\\nreduce\\nthe\\nhigh-frequency noise and the effect of low-frequency drift.\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content=\"follows. Time series of all brain voxels was smoothed\\nthrough\\na\\n3-point\\nhamming\\nwindow\\nto\\nreduce\\nthe\\nhigh-frequency noise and the effect of low-frequency drift.\\nAfter this, to minimize the effect of global drift, voxel\\nintensities were scaled by dividing each time point's value\\nby the mean value of the whole-brain image at that time\\npoint. To perform the fMRI analysis, time series from the\\nscan were extracted for the seed by weighted-averaging the\\ntime series of all voxels in the seed. The weight for every\\nvoxel in the seed was the proportion of its t score to the sum\\nof\\nall\\nthe\\nt\\nscores\\nin\\nthe\\nseed.\\nThe\\nresulting\\nweighted-averaged time series served as a reference model\\nfor cross-correlation analysis with the whole brain voxels,\\nyielding a spatial zero-lag cross-correlation map.\\nC. DTI data processing\\nDiffusion weighted images\\nS\\napplied with\\ndiffusion\\nsensitizing field gradient G were acquired based on the\\nStejeskal-Tanner spin echo Scheme and signal value So\\nwithout the G:\\n_S\\n2 G 2\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 1}, page_content='Diffusion weighted images\\nS\\napplied with\\ndiffusion\\nsensitizing field gradient G were acquired based on the\\nStejeskal-Tanner spin echo Scheme and signal value So\\nwithout the G:\\n_S\\n2 G 2\\n2 ( A --) D\\nSe-\\nb\\n0O\\n(1)\\nWhere b is the diffusion sensitization. In this case, the\\nnonlinear directions ofthe gradient were six:\\ngl = 1 /2(1,0, 1-)\\ng2 = 1/ V/2(1, -1,0)\\ng3 = 1/\\n2(0,-1,1)\\ng4 = 1/\\n2(1,0,1)\\ng5 = I /\\n/2(1, 1,0)\\ng6 = I/+2(0,1,l) ,\\nFig. 1. Several slices of brain activated regions. Colors denote the strong\\ndegree of activity.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 2}, page_content=\"2007 IEEE/ICME International Conference on Complex Medical Engineering\\n1/\\n...\\ni.\\n:M\\nFig. 2. Location of ROI (white area). It's in the middle occipital gyrus at\\n[-36 -90 0] (r=8mm, Talairach coordinates)\\nThe images of the S and SO\\nis presented as follows\\n(Fig.3):\\n4;/\\nFig. 4. the diffusion ellipsoid\\nFig. 3. images at six nonlinear gradient directions and without gradient\\nThus, tensor matrix is derived from (1):\\nD\\n-ln\\nb\\nS0\\n(2)\\nWhere D is a 3*3 symmetric matrix with six freedom\\ndegrees:\\n(3)\\nThen the diffusion tensor matrix D was diagonalized for\\nobtaining three eigenvalues (XQl>2>23) and corresponding\\neigenvectors (el>e2>e3) that represented three axes of the\\ndiffusion ellipsoid (Fig.4).\\nWe can calculate some diffusion index from the three\\neigenvalues to scale the degree of the diffusion. The most\\ncommon index\\nis the\\nfractional anisotropy (FA) which\\ndenoted\\nthe\\ndegree\\nof\\nthe\\ndiffusion\\nanisotropy\\nwas\\ncalculated\\nfrom\\nthe\\nthree\\neigenvalues\\nranging\\nfrom\\n0\\n(isotropy) to\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 2}, page_content='common index\\nis the\\nfractional anisotropy (FA) which\\ndenoted\\nthe\\ndegree\\nof\\nthe\\ndiffusion\\nanisotropy\\nwas\\ncalculated\\nfrom\\nthe\\nthree\\neigenvalues\\nranging\\nfrom\\n0\\n(isotropy) to\\n1 (maximum anisotropy) [10]:\\nFA = 13[(A1- A\\n+ (A2 - A )2 + (A3 - A )2]\\n_ aA12\\n+ A\\n2\\n+\\n)\\n(4)\\nWhere X denotes the mean of the three eigenvalues, which\\nis equal to the directionally averaged diffusivity.\\nThe major eigenvector\\ncorresponding\\nto\\nthe maximum\\neigenvalue denotes the local fiber orientation. The vector\\norientation maps were generated by mapping the major\\neigenvector directional components into RGB color along x,\\ny,\\nand\\nz\\nand weighting\\nthe\\ncolor\\nbrightness by\\nFA.\\nConventionally the directional RGB color mapping is red\\nfor\\nleft-right,\\ngreen\\nfor\\nanteroposterior,\\nand\\nblue\\nfor\\nsuperior-inferior [17].\\nD. combination ofthefMRI and DTI\\nThe fMRI and DTI data were both normalized to a same\\nEPI-template to acquire some normalized parameters before\\nthe individual processing of the two kinds of data. These'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 2}, page_content=\"The fMRI and DTI data were both normalized to a same\\nEPI-template to acquire some normalized parameters before\\nthe individual processing of the two kinds of data. These\\nparameters were preserved until the combination. After\\nobtaining\\nthe\\nfunctional\\nconnectivity\\nmaps\\nand\\nthe\\ndirectional\\neigenvector\\ncolor\\nmaps,\\nthese\\nnormalized\\nparameters were applied to them for normalization. Finally,\\nwe overlaid the normalized functional connectivity maps\\nupon the directional color maps.\\nIII. RESULTS AND CONCLUSIONS\\nAll subjects' fMRI data and DTI data were processed and\\ncombined correspondingly. We chose the representative\\nresult of 12th subject to show the correlation of the structure\\nand function.\\nSignificantly activated brain regions that covary with one\\nanother of PN task are interpreted as functional connected.\\nFrom the functional connectivity maps, we could observe\\nthat there was evident strong inter-hemispheric correlation\\nbetween occipital lobe, which was responsible for visual\\n|\\nl\\n1\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 2}, page_content=\"From the functional connectivity maps, we could observe\\nthat there was evident strong inter-hemispheric correlation\\nbetween occipital lobe, which was responsible for visual\\n|\\nl\\n1\\n......................\\ncorr\\nM0.5\\nFig. 5. Correlation maps showing the interregional temporal correlation\\nbetween ROI (white area) and all brain voxels. The colored regions are\\nconsidered in a same functional connectivity network.\\nP.\\ni.u\\nD-,(-.'\\nD=, .DOW\\ni.u\\nZ.z\\nD.z\\nD!7\\nD=\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content='2007 IEEE/ICME International Conference on Complex Medical Engineering\\nFig. 6. Two slices of directional color maps overlaid by functional\\nconnectivity network. The color indicates direction as follows: red means\\nleft-right, green means antero-posterior, and blue means superior-inferior.\\nBrightness is proportional to FA. Functional connected regions were\\nprofiled in white color.\\nprocessing,\\nand\\nthere\\nwas\\nalso\\nsignificant\\ncorrelation\\nbetween middle occipital gyrus and middle frontal gyrus in\\nleft\\ncerebrum,\\nwhich\\nwas\\nresponsible\\nfor\\nsemantic\\nprocessing (Fig. 5).\\nThe\\ndirectional\\ncolor\\nmaps\\noverlaid\\nby\\nfunctional\\nconnectivity networks obviously depicted the fiber pathways\\nbetween the functional connected regions. From the maps,\\nthe green color compound with blue indicated the structural\\nconnectivity between the occipital gyrus and the middle\\nfrontal\\ngyrus\\nboth\\nin\\nleft\\ncerebrum\\nalong\\nthe\\nantero-posterior direction (Fig. 6 left). The green compound'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content=\"connectivity between the occipital gyrus and the middle\\nfrontal\\ngyrus\\nboth\\nin\\nleft\\ncerebrum\\nalong\\nthe\\nantero-posterior direction (Fig. 6 left). The green compound\\nwith red color indicated that the optic radiant fiber derived\\nfrom\\nthe\\nbilateral\\nvisual\\ncortex\\ntrack\\nalong\\nthe\\nantero-posterior direction a short distance and then joint at\\nthe lateral geniculate body along the left-right direction (Fig.\\n6 right).\\nIV. DISCUSSION\\nThe combination of fMRI and DTI is related to many\\naspects. First, data acquisition is important for combination\\nof fMRI and DTI. The ideal data are acquired by employing\\nclosely matched EPI sequences for both datasets. The most\\ninfluential parameters are the voxel size and the number of\\nwhole brain\\nslices. Though there\\nare some differences\\nbetween the two scan technologies, we suggest the data\\nparameters be as matched as possible. If this condition can't\\nbe satisfied, we should do normalization before combination.\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content=\"are some differences\\nbetween the two scan technologies, we suggest the data\\nparameters be as matched as possible. If this condition can't\\nbe satisfied, we should do normalization before combination.\\nAll the original images of fMRI and DTI were normalized to\\nthe EPI template and preserved the parameters. After the\\nindividual processing of the two kinds of data, we used the\\nparameters\\nobtained\\nbefore\\nto\\nnormalize\\nthem.\\nThis\\nprocedure will bring some distortion to the images. In\\nfurther researches, the methods of matching the two data\\nwill be also the main aspects of study on combination.\\nFunctional connectivity networks obtained in our work\\nare temporal correlation method. The chosen of ROI and the\\nweighted values of the time series in ROI is the key point of\\nthis method and can be optimized in further researches. Also\\nwe could use ICA or other models to acquire the functional\\nconnectivity networks.\\nStructure\\nconnections\\ncan\\nbe\\nobtained\\nby\\nfiber\"),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content='this method and can be optimized in further researches. Also\\nwe could use ICA or other models to acquire the functional\\nconnectivity networks.\\nStructure\\nconnections\\ncan\\nbe\\nobtained\\nby\\nfiber\\ntractography, FMT algorithm, PICo tractography algorithm\\nor directional eigenvector color maps as mentioned before.\\nFMT\\nand\\nPICo\\ntractography\\nalgorithm\\nonly\\nprovide\\nconnectivity probability between brain regions, so they are\\nless visual than the other two. In this work, we chose\\ndirectional color maps to denote the fiber orientation. This\\nmethod provide a concise approach to typically visualize the\\nfiber\\nmajor\\ntract\\nwithout\\nthe\\nmore\\nadvanced\\nand\\nsophisticated tractography techniques which will take longer\\nto reach practice.\\nSince\\nit\\nalso can indicate the\\nfiber\\norientation, we often use it when we need rapid analysis of\\nsome cognitive functional connectivity or some clinical\\ncases. Furthermore, basing on the results of this method,\\nresearchers\\ncan\\ndetermine\\nweather\\nthey\\nneed\\nfiber'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content='some cognitive functional connectivity or some clinical\\ncases. Furthermore, basing on the results of this method,\\nresearchers\\ncan\\ndetermine\\nweather\\nthey\\nneed\\nfiber\\ntractography to provide more specific information of white\\nmatter structure.\\nPicture Naming (PN) cognitive task was chosen in our\\nstudy\\nfor\\nits\\nsimple\\nand\\nacknowledged\\nfunctional\\nconnectivity network [19]. The results of combination are\\nconsistent\\nwith\\nour\\nhypothesis\\nthat\\ncerebra\\nfunctional\\nconnectivity\\nhas\\nstructure\\nconnectivity\\nas\\nits\\nbasis\\nin\\nspecific cognitive task.\\nThis study may provide a basis for future studies to reveal\\nthe underlying neural networks associated with function and\\nstructure.\\nACKNOWLEDGMENT\\nThis\\nwork\\nis\\nsupported\\nby\\nNatural\\nScience\\nKey\\nFoundation of Beijing (4061004), National Natural Science\\nFoundation of China (60628101), and National High-tech\\nR&D Program (863 Program, 2006AAOlZ132).\\nREFERENCES\\n[1] D.J. Werring, C.A. Clark, and G.J.M. Parker, \"A Direct Demonstration'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content='Foundation of China (60628101), and National High-tech\\nR&D Program (863 Program, 2006AAOlZ132).\\nREFERENCES\\n[1] D.J. Werring, C.A. Clark, and G.J.M. Parker, \"A Direct Demonstration\\nof both Structure and Function in the Visual System: Combining\\nDiffusion\\nTensor\\nImaging\\nwith\\nFunctional\\nMagnetic\\nResonance\\nImaging\", NeuroImage, Vol. 9, pp.352-361, 1999\\n[2]\\nM.\\nGuye,\\nand\\nG.J.M.\\nParker,\\n\"Combined\\nfunctional MRI\\nand\\ntractography to demonstrate the connectivity of the human primary\\nmotor cortex in vivo\", NeuroImage, Vol. 19, pp.1349-1360, 2003\\n[3] K.J. Friston, \"Functional and effective connectivity in neuroimaging: a\\nsynthesis\", Hum. Brain Mapp., Vol. 2, pp. 56-78, 1994\\n[4] M. Hampson, B.S. Peterson, P. Skudlarski, J.C. Gatenby, and J.C, Gore,\\n\" Detection of functional connectivity using temporal correlations in\\nMR Images\", Hum. Brain Mapp., Vol. 15, pp.247-262, 2002\\n[5] C.F. Westin, S.E. Maier, H. Mamata, \"Processing and visualization for'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 3}, page_content='\" Detection of functional connectivity using temporal correlations in\\nMR Images\", Hum. Brain Mapp., Vol. 15, pp.247-262, 2002\\n[5] C.F. Westin, S.E. Maier, H. Mamata, \"Processing and visualization for\\ndiffusion tensor MRI\", Mledical Image Analysis, Vol. 6, pp.93-108,\\n2002\\n[6] E.R. Melhem, S. Mori, and G. Mukundan, \"Diffusion Tensor MR\\nImaging of the Brain and White Matter Tractography\", AJR , ppI78,\\n2002\\n[7]\\nP.J.\\nBasser,\\nJ.\\nMatiello,\\nand D.L.\\nBihan, \"MR diffusion tensor\\nspectroscopyand imaging\", Biophys, Vol. 66, pp. 259-267,1994\\n[8] S.L. Heller, and L.A. Heiera, \"Evidence of cerebral reorganization\\nfollowing\\nperinatal\\nstroke\\ndemonstrated\\nwith\\nfMRI\\nand\\nDTI\\ntractography\", Clin. Imaging, Vol. 29, pp.283-287, 2005\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 4}, page_content='2007 IEEE/ICME International Conference on Complex Medical Engineering\\n[9] T. Schonberg, P. Pianka, and T. Hendler, \"Characterization of displaced\\nwhite\\nmatter by\\nbrain tumors\\nusing combined DTI\\nand fMRI\",\\nNeuroImage, Vol. 30, pp. 1100-1111, 2006\\n[10]\\nP.\\nBasser,\\nand C.\\nPierpaoli,\\n\"Microstructural\\nand physiological\\nfeaturesof tissues elucidated by quantitative-diffusion-tensor\", MRI[J].J\\nMagn Reson B, Vol. 111, pp.209-219, 1996\\n[11] Y. Masutani, and S. Aoki, \"MR diffusion tensor imaging: recent\\nadvance and new techniques for diffusion tensor visualization\", Eur. J.\\nRadio., Vol. 46, pp.53-66, 2003\\n[12] D. Le Bihan, J.F. Mangin, and C. Poupon, \"Diffusion Tensor Imaging:\\nConcepts and Applications\", J. magn. reson. imaging, Vol.\\n13, pp.\\n534-546, 2001\\n[13]\\nA.T.\\nToosy,\\n0.\\nCiccarelli,\\nand\\nG.J.M.\\nParker,\\n\"Characterizing\\nfunction-structure\\nrelationships\\nin the human\\nvisual\\nsystem with\\nfunctional MRI and diffusion tensor imaging\", NeuroImage, Vol. 21,\\npp. 1452- 1463, 2004'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 4}, page_content='Ciccarelli,\\nand\\nG.J.M.\\nParker,\\n\"Characterizing\\nfunction-structure\\nrelationships\\nin the human\\nvisual\\nsystem with\\nfunctional MRI and diffusion tensor imaging\", NeuroImage, Vol. 21,\\npp. 1452- 1463, 2004\\n[14] M.L. Seghier, F. Lazeyras, and S. Zimine, \"Combination of\\nevent-related fMRI and diffusion tensor imaging in an infant with perinatal\\nstroke\\', NeuroImage, Vol. 21, pp.463-472, 2004\\n[15]\\nP.J.\\nBasser,\\nand\\nD.K.\\nJones,\\n\"Diffusion-tensor\\nMRI:\\ntheory,\\nexperimental\\ndesign and\\ndata\\nanalysis-a technical\\nreview\", NMR\\nBiomed., Vol. 15, pp.456-467, 2002\\n[16] C.R. Tench, P.S. Morgan, M. Wilson, and L.D. Blumhardt, \"White\\nMatter Mapping Using Diffusion Tensor MRI\", Magn. Reson. Med.,\\nVol. 47, pp.967-972, 2002\\n[17] B.J. Jellison, A.S. Field, J. Medow, and M. Lazar, \"Diffusion Tensor\\nImaging of Cerebral White Matter: A Pictorial Review of Physics, Fiber\\nTract Anatomy, and Tumor Imaging Patterns\", AJNR Am JNeuroradiol,\\nVol. 25, pp. 356-369, 2004\\n[18]\\nS.\\nMori,\\nand van\\nZijl\\nP.C.M.,\\n\"Fiber tracking:'),\n",
       " Document(metadata={'producer': '', 'creator': '', 'creationdate': '2007-10-15T13:03:01+05:30', 'source': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'file_path': '..\\\\data\\\\literature\\\\A_Study_of_Combination_fMRI_and_DTI_in_A_Cognitive_Task.pdf', 'total_pages': 5, 'format': 'PDF 1.6', 'title': '', 'author': '', 'subject': '', 'keywords': '', 'moddate': \"D:20071105111301-05'\", 'trapped': '', 'modDate': \"D:20071105111301-05'\", 'creationDate': \"D:20071015130301+05'30'\", 'page': 4}, page_content='Tract Anatomy, and Tumor Imaging Patterns\", AJNR Am JNeuroradiol,\\nVol. 25, pp. 356-369, 2004\\n[18]\\nS.\\nMori,\\nand van\\nZijl\\nP.C.M.,\\n\"Fiber tracking:\\nprinciples\\nand\\nstrategies\\na technical review\", NMR Biomed., Vol. 15, pp. 468-480,\\n2002\\n[19] C. Whatmough, and H. Chertkow, \"Dissociable brain regions process\\nobject\\nmeaning\\nand\\nobject\\nstructure\\nduring\\npicture\\nnaming\",\\nNeuropsychologia, Vol. 40, pp.174-186, 2002\\nw90\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:36:26 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='Hybrid CNN-RNN Models for Multimodal \\nAnalysis of Autism Spectrum Disorder \\nNeuroimaging \\nRajat Kishor Varshney \\nSchool of Computing Science & \\nEngineering, Research Scholar, Galgotias \\nUniversity, Greater Noida ,203201, India \\nrajatvarshney1207@gmail.com \\n \\n \\nAlok Katiyar \\nSchool of Computing Science & \\nEngineering, Professor, Galgotias \\nUniversity, Greater Noida ,203201, India \\nalok_katiyar@hotmail.com \\n \\n \\nPrashant Johri \\nSchool of Computer Application & \\nTechnology, Professor, Galgotias \\nUniversity, Greater Noida ,203201, India \\njohri.prashant@gmail.com \\n \\nAbstract. ASD is a developmental disorder in the \\nneurological category and has numerous and diverse symptoms. \\nDiagnosis \\nis \\nstill \\nproblematic \\nbecause \\nsymptoms \\nof \\nparkinsonism are diverse, and the dimensionality of the \\nneuroimaging data is high. This paper presents a combination \\nof CNN and RNN for classifying the ASD by analyzing \\nmultimodal neuroimaging data such as fMRI data, MRI data,'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='neuroimaging data is high. This paper presents a combination \\nof CNN and RNN for classifying the ASD by analyzing \\nmultimodal neuroimaging data such as fMRI data, MRI data, \\nand DTI data. The CNN component analyzes spatial \\ncharacteristics of the neuroimaging data, and the RNN learns \\ntemporal patterns in fMRI series. Enhancements include a novel \\npreprocessing step called 3D Brain Region Alignment (3D-BRA) \\nas well as the 3D version of Adaptive Temporal, minimizing \\ndimensionality through an innovative feature selection \\ntechnique employing EN-CC. Predictive analysis with the \\nhybrid model utilizes standard benchmark datasets such as \\nABIDE I, ABIDE II, HCP, and SFARI and records classification \\nperformances \\nover \\n95 \\npercentage. \\nTo \\nenhance \\nthe \\ninterpretability of the model, the Grad-CAM algorithm is used, \\nwhich marks specific regions of the brain important in the \\ncontext of ASD. The proposed framework seems to possess'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='To \\nenhance \\nthe \\ninterpretability of the model, the Grad-CAM algorithm is used, \\nwhich marks specific regions of the brain important in the \\ncontext of ASD. The proposed framework seems to possess \\nconsiderable feasibility for enhancing the diagnosis of ASD and \\nimproving the understanding of its neural substrates, making \\nthe developed method highly advantageous for applications in \\nneuroinformatics and clinical utility. \\nKeywords: Autism Spectrum Disorder, Neuroimaging, fMRI, \\nHybrid CNN-RNN, Deep Learning, Brain Connectivity, Feature \\nSelection, Multimodal Analysis. \\nI. \\nINTRODUCTION \\nASD is a developmental disorder with notable deficits in \\nsocial skills acts, verbal and nonverbal communication and \\nrepetitive and stereotyped behavior. ASD identification has \\nhistorically been made based on behavioral approaches; \\nnevertheless, such procedures are limited as a result of diverse \\nsymptomatology. \\nBrain \\nimaging \\nmethods, \\nincluding'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='historically been made based on behavioral approaches; \\nnevertheless, such procedures are limited as a result of diverse \\nsymptomatology. \\nBrain \\nimaging \\nmethods, \\nincluding \\nfunctional Magnetic Resonance Imaging (fMRI) and \\nStructural MRI, are also beneficial to the organization in \\nstudying ASD, since they help to establish a different brain \\nconnectivity and structure. These advancements have created \\na new avenue and strategy in which neuroimaging data can be \\napplied in computational models to aid in ASD diagnosis and \\nits categorization[1]. \\nUnlike conventional machine learning techniques that \\nattempt to develop hypothesis for single layer inputs, deep \\nlearning has made it possible to extract high level abstractions \\nfrom high dimensional medical imaging data. Earlier, \\nConvolutional Neural Networks (CNNs) have performed well \\nin identifying spatial patterns of neuroimaging data, and \\nRecurrent Neural Networks (RNNs) are quite effective in'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='Convolutional Neural Networks (CNNs) have performed well \\nin identifying spatial patterns of neuroimaging data, and \\nRecurrent Neural Networks (RNNs) are quite effective in \\ncapturing temporal dynamic[2]. Although both of these \\narchitectures have their weakness when using neuroimaging \\ndata for ASD classification, integrating CNN and RNN into a \\nCNN-RNN hybrid model advantageously, thus making it a \\nsuitable framework for the classification of ASD. Working \\nwith neuroimaging data is very difficult because such data is \\nhigh-dimensional, noisy, and can differ greatly between \\nindividuals. Moreover, while fMRI data and MRI data are \\ncollected with temporal resolution in terms of time, and spatial \\ndistribution in terms of space, the complexity of these two \\ndimensions in the data highlights the need for adequately \\ncapturing both temporal and spatial components of the data. \\nThese then pose challenge to standard approaches hence the'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='dimensions in the data highlights the need for adequately \\ncapturing both temporal and spatial components of the data. \\nThese then pose challenge to standard approaches hence the \\npoor results performances in categorizing ASD. These \\nchallenges can be tackle by a hybrid CNN-RNN framework \\nthat can capture the spatial and temporal features of an image \\nand thus performs a comprehensive feature extraction.  \\nMulti-modal neuroimaging combines fMRI, structural \\nMRI, and DTI, which shows superior informative content in \\ncomparison with only fMRI or sMRI. All these modalities \\nbring in differing information; fMRI reveals functional \\nconnection \\nwhile \\nMRI \\nand \\nDTI \\nreveal \\nstructural \\ncharacteristics [3]. These should be integrated models where \\nadditional data can enrich the result by adding complementary \\ndata from the other modalities. Convolutional Neural \\nNetworks are applied successfully for analyzing the spatial \\ncharacteristics of the neuroimaging data. To localize the ASD'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='data from the other modalities. Convolutional Neural \\nNetworks are applied successfully for analyzing the spatial \\ncharacteristics of the neuroimaging data. To localize the ASD \\nsymptoms, CNNs used contextual convolutional filters and \\nsuccessfully identified the active regions in the brain as well \\nas the structural changes. The architectures including ResNet-\\n50 and Efficient Net have been tested on large-scale datasets, \\ntherefore can extract hierarchical features for spatial feature \\nextraction in this hybrid way [4]. \\nDue to the time dependency of the data, especially in fMRI \\nrecordings, mod- els that can handle time series data are \\nrequired. Recurrent Neural Networks, especially Long Short \\nTerm Memory (LSTM) networks, are ideal in observing \\ntemporal behaviors and variations in the neural activity. \\nIncorporation of the RNNs into the hybrid model affords \\nidentification of dynamic connectivity pat- terns important in \\nelucidating ASD. The combination of CNN and RNN in the'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='Incorporation of the RNNs into the hybrid model affords \\nidentification of dynamic connectivity pat- terns important in \\nelucidating ASD. The combination of CNN and RNN in the \\nproposed architecture takes advantage of spatial and temporal \\ncharacteristics of the neuroimaging data. CNNs capture spatial \\nrelations which are then provided to RNNs to take temporal \\nrelations. Such integration guarantees that the hybrid model \\ncaptures temporal dynamics of the interactions between \\ncortical regions and underlying neural activities, which will \\nimprove the accuracy of classifying ASD [5]. \\nPreprocessing remains integrated for sealing for \\nneuroimaging hybrid model outcomes. To overcome such \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n155\\n979-8-3315-4237-5/25/$31.00 ©2025 IEEE\\n2025 International Conference on Automation and Computation (AUTOCOM) | 979-8-3315-4237-5/25/$31.00 ©2025 IEEE | DOI: 10.1109/AUTOCOM64127.2025.10956945'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 0}, page_content='155\\n979-8-3315-4237-5/25/$31.00 ©2025 IEEE\\n2025 International Conference on Automation and Computation (AUTOCOM) | 979-8-3315-4237-5/25/$31.00 ©2025 IEEE | DOI: 10.1109/AUTOCOM64127.2025.10956945\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='issues, brain region alignment, motion correction, and \\ntemporal filtering are performed to reduce noise as well as \\ninter-subject variations. Moreover, other techniques such as \\nElastic Net Correlation Coefficient (EN-CC) used to feature \\nselection, allow elimination of features with low correlation \\nand many more, thus making the hybrid model the best [6]. \\nHaving reliable information for medical imaging is a \\nsignificant advantage when applying deep learning, though \\none drawback is the interpretability. To overcome these \\nchallenges, Hybrid CNN-RNN models are designed equipped \\nwith application of Explainable AI techniques including \\nLayer-wise Relevance Propagation (LRP) and Grad-CAM. \\nThese methods bring the focus to propose the most distinct \\nand crucially connected brain parts necessary for the model’s \\nprediction, which can be effectively explained to clinicians \\nabout the neural basis of ASD and help in clinical decision'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='and crucially connected brain parts necessary for the model’s \\nprediction, which can be effectively explained to clinicians \\nabout the neural basis of ASD and help in clinical decision \\nmaking [7]. The purpose of this work is therefore to design a \\nCNN-RNN model for simultaneous analysis of neuroimaging \\ndata for ASD. Thus, the proposed framework aims at high \\nclassification accuracy and at the same time strives to generate \\nclinically relevant information by incorporating the advanced \\npreprocessing, spatial-temporal modeling, and explainable AI. \\nThis work contributes to the development of Advanced AI \\ntechniques dealing with neuroimaging and may have \\nimplications for the actual diagnosis of ASD and advancing \\nthe knowledge of its neurological basis [8]. \\nII. \\nRELATED WORKS \\nEarlier ASD detection based on neuroimaging data \\nemployed conventional ma- chine learning paradigms like \\nSupport Vector Machines (SVM) and Random Forests. They'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='II. \\nRELATED WORKS \\nEarlier ASD detection based on neuroimaging data \\nemployed conventional ma- chine learning paradigms like \\nSupport Vector Machines (SVM) and Random Forests. They \\ninvolved significant feature extraction and dimensionality \\nreduction methods such as Principal Component Analysis \\n(PCA) and Independent Component Analysis (ICA), as most \\nneuroimaging data are high dimensional. Although useful up \\nto some extent, these kinds of approaches did not allow the \\nauthors to decompose the relationships within the data fields \\nbased on their spatial and temporal components [9]. Deep \\nlearning emerged and made analyzing neuroimaging data \\neven easier because it allowed for the automation of feature \\nextraction. CNNs have been used more often for the \\nclassification of ASD since the method is good at finding \\nspatial features of the brain activity and structure. For \\ninstance, there are ResNet , VGG models, which have large'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='classification of ASD since the method is good at finding \\nspatial features of the brain activity and structure. For \\ninstance, there are ResNet , VGG models, which have large \\naccuracy rates for classifying ASD from MRI and fMRI data. \\nNevertheless, such approaches do not consider the time nature \\nof most neuroimaging studies [10]. \\nWhile CNNs are very efficient in extracting spatial \\nfeatures, one disadvantage is that spatial CNNs do not \\nincorporate a mechanism to encode temporal dynamics \\ninherent to most functional neuroimaging data including \\nfMRI. Temporal dimension turns out to be a critical factor in \\nstudying brain connectivity and neural activation. Therefore, \\nmore research is being done on hybrid solutions that will act \\nas temporal modeling capabilities [11]. \\n RNNs, especially the LSTM, has been used to process \\ntemporal data in neuroimaging research. LSTMs exploit \\ntemporal dependence structure in fMRI data, making it ideal'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='RNNs, especially the LSTM, has been used to process \\ntemporal data in neuroimaging research. LSTMs exploit \\ntemporal dependence structure in fMRI data, making it ideal \\nfor analysis in time-series neural activity. Research \\nincorporating RNNs in diagnosing ASD has provided positive \\noutcomes, and incorporating the CNN for spatial feature \\nextraction can improve their performance [12]. CNN- RNN \\nhybrid models have become the standard approach for \\ncombining spatial and temporal analysis models. These \\nmodels, which integrate the advantages of CNNs and RNNs, \\nafford extensive feature analysis over neuroimaging data. The \\nexperiments have demonstrated that the proposed hybrid \\narchitectures enhance classification accuracy and reveal the \\nspatial and temporal patterns of brain activity. For example, \\nthe combination of CNN and LSTM has been revealed to \\nafford better performance in ASD classification compared to \\nsingle models of CNN or RNN [13]. There is much interest in'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='the combination of CNN and LSTM has been revealed to \\nafford better performance in ASD classification compared to \\nsingle models of CNN or RNN [13]. There is much interest in \\ncombining fMRI, sMRI, and DTI, which are imaging \\ntechniques yielding data from a patient’s brain. Continuous \\nintegration of functional, structural, and connectivity \\ninformation improves the diagnostic yield of multimodal \\nmethods. The hybrid CNN-RNN structures are more relevant \\nfor solving problems with multimedia data since it is possible \\nto use the benefits of each modality in parallel [14]. \\nThe neuroimaging datasets of high dimension typically \\nsuffer from noisy or less relevant features, so the process of \\nfeature selection becomes mandatory. There is the use of \\nfeatures like Elastic Net Correlation Coefficient (EN-CC) \\ntogether with Recursive Feature Elimination (RFE) to \\nimprove the model. The latest heterogeneous CNN-RNN \\nmodel-based investigations integrate feature se- lection'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='together with Recursive Feature Elimination (RFE) to \\nimprove the model. The latest heterogeneous CNN-RNN \\nmodel-based investigations integrate feature se- lection \\ntechniques in an attempt to avoid high dimensionality, thus \\nfocusing the model towards the important connectivity \\nfeatures of the brain [15]. Data preprocessing is another \\nimportant step in filtering and enhancing neuroimaging data \\nso that they can be easily interpreted. Standardization and re- \\nduction of noise can be performed through the utilization of \\nmethods such as 3D Brain Region Alignment (3D-BRA) and \\nmotion correction. A number of temporal filtering methods \\nare then employed to remove artifacts from fMRI data while \\npreserving important connectivity patterns within the \\nbrain[16]. These have laid a strong background for hybrid \\nCNN-RNN models to reduce robustness analysis on. \\nWhen deep learning is applied to medical experiences and \\napplications, one major concern is how to explain the model’s'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='CNN-RNN models to reduce robustness analysis on. \\nWhen deep learning is applied to medical experiences and \\napplications, one major concern is how to explain the model’s \\noutput. Addressing this problem, some novel approaches, \\ncalled Explainable AI methods, including Grad-CAM and \\nLayer-wise Relevance Propagation (LRP), have been \\ndeveloped. Such techniques serve to pinpoint areas and \\nconnections most consistent with the model predictor, and \\nmay be useful to specialists[17-19]. The CNN-RNN models \\nare entirely accurate for ASD diagnosis yet their clinical \\ninterpretability \\nremains \\nhigh \\nwhen \\nintegrated \\nwith \\nexplainability tools. However, there are several issues with \\ndeep learning and neuroimaging even today. Some of the \\nchallenges they were posed stay relevant, including limited \\navailability of data, inter-subject variability, and the call for \\nmore interpretable models. Multimodal analysis with CNN–\\nRNN systems and with more involved preprocessing is'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='availability of data, inter-subject variability, and the call for \\nmore interpretable models. Multimodal analysis with CNN–\\nRNN systems and with more involved preprocessing is \\nsuggested as the promising direction for further research \\nMoreover, it is feasible to enhance the quantity and variability \\nof data through such a possibility as using larger sized data \\nsets and from other sources, e.g., ABIDE or SFARI. \\nTherefore, the future studies need to aim at various directions \\nfor the improvement of the hybrid architectures and the \\nintegration of XAI in order to establish connections between \\nnew and innovative algorithms’ solutions and feasible clinical \\nuse. \\nIII. \\nPROPOSED METHODOLOGY \\nTo this end, a novel multimodal brain image analysis \\nframework using both the CNNs and RNNs has been proposed \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n156'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 1}, page_content='To this end, a novel multimodal brain image analysis \\nframework using both the CNNs and RNNs has been proposed \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n156\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 2}, page_content='to classify patients with ASD. CNN is used for finding spatial \\nfeatures and RNN for temporal dependencies of the desired \\nobject. Such a model guarantees the significant feature \\ncoverage and uses the distinctive advantages of both networks. \\nThis is developed in Python with TensorFlow/ PyTorch and \\nwith MATLAB for data preprocessing and visualization in \\nTableau. \\n \\nFig. 1. Proposed Flowchart for Multimodal Analysis of Autism Spectrum \\nDisorder \\nA.  Data Preprocessing and Background Subtraction \\nWith preprocessing, neuroimaging data is well prepared \\nfor analysis prior to any other procedure being undertaken. \\nSubstantial noise and extraneous information in fMRI images \\nare abstracted using 3D Brain Region Alignment (3D-BRA) \\nand motion correction They include What was done during the \\npreprocessing steps: Towards this goal, a temporal mean filter \\nis used to principally filter out noise as a precursor to \\nextracting significant signals. \\nB (x, y ,t)=f (x, y ,t)− T/1 ∑்\\n௧ୀଵ'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 2}, page_content='preprocessing steps: Towards this goal, a temporal mean filter \\nis used to principally filter out noise as a precursor to \\nextracting significant signals. \\nB (x, y ,t)=f (x, y ,t)− T/1 ∑்\\n௧ୀଵ\\nf(x,y,t)                               (1) \\nwhere f (x, y, t) represents the pixel intensity at time t, and \\nT is the total number of frames.  \\nMotion correction and spatial normalization align all scans \\nto a common anatomical template.. \\nB. Feature Extraction with CNN \\nThe CNN module is the layer that helps identify the spatial \\nfeatures from the processed neuroimaging raw data. ResNet-\\n50 is used here because it is able to perform residual learning \\neffectively in a comparatively low number of layers. The \\nconvolutional layers analyze small areas of brain activation. \\n                             y = ReLU (W · x + b)    \\n (2) \\nwhere x is the input image, W are the weights of the \\nconvolutional filter, and b is the bias. Subsequent to \\nconvolution to help decrease the spatial dimensions while'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 2}, page_content='(2) \\nwhere x is the input image, W are the weights of the \\nconvolutional filter, and b is the bias. Subsequent to \\nconvolution to help decrease the spatial dimensions while \\npreserving significant features, there is max pooling. \\nC. Temporal Analysis with RNN \\nThe extracted spatial features are flattened and fed into the \\nRNN module, specif- ically a Long Short-Term Memory \\n(LSTM) network. LSTM is ideal for capturing temporal \\ndependencies in fMRI data. The LSTM cell operations are \\ndefined as follows [?]: \\nft = σ(Wf · [ht−1, xt] + bf )  \\n (3) \\nit = σ(Wi · [ht−1, xt] + bi)  \\n (4) \\nCt = ft ∗ Ct−1 + it ∗ tanh(WC · \\n[ht−1, xt] + bC)  \\n  \\n    (5) \\not = σ(Wo · [ht−1, xt] + bo)  \\n (6) \\nht = ot ∗ tanh(Ct)  \\n \\n (7) \\nHere   f_t, i_t, and o_t represent the forget, input, and \\noutput gates, respectively, while Ct is the cell state.  \\nD.  Multimodal Integration \\nIn this case, the fMRI, MRI, and DTI data are integrated \\nby combining the features of each modality together in a'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 2}, page_content='D.  Multimodal Integration \\nIn this case, the fMRI, MRI, and DTI data are integrated \\nby combining the features of each modality together in a \\ncascading manner and then fed into the RNN. This integration \\ntakes advantage of common information between two \\nmodalities to increase the classification rate. To ensure each \\nmodality is equally represented, a weighting mechanism is \\nadopted: \\nF = αFfMRI + βFMRI + γFDT I  \\n \\n(8) \\nWhere α, β, γ are the weights assigned to each modality’s \\nfeatures. \\nE. Model Training and Evaluation \\nThe hybrid model is trained using the Adam optimizer to \\nminimize the cross- entropy loss: \\nL=−\\nଵ\\nே(+ܽ) 1 ∑ N  [y i  log ( y ^  i  )+(1−y i  )log(1− y \\n^  i  )]       (9) \\nwhere ݕ ݅  is the true label, and ݕ ^ ݅  is the predicted \\nprobability. The model undergoes Stratified Group Cross-\\nValidation (SGCV) to ensure reliable performance evaluation \\nacross subject groups. Metrics such as accuracy, precision, \\nrecall, and F1-score are computed.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 2}, page_content='Validation (SGCV) to ensure reliable performance evaluation \\nacross subject groups. Metrics such as accuracy, precision, \\nrecall, and F1-score are computed. \\nF. Software and Implementation Details \\nx Programming Environment: TensorFlow and PyTorch \\nbased on Python for model development. \\nx Preprocessing Tools: MATLAB for motion correction \\nand background subtraction. \\nx Visualization: Tableau for presenting results and \\ninsights. \\nx Hardware: The datasets for training include NVIDIA \\nGPUs for accelerated training. \\nx Libraries: For data handling and preprocessing, Python \\nlibraries including NumPy, Pandas, OpenCV, and \\nScikit-learn are used. \\n \\nG. Dataset \\nThe ASD datasets employed in the present paper are \\nbenchmark datasets such as ABIDE I, ABIDE II, HCP, and \\nSFARI, which contain significant ASD data for neuroimaging \\ninvestigation. In ABIDE I and ABIDE II, data includes \\nresting- state fMRI and structural MRI from individuals with'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 2}, page_content='SFARI, which contain significant ASD data for neuroimaging \\ninvestigation. In ABIDE I and ABIDE II, data includes \\nresting- state fMRI and structural MRI from individuals with \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n157\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 3}, page_content='ASD and from typically developing controls, arriving from \\nmore than a thousand participants across several sites. The \\nreleased HCP dataset provides RS-fMRI, task-fMRI, \\nStructural MRI, and DTI data from more than 1200 healthy \\nadults. SFARI has a vast resource of gene, clinical, and \\nneuroimaging \\ndata \\nestablishing \\nthe \\nneuroanatomical \\nfoundation of ASD as well as about thousands of participants. \\nThe availability of these datasets provides a rich database for \\nunderstanding BCT profiles, structural alterations, and \\ngenomic architecture underlying ASD, as well as for training \\nand testing complex machine-learning algorithms for ASD \\nidentification. \\nSo as to allow for reproducing the results, the specifics of \\nthe proposed technique have been described in detail. This \\ninclude TensorFlow and PyTorch for model compiling, \\nMATLAB for preprocessing data and Tableau for visualizing \\nthe data. The preprocessing pipeline, containing such tools as'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 3}, page_content='include TensorFlow and PyTorch for model compiling, \\nMATLAB for preprocessing data and Tableau for visualizing \\nthe data. The preprocessing pipeline, containing such tools as \\n3D BRA and Elastic Net Correlation Coefficient for feature \\nselection, \\nis \\nexplained in detail. \\nIn \\naddition, \\nthe \\nhyperparameters used in training and evaluation are described \\nnext. In subsequent versions of this paper, there will be a \\nsecondary repository with the code, pretrained models, and \\nlinks to the datasets for research purposes. \\nIV. \\nRESULT AND DISCUSSION \\nThe hybrid CNN-RNN model was evaluated on four well-\\nestablished neuroimaging datasets: ABIDE I, ABIDE II, \\nHuman Connectome Project (HCP), and the Simons \\nFoundation Autism Research Initiative (SFARI). Data \\ncollection was obtained from online surveys. This section \\nprovides the outcome of the method suggested in this paper \\nand the analyses of the findings, complemented by tables to \\nenhance readability. \\nTABLE I.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 3}, page_content='provides the outcome of the method suggested in this paper \\nand the analyses of the findings, complemented by tables to \\nenhance readability. \\nTABLE I.  \\nCLASSIFICATION ACCURACY OF THE HYBRID CNN-RNN \\nMODEL \\nDataset \\nAccuracy (%) \\nPrecision (%) \\nRecall \\n(%) \\nF1-\\nScore \\n(%) \\nABIDE I \\n96.5 \\n95.8 \\n96.2 \\n96.0 \\nABIDE II \\n97.3 \\n97.0 \\n96.9 \\n97.0 \\nHCP \\n98.1 \\n98.0 \\n97.8 \\n97.9 \\nSFARI \\n99.0 \\n98.9 \\n99.1 \\n99.0 \\n \\nx ABIDE I: The Autism Brain Imaging Data Exchange \\nI. \\nx ABIDE II: The Autism Brain Imaging Data Exchange \\nII. \\nx HCP: Human Connectome Project. \\nx SFARI: Autism Research Initiative of the Simons \\nFoundation. \\nThe proposed hybrid CNN-RNN yielded marvelous \\nclassification accuracy ranging from 96.5% to 99.0% across \\nall datasets. Among the various datasets used in the study, the \\nhighest accuracy was observed in the SFARI dataset. This can \\nbe attributed to the larger sample size and diverse \\nrepresentation of data, which demonstrates the scalability of'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 3}, page_content='highest accuracy was observed in the SFARI dataset. This can \\nbe attributed to the larger sample size and diverse \\nrepresentation of data, which demonstrates the scalability of \\nthe proposed model. High precision and recall values justify \\nthe referability of the model to accurately classify ASD \\ninstances and to reduce the incidence of Type I and II errors. \\n            \\n \\nFig. 2. Bar graph showing the classification accuracy of the hybrid \\nCNNRNN   model on different datasets \\nThe proposed hybrid CNN-RNN model yields robust \\nclassification of ASD. Across multiple datasets, with \\nespecially remarkable accuracy on the SFARI dataset. \\nTABLE II.  \\nIMPACT OF FEATURE SELECTION ON MODEL \\nPERFORMANCE \\nModel \\n \\nAccuracy Without \\nFeature Selection \\n(%) \\nAccuracy With \\nFeature Selection \\n(%) \\nResNet-50 (CNN) \\n85.2 \\n96.5 \\nVGG-16 (CNN) \\n87.1 \\n97.3 \\nInception-v3 (CNN) \\n88.3 \\n98.1 \\nHybrid \\nCNN-RNN \\n(Proposed) \\n89.5 \\n99.0 \\nApplication of Elastic Net Correlation Coefficient (EN-'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 3}, page_content='(%) \\nResNet-50 (CNN) \\n85.2 \\n96.5 \\nVGG-16 (CNN) \\n87.1 \\n97.3 \\nInception-v3 (CNN) \\n88.3 \\n98.1 \\nHybrid \\nCNN-RNN \\n(Proposed) \\n89.5 \\n99.0 \\nApplication of Elastic Net Correlation Coefficient (EN-\\nCC) for feature selection enhanced the accuracy by about 10-\\n15% in all the investigated architectures. This result indicates \\nthe crucial need for dimensionality reduction and feature \\nselection, especially when dealing with multifaceted \\nneuroimaging datasets \\n \\n \\nFig. 3. When considering feature selection, we obtain the following bar \\ngraph of the model performance: All bars on the left side of the figure \\nindicate that the presented results have been obtained without feature \\nselection, while all bars on the right side have been obtained with feature \\nselection. \\n \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n158'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 3}, page_content='selection, while all bars on the right side have been obtained with feature \\nselection. \\n \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n158\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='To think that feature selection is still a big deal in \\nenhancing the performance of a model, and the application of \\nEN-CC significantly improved the classification rate of the \\nCNN-RNN hybrid model. \\nTABLE III.  \\nCOMPARISON WITH STATE-OF-THE-ART MODELS \\nModel \\nAccuracy \\n(%) \\nPrecision \\n(%) \\nRecall \\n(%) \\nF1-\\nScore \\n(%) \\nSVM + PCA \\n81.2 \\n79.5 \\n80.1 \\n79.8 \\nCNN (ResNet-\\n50) \\n96.5 \\n95.8 \\n96.2 \\n96.0 \\nCNN + RNN \\n(LSTM) \\n97.8 \\n97.5 \\n97.6 \\n97.5 \\nHybrid CNN-\\nRNN \\n(Proposed) \\n99.0 \\n98.9 \\n99.1 \\n99.0 \\n The proposed hybrid CNN-RNN model yielded better \\nresults than earlier methods, particularly in accuracy and F1-\\nscores. By combining CNNs to ana- lyze spatial features and \\nRNNs to determine temporal features, the model was able to \\nidentify intricate patterns in the neuroimaging datasets. The \\ncomparison with traditional models such as SVM+PCA  \\n \\nFig. 4. The bar graphical representation showing a comparison between the \\nmodel-M3 and the model-M4 on the basis of Accuracy, precision, recall, and'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='Fig. 4. The bar graphical representation showing a comparison between the \\nmodel-M3 and the model-M4 on the basis of Accuracy, precision, recall, and \\nF1 score. \\nTABLE IV.  \\nEXPLAINABILITY ANALYSIS USING GRAD-CAM \\nDataset \\nKey Brain Regions \\nIdentified \\nRelevance to ASD \\nABIDE I \\nPrefrontal Cortex, \\nTemporal Lobe \\nAssociated with social and \\ncommunication skills. \\nABIDE II \\nAmygdala, \\nHippocampus \\nKey roles in emotion and \\nmemory processing. \\nHCP \\nDefault Mode Network \\n(DMN) \\nCritical for self-referential \\nthought. \\nSFARI \\nSocial Cognition \\nCircuitry, Insula \\nRelevant to empathy and \\nemotional regulation \\nFurthermore, the Grad-CAM analysis offered insights into \\nthe brain regions from which the model draws its decisions. \\nFor instance, the Default Mode Net- work (DMN) detected in \\nthe HCP dataset corresponds to regions frequently implicated \\nin ASD individuals, particularly in terms of self-related \\nprocessing. The prefrontal cortex and temporal lobe were'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='the HCP dataset corresponds to regions frequently implicated \\nin ASD individuals, particularly in terms of self-related \\nprocessing. The prefrontal cortex and temporal lobe were \\nmentioned most often, as expected, due to social and \\ncommunicative dysfunction in the disorder \\nThe hybrid CNN-RNN used for sound analysis is coupled \\nwith Grad-CAM for better explain- ability and it identified key \\nregions of the brain related to ASD including the prefrontal \\ncortex, the amygdala, and the Default Mode Network (DMN). \\nThis kind of interpretability is fundamental to reconcile AI-\\ndriven predictions with clinical practice. The approach \\nguarantees the effective utilisation of the multitask \\npreprocessing which focuses on machines to apply different \\npreprocessing techniques and also ensures merging of \\nmultimodal neuroimaging data; In other words, the \\npreprocessing stage plays a crucial role of ensuring effective \\nfeature extraction throughout the diagnostic procedure as a'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='multimodal neuroimaging data; In other words, the \\npreprocessing stage plays a crucial role of ensuring effective \\nfeature extraction throughout the diagnostic procedure as a \\nresult of advanced spatial–temporal analyses. The study \\nstrengths also relate to the clinical applicability for \\ninterpersonal psychotherapy and the research utility of the \\nproposed framework. \\nV. \\nCONCLUSION \\nThis paper proposes a new CNN–RNN system for the \\nmultimodal investigation of ASD based on neuroimaging \\ndata. An important facet of the proposed method- ology \\nprovides compatibility with spatial and temporal feature \\nextraction while overcoming the problems introduced by \\nfMRI, structural MRI, and DTI streams given the high \\ndimensionality of the data. Program enhancements like 3D \\nBrain Region Alignment (3D-BRA) and Adaptive Temporal \\nFiltering pre-processing essentially improve the data quality \\nof this model. Overall, the proposed hybrid CNN-RNN model'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='Brain Region Alignment (3D-BRA) and Adaptive Temporal \\nFiltering pre-processing essentially improve the data quality \\nof this model. Overall, the proposed hybrid CNN-RNN model \\nexhibited high classification accuracy of above 95 percentage \\nin different data sets, that are ABIDE I, ABIDE II, HCP and \\nSFARI. The inclusion and exclusion step based on Elastic Net \\nCorrelation Coefficient (EN-CC) increased the model \\naccuracy by shrinking the model down to only 10, 15 or 20 \\nimportant features out of thousands of gene features that are \\nrelevant for separating cancerous tissues from normal tissues. \\nAdding methods like Grad-CAM as a part of explainable AI, \\nthe authors were able to determine that the model focused on \\nthe most critical regions of the brain related to ASD and that \\nthe results will be beneficial for clinicians to make decisions \\nbased on the AI model’s predictions. The findings show that \\nthe application of the combined deep learning architecture can'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='the results will be beneficial for clinicians to make decisions \\nbased on the AI model’s predictions. The findings show that \\nthe application of the combined deep learning architecture can \\nenhance the significance of ASD diagnosis and boosted clf by \\nmaking it more accurate and easier to understand. Through \\nintegration \\nof \\nmultimodal \\nneuroimaging \\ndata \\nand \\ndevelopment of highly efficient computational methods, this \\napproach provides a possibility to improve the clinical \\npractice and reveal the mechanisms of ASD’s development. \\nKalecki-type equations and nonlinear least squares estimation \\nused in this paper can be subjected to more extensive \\nvariations and use a much broader range of data from different \\npopulations in future studies. \\nThe study also recognizes that the Crowdbase was \\nsomewhat limited by the kind of datasets available and the \\nlack of much greater diversity and representation. Despite \\nusing benchmark datasets including ABIDE I and II, HCP,'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='somewhat limited by the kind of datasets available and the \\nlack of much greater diversity and representation. Despite \\nusing benchmark datasets including ABIDE I and II, HCP, \\nand SFARI with their distinct neuroimaging and behavioral \\ncontributors, more optimization is required. In future work, we \\nplan to add more datasets from other demographic and clinical \\npoint of view. The extension of this study will increase the \\nstability and applicability of the enshrouded model to other \\ndifferent groups and diseases. \\nREFERENCES \\n[1] \\nAlhussen, A., Vinoth, N., Jenis, M.R.A., Surendran, S., Ganesh, V.D., \\nThangaraj, S.J.J.: Development of weighted ensemble deep learning \\nnetwork for surface rough- ness prediction and flank wear \\nmeasurement. \\nJ. \\nMater. \\nEng. \\nPerform. \\n(2024). \\nhttps://doi.org/10.1007/s11665-024-09726-7 \\n[2] \\nDeepa, S.N., Devi, B.A.: A survey on artificial intelligence approaches \\nfor medical image classification. Indian J. Sci. Technol. 4(11), 1583–'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 4}, page_content='https://doi.org/10.1007/s11665-024-09726-7 \\n[2] \\nDeepa, S.N., Devi, B.A.: A survey on artificial intelligence approaches \\nfor medical image classification. Indian J. Sci. Technol. 4(11), 1583–\\n1595 (2011). https://doi.org/10.17485/ijst/2011/v4i11/30291 \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n159\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 5}, page_content='[3] \\nDhamale, T.D., Bhandari, S.U., Harpale, V.K., Nandan, D.: Autism \\nspec- trum disorder detection using parallel deep convolution neural \\nnetwork and generative adversarial networks. Trait. du Signal 41(2), \\n643–652 (2024). https://doi.org/10.18280/ts.410208 \\n[4] \\nEslami, T., Saeed, F.: Auto-asd-network: A technique based on deep \\nlearning and support vector machines for diagnosing autism spectrum \\ndisorder using fmri data. In: ACM-BCB 2019 - Proc. 10th ACM Int. \\nConf. Bioinformatics, Comput. Biol. Heal. Informatics. pp. 646–651 \\n(2019). https://doi.org/10.1145/3307339.3343482 \\n[5] \\nGanesh, D.V., Bommi, R.M.: Neural network based predictive analysis \\nof sur- face roughness using bayesian regularization in turning of \\nmonel k500. In: 6th Int. Conf. Inven. Comput. Technol. ICICT 2023 - \\nProc. \\npp. \\n448–452 \\n(2023). \\nhttps://doi.org/10.1109/ICICT57646.2023.10134220 \\n[6] \\nGanesh, V.D., Bommi, R.: Flankpix: An image segmentation algorithm'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 5}, page_content='Proc. \\npp. \\n448–452 \\n(2023). \\nhttps://doi.org/10.1109/ICICT57646.2023.10134220 \\n[6] \\nGanesh, V.D., Bommi, R.: Flankpix: An image segmentation algorithm \\nfor flank wear analysis in monel k500 turning. Trait. du Signal 41(3), \\n1625–1631 (2024). https://doi.org/10.18280/ts.410349 \\n[7] \\nJung, Y.J., Han, S.H., Choi, H.J.: Explaining cnn and rnn using selec- \\ntive layer-wise relevance propagation. IEEE Access 9, 18670–18681 \\n(2021). https://doi.org/10.1109/ACCESS.2021.3051171 \\n[8] \\nKe, F., Choi, S., Kang, Y.H., Cheon, K.A., Lee, S.W.: Exploring the \\nstructural and strategic bases of autism spectrum disorders with deep \\nlearning. \\nIEEE \\nAccess \\n8, \\n153341–153352 \\n(2020). \\nhttps://doi.org/10.1109/ACCESS.2020.3016734 \\n[9] \\nNandhini, T.J., Thinakaran, K.: An improved crime scene detection \\nsystem based on convolutional neural networks and video surveillance. \\nIn: 2023 5th Int. Conf. Electr. Comput. Commun. Technol. ICECCT \\n2023. \\npp. \\n1–6 \\n(2023).'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 5}, page_content='system based on convolutional neural networks and video surveillance. \\nIn: 2023 5th Int. Conf. Electr. Comput. Commun. Technol. ICECCT \\n2023. \\npp. \\n1–6 \\n(2023). \\nhttps://doi.org/10.1109/ICECCT56650.2023.10179852 \\n[10] Saleh,  A.Y.,  Chern,  L.H.:  Autism  spectrum  disorder  classification  \\nus- ing deep learning. Int. J. online Biomed. Eng. 17(8), 103–114 \\n(2021). https://doi.org/10.3991/ijoe.v17i08.24603 \\n[11] Shah, S.K., Kumar, N.S., Gurumoorthi, E., Shnain, A.H., Kumar, A., \\nNandhini, T.J.: A detailed analysis of speculative based algorithm \\nimplementation for modish house. In: 2024 4th Int. Conf. Adv. \\nComput. \\nInnov. \\nTechnol. \\nEng. \\npp. \\n984–987 \\n(2024). \\nhttps://doi.org/10.1109/ICACITE60783.2024.10616747 \\n[12] Shao, L., Fu, C., You, Y., Fu, D.: Classification of asd based on fmri \\ndata with deep learning. Cogn. Neurodyn. 15(6), 961–974 (2021). \\nhttps://doi.org/10.1007/s11571-021-09683-0 \\n[13] Sherkatghanad, Z., et al.: Automated detection of autism spectrum'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 5}, page_content='data with deep learning. Cogn. Neurodyn. 15(6), 961–974 (2021). \\nhttps://doi.org/10.1007/s11571-021-09683-0 \\n[13] Sherkatghanad, Z., et al.: Automated detection of autism spectrum \\ndisorder us- ing a convolutional neural network. Front. Neurosci. \\n13(January), 1–12 (2020). https://doi.org/10.3389/fnins.2019.01325 \\n[14] Tang, M., Kumar, P., Chen, H., Shrivastava, A.: Deep multimodal \\nlearn- ing for the diagnosis of autism spectrum disorder. J. Imaging \\n6(6) (2020). https://doi.org/10.3390/jimaging6060047 \\n[15] Yang, X., Sarraf, S., Zhang, N.: Deep learning-based framework for \\nautism func- tional mri image classification. Journal of the Arkansas \\nAcademy of Science 72 (2018) \\n[16] Jahani, A., Jahani, I., Khadem, A. et al. Twinned neuroimaging \\nanalysis contributes to improving the classification of young people \\nwith autism spectrum disorder. Sci Rep 14, 20120 (2024). \\nhttps://doi.org/10.1038/s41598-024-71174-z \\n[17] Jainy Sachdeva, Riyaansh Mittal, Jiya Mehta, Riya Jain, Anmol'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 5}, page_content='with autism spectrum disorder. Sci Rep 14, 20120 (2024). \\nhttps://doi.org/10.1038/s41598-024-71174-z \\n[17] Jainy Sachdeva, Riyaansh Mittal, Jiya Mehta, Riya Jain, Anmol \\nRanjan, Resolving autism spectrum disorder (ASD) through brain \\ntopologies using fMRI dataset with multi-layer perceptron (MLP), \\nPsychiatry Research: Neuroimaging, Volume 343, 2024, 111858, ISSN \\n0925-4927, https://doi.org/10.1016/j.pscychresns.2024.111858. \\n[18] A. K. Singh, I. R. Khan, S. Khan, K. Pant, S. Debnath, and S. Miah, \\n“Multichannel CNN Model for Biomedical Entity Reorganization,” \\nBioMed \\nResearch \\nInternational, \\nvol. \\n2022, \\n2022, \\ndoi: \\n10.1155/2022/5765629. \\n[19] K. Maithili et al., “Optimized CNN Model for Diabetic Retinopathy \\nDetection and Classification,” International Journal of Intelligent \\nSystems and Applications in Engineering, vol. 11, no. 7s, pp. 317–331, \\n2023, [Online]. Available: Scopus Link. \\n \\n \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n160'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 18.0 (Windows); modified using iTextSharp 5.4.1 ©2000-2012 1T3XT BVBA (AGPL-version); modified using iText® Core 7.2.4 (AGPL version) ©2000-2022 iText Group NV', 'creator': 'Microsoft® Word for Microsoft 365', 'creationdate': '2025-03-09T00:31:21+05:30', 'source': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'file_path': '..\\\\data\\\\literature\\\\Hybrid_CNN-RNN_Models_for_Multimodal_Analysis_of_Autism_Spectrum_Disorder_Neuroimaging.pdf', 'total_pages': 6, 'format': 'PDF 1.6', 'title': 'Hybrid CNN-RNN Models for Multimodal Analysis of Autism Spectrum Disorder Neuroimaging', 'author': '', 'subject': '2025 International Conference on Automation and Computation (AUTOCOM);2025; ; ;10.1109/AUTOCOM64127.2025.10956945', 'keywords': '', 'moddate': '2025-04-11T15:29:29-04:00', 'trapped': '', 'modDate': \"D:20250411152929-04'00'\", 'creationDate': \"D:20250309003121+05'30'\", 'page': 5}, page_content='Systems and Applications in Engineering, vol. 11, no. 7s, pp. 317–331, \\n2023, [Online]. Available: Scopus Link. \\n \\n \\n2025 International Conference on Automation and Computation (AUTOCOM)\\n160\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 12,2025 at 08:33:17 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='Preprocessing FMRI Data In SPM12\\nFatima Ez-zahraa Bazay\\nLRIT - CNRST URAC 29\\nRabat IT Center, FS\\nMohammed V University in Rabat\\nMorocco\\nFatima Zahra Benabdallah\\nLRIT - CNRST URAC 29\\nRabat IT Center, FS\\nMohammed V University in Rabat\\nMorocco\\nAhmed Drissi El Maliani\\nLRIT - CNRST URAC 29\\nRabat IT Center, FS\\nMohammed V University in Rabat\\nMorocco\\nAbstract—\\nFunctional magnetic resonance imaging (fMRI) has ended\\nup the most famous technique for imaging brain functions.\\nCurrently, there is a giant range of software packages for the\\nanalysis of fMRI data, such as AFNI, SPM, FreeSurfer, FSL,\\nfMRIPrep and Workbench.In this work, we propose an fMRI\\npreprocessing on SPM applied to ABIDE II database and we\\nevaluate this processing by applying an MLG.\\nIndex\\nTerms—Functional\\nmagnetic\\nresonance\\nimaging\\n(fMRI),\\nPreprocessing,\\nSegmentation,Slice\\ntiming,coregistration,normalization,smoothing,Realignment\\n,ABIDE II,NIFTI.\\nI. INTRODUCTION\\nFunctional\\nMagnetic\\nresonance\\nimaging\\n(fMRI),\\na'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='resonance\\nimaging\\n(fMRI),\\nPreprocessing,\\nSegmentation,Slice\\ntiming,coregistration,normalization,smoothing,Realignment\\n,ABIDE II,NIFTI.\\nI. INTRODUCTION\\nFunctional\\nMagnetic\\nresonance\\nimaging\\n(fMRI),\\na\\ncontemporary imaging technique is an effective non-invasive,\\nsecure device used to study the brain’s features primarily\\nbased on a measure of the brain neural activation. [1]\\nThe preprocessing of raw fMRI records is a crucial step\\nearlier than performing further analyses for the following\\nreasons. First, fMRI measures spontaneous fluctuations of\\nblood-oxygen-level dependent (BOLD) signals associated with\\nneuronal activities. However, BOLD indicators comprise non-\\nneuronal contributions, such as head motion, physiological\\ncontributions, tissues outside the scope of interest, and\\nMRI-induced artifacts, as nicely as neuronal signals.( [2], [3],\\n[4])The non-neuronal aspects of BOLD signals complicate\\nthe interpretation of fMRI signals. Secondly, the remarkable'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='MRI-induced artifacts, as nicely as neuronal signals.( [2], [3],\\n[4])The non-neuronal aspects of BOLD signals complicate\\nthe interpretation of fMRI signals. Secondly, the remarkable\\nfMRI facts largely rely upon the image acquisition parameters.\\nDepending on the acquisition parameters, different MRI data\\nmight have a distinctive variety of intensity values, matrix\\nsizes, and orientations. So preprocessing steps for fMRI data\\nare required to care for these problems.\\nIn\\nprevious\\nstudies,\\nresearchers\\nhave\\ndeveloped\\nfreely\\naccessible\\nopen-source\\nneuroimaging\\ndata\\npreprocessing\\ntools,\\nsuch\\nas\\nstatistical\\nparametric\\nmapping\\n(SPM)1,\\nfMRIPrep [5],Workbench [6] ,FreeSurfer [7],FMRIB software\\nlibrary (FSL) [8], and analysis of functional neuroimages\\n(AFNI) [9] . These are broadly used software program\\ntools, but everyone employs a specific data preprocessing\\nstrategy. FreeSurfer is appropriate for surface data and gives\\na fully automated command line-based pipeline.SPM and'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='tools, but everyone employs a specific data preprocessing\\nstrategy. FreeSurfer is appropriate for surface data and gives\\na fully automated command line-based pipeline.SPM and\\nIdentify applicable funding agency here. If none, delete this.\\n1http://www.fil.ion.ucl.ac.uk/spm/\\nFSL grant fully intuitive graphical user interface (GUI)-based\\npreprocessing pipelines and are suitable for volume data.\\nAFNI and Workbench system each volume and surface data.\\nThere are many strides in a given preprocessing pipeline,\\nincluding\\ndiscipline\\ninhomogeneity\\ncorrection,\\nmotion\\ncorrection, registration, and segmentation steps. This pre-\\nprocessing of FMRI images is intended to be used in several\\nworks, such as the detection of Autism Spectrum Disorders.\\n[10], [11], [12]\\nHere, we are interested in the different stages of FMRI\\ndata preprocessing in SPM. This preprocessing is applied to\\nthe ABIDE II database and has not yet been done on this\\ndatabase. And we evaluate our processing using a general'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='data preprocessing in SPM. This preprocessing is applied to\\nthe ABIDE II database and has not yet been done on this\\ndatabase. And we evaluate our processing using a general\\nlinear model (GLM) [13], which is a tool for statistical\\nanalysis of fMRI data, which allows us to evaluate the\\nactivated brain areas by defining and estimating contrasts that\\nenable us to determine which are the significantly activated\\nregions detected by the model.\\nII. METHODS\\nIn this paper, we will apply the FMRI data preprocessing\\non SPM(Statistical Parametric Mapping).\\nStatistical Parametric Mapping is a statistical technique to\\ndecide the difference in recorded brain activity by construction\\nof spatially prolonged statistical processes used to check\\nhypotheses during the practical neuroimaging experiments\\nand the usage of neuroimaging applied sciences [14]. These\\nthoughts have been blanketed in software that is known as\\nSPM. Brain imaging information sequences are applied to'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='and the usage of neuroimaging applied sciences [14]. These\\nthoughts have been blanketed in software that is known as\\nSPM. Brain imaging information sequences are applied to\\nthe SPM software program for analysis. The sequences can\\nbe a series of images from different subjects or time series\\nfrom identical subjects. SPM is made freely on hand to\\nthe neuroimaging community throughout laboratories. The\\nsoftware is an entire evaluation package deal that represents\\nthe implementation of the theoretical concept of Statistical\\nParametric Mapping.\\nThe SPM software is a suite of MATLAB (MathWorks,\\nInc) functions and subroutines with some externally compiled\\nC routines. The current version of SPM is SPM12.\\nFMRI data processing in SPM includes two types of\\nprocessing; Temporal Processing, which includes slice timing\\noperation, and also has Spatial Processing, which provides\\nfor Realign, Co-registration, Segmentation, Normalization,\\n978-1-6654-5276-2/22/$31.00 © 2022 IEEE'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 0}, page_content='operation, and also has Spatial Processing, which provides\\nfor Realign, Co-registration, Segmentation, Normalization,\\n978-1-6654-5276-2/22/$31.00 © 2022 IEEE\\n2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM) | 978-1-6654-5276-2/22/$31.00 ©2022 IEEE | DOI: 10.1109/WINCOM55661.2022.9966463\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='and Smoothing.\\nThe ABIDE database consists of the first (ABIDE I) and\\nsecond (ABIDE II) data releases provided by the University of\\nUtah. Seventeen sites provided 1112 individuals in the ABIDE\\nI version, including 539 autism patients and 573 controls.\\nThe ABIDE II data was compiled from 19 sites, a total\\ndataset of 1114 individuals, including 521 individuals with\\nautism and 593 controls. Site-specific details on diagnostic\\ncriteria and more detailed characterization For clinical and\\nbehavioral phenotypes, see [15]. The current study is based\\non the ABIDE II database only.\\nA. Slice Timing Correction\\nA total volume is acquired in a TR ranging, in general,\\nfrom a few milliseconds to several seconds. A time lag\\nbetween slices is, therefore, present, depending on the\\nposition of the slice within the volume and the number\\nof cuts. As a result, this problem can cause a time lag in\\nthe measured hemodynamic response between the different'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='position of the slice within the volume and the number\\nof cuts. As a result, this problem can cause a time lag in\\nthe measured hemodynamic response between the different\\nslices, compromising the reliability of the statistical results in\\ndetecting activations.\\nThe impact of this time lag is all the more important for a\\nlong TR and non-interlaced acquisitions. To compensate for\\nthis time lag, one solution consists of a time interpolation to\\nbring all the slices back to a common time, which is the time\\nof acquisition of one of the slices of the volume (reference\\nslice). The reference cut is generally the central cut in the\\nvolume to reduce the interpolation to a TR/2. Subsequently,\\nwe could consider that all the slices of a volume were\\nacquired at the same time [16].\\nAnother time lag model was proposed to avoid modifying\\nthe data before statistical analysis. This involves including\\na regressor in the linear model GLM, whose principle is'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='Another time lag model was proposed to avoid modifying\\nthe data before statistical analysis. This involves including\\na regressor in the linear model GLM, whose principle is\\nbased on the time derivatives of the HRF [17]. Generally,\\nthis method is used for a long TR (greater than 2 seconds) to\\neliminate any artefactual variance components.\\nB. Realignment\\nThe subject’s movement during the experiment is one of\\nthe significant artifacts and inevitable, given an acquisition\\ntime of several minutes. A rigid readjustment allows partial\\ncorrection of motion artifacts. It consists first of an estimate\\nof the six translation parameters (”right,” ”forward,” ”up,”\\nin mm) and rotation (”pitch,” ”roll,” ”yaw,” in radians) of\\nthe volumes acquired for the subject relative to a reference\\nvolume (usually the first volume). Then, an interpolation\\nconcerning the estimate makes it possible to readjust the\\nvolumes on the reference volume.\\nNote that there is an interdependence between the motion'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='concerning the estimate makes it possible to readjust the\\nvolumes on the reference volume.\\nNote that there is an interdependence between the motion\\ncorrection and slice-timing steps, and the order in which\\nthese two pre-processing operations are performed impacts\\nthe sensitivity of the results. Usually, slice timing is applied\\nbefore motion correction when slices are acquired in an\\ninterleaved way so as not to resample voxels from one\\nslice to an adjacent slice acquired at a distance in time.\\nConversely, if the acquisition was performed in sequential\\nmode, the motion correction is applied first, not temporarily,\\nto interpolate signals from distant brain regions.\\nC. Coregistration\\nIn fMRI, this step is usually performed to align the func-\\ntional images with each other on the anatomical image to\\nallow the final individual activations to be represented on a\\nhighly resolved image. The anatomy is often readjusted to the\\naverage functional volumes when one wishes to remain in the'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='allow the final individual activations to be represented on a\\nhighly resolved image. The anatomy is often readjusted to the\\naverage functional volumes when one wishes to remain in the\\nnative space without disturbing the functional data. As the\\nmodalities are different, the similarity measurement method is\\nusually based on mutual information [18], as anatomical and\\nfunctional images are often weighted differently.\\nD. Segmentation\\nThe segmentation step creates, from a single image, a white\\nmatter image, a grey matter image, and a cerebrospinal fluid\\n(CSF) image. These images can display a 3D rendering of the\\nactivations overlaid on the anatomy. The segmentation in SPM\\nis based on a modified mixed Gaussian model. The Bayesian\\nrule assigns the probability for each voxel belonging to a tissue\\nclass by combining the probability of belonging to that tissue\\nclass and the prior probability derived from probability maps\\nfrom many subjects.\\nE. Normalization'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='class by combining the probability of belonging to that tissue\\nclass and the prior probability derived from probability maps\\nfrom many subjects.\\nE. Normalization\\nThis operation is crucial for group analysis. Indeed, two\\ninter-individual variability sources must be considered when\\nconducting a group study: anatomical variability (shape, size\\nof the brain) and functional variability (localization of regions\\nactivated during stimulation). Spatial normalization allows all\\nimages to be placed in the same reference space. (reference\\nor template) to compare the functional fMRI results obtained\\nfor different subjects.\\nSeveral stereotactic references are used in neuroimaging\\nstudies: the one based on the Talairach and Tournoux atlas\\n[19], and that of the Montreal Neurological Institute, of\\nwhich several versions have been developed [20] and the\\nAnatomical Automatic Labeling atlas (AAL) [21].\\nThese two first atlases differ radically in their conception, the'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 1}, page_content='which several versions have been developed [20] and the\\nAnatomical Automatic Labeling atlas (AAL) [21].\\nThese two first atlases differ radically in their conception, the\\nfirst being based on the post-mortem study of the brain of a\\nsingle individual, the second being constructed by averaging\\nT1-weighted MRI images of dozens of brains (305 in the first\\nversion [22], then 152 in the currently most used version).\\nOnce spatial normalization has been achieved, the activations\\ncan be localized in the common space and compared between\\ndifferent subjects. It is then assumed that voxel to voxel\\ncorrespondence exists for all subjects, which allows the\\ncalculation of group of group statistics.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 2}, page_content='F. Smoothing\\nAn inherent problem with fMRI acquisition is that adjacent\\nvoxels do not have independent signals from each other. As\\na result, the signals have spatial correlations that are difficult\\nto estimate. The functional images undergo spatial smoothing\\nto impose known and controlled correlation characteristics.\\nThis smoothing consists of convolution with an isotropic\\nGaussian filter with a Full Width at Halh Maximum (FWHM),\\ngenerally fixed at 2 or 4 times a voxel size. However, this\\nchoice of Gaussian kernel must be adapted to the study being\\nconducted [23].\\nThis\\nstep\\nmakes\\nit\\npossible\\nto\\nminimize\\nthe\\nresidual\\nrealignment and normalization faults and to increase the\\nsignal-to-noise ratio. On the other hand, the accuracy of the\\nlocalization of activations may be reduced, especially by\\nextension.\\nG. General linear model (GLM)\\nThe GLM is a fundamental tool for the statistical analysis\\nof functional data. The linear approach has appeared in'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 2}, page_content='extension.\\nG. General linear model (GLM)\\nThe GLM is a fundamental tool for the statistical analysis\\nof functional data. The linear approach has appeared in\\nthe literature through the parametric statistical map method\\n[24], [25], [26]. This method has become a reference in\\nthe univariate analysis of functional data. It consists in\\nmodeling the observed time series for each voxel by a linear\\ncombination of explanatory functions with a residual error\\nterm.\\nMore precisely, this model is a multiple linear regression in\\nwhich the BOLD signal yj,t is acquired at voxel j at time t.\\nIt is defined through three types of signals:\\n•The useful signal, which is assumed to be the sum of the\\neffects induced by the different conditions of the experimental\\nparadigm M;\\n•Low-frequency drifts due to physical (thermal noise, etc.),\\nphysiological (folding of respiratory and cardiac frequencies),\\nand effects induced by the interaction between movements\\nand the static magnetic field;'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 2}, page_content='physiological (folding of respiratory and cardiac frequencies),\\nand effects induced by the interaction between movements\\nand the static magnetic field;\\n•The noise expressing the modeling error, which is generally\\nassumed to be Gaussian and dependent on voxel j.\\nyj,t = Xβj + ej =\\nR\\nX\\nr=1\\nβr\\nj + ej,t\\n(1)\\nwhere yj,t is a vector of dimension N (the number of\\nscans) representing the preprocessed BOLD signal at voxel j\\nmeasured at time t,with j = n∗TR, n ∈[1, N].\\nThe wanted signal, as well as the low-frequency drift, is\\nrepresented by X∗βj,where X is a matrix of dimension N x\\nR called the drawing matrix, and βj are the linear regression\\ncoefficients (vector of dimension R). Finally ej represents the\\nresidual error.\\nThe estimation of the parameters βj is done in the classical\\nway by least squares adjustment. This minimizes the squared\\nerror between the signal yj,t measured at time t and the\\nestimated one X∗bβj. The estimation of the parameters βj is'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 2}, page_content='way by least squares adjustment. This minimizes the squared\\nerror between the signal yj,t measured at time t and the\\nestimated one X∗bβj. The estimation of the parameters βj is\\ntherefore performed independently for each voxel of the brain\\n(massively univariate estimation). Let bβj be the value of βj\\nadjusted by the least squares method. The expression for bβj\\nis then written :\\nbβj = (XT X)\\n−1XT yj,t\\n(2)\\nThe estimated parameter bβj thus follows a normal distribution,\\ncentered on βj and with matrix of covariance Var[bβj]=\\nσ2\\nj (XT X)\\n−1. The estimator of the noise variance σ2\\nj at voxel\\nj is estimated by :\\nbσ2\\nj = (yj −X bβj)T (yj −X bβj)\\nN −R\\n(3)\\nAfter estimating the linear regression coefficients, statistical\\ntests should be chosen according to the nature and distribution\\nof the variables studied. A test is used to test whether a con-\\ntrast (i.e., a linear combination) of parameters is significantly\\ndifferent from a valued, under the following two assumptions:\\nH0 : cT βj = d\\n(4)'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 2}, page_content='trast (i.e., a linear combination) of parameters is significantly\\ndifferent from a valued, under the following two assumptions:\\nH0 : cT βj = d\\n(4)\\nH0 : cT βj ̸= d\\n(5)\\nWhere cT = [cr; r = 1 : R] is a row vector determining\\nthe coefficients of the desired linear combination with\\ncr{−1, 0, 1}. It is therefore important to know the distribution\\nof βj. As we have previously predicted that the estimator of\\neach parameter βj follows a normal distribution, the linear\\ncombination of several parameters cT βj also follows a normal\\ndistribution:cT bβjN(cT βj, cT σ2\\nj (XT X)\\n−1c\\nNote also that\\n(N−R)σ2\\nj\\nσ2\\nj\\nis thus expressed as the sum of\\nthe squares of N random variables following a normal\\ndistribution, hence :\\nbσ2\\nj\\ncT\\nx2\\nN−R\\n(N −R)\\n(6)\\n(Chi-square law with N - R degrees of freedom). As a\\nresult, we can use the following statistical test which follows\\nStudent’s law with N - R degrees of freedom:\\nTj =\\ncT bβj −cT βj\\nq\\ncT σ2\\nj (XT X)−1c\\n(7)\\nIII. RESULTS'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 2}, page_content='result, we can use the following statistical test which follows\\nStudent’s law with N - R degrees of freedom:\\nTj =\\ncT bβj −cT βj\\nq\\ncT σ2\\nj (XT X)−1c\\n(7)\\nIII. RESULTS\\nPreprocessing is applied on the the ABIDE II database\\non the OHSU site, on 276 functional images, and the pre-\\nprocessing steps will be in the following order: Slice tim-\\ning correction, motion correction (realignment), anatomical-\\nfunctional realignment (Coregistration), segmentation, spatial\\nnormalization (Normalize), and smoothing.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 3}, page_content='FIGURE 1. SPM Preprocessing Pipeline of fMRI data.\\nTable I shows the preprocessing steps with input and output\\ndata for each level, assuming that the name of the fMRI data\\ninput is rest.\\nmethod Name\\nInput Data\\nOutput Data\\nSlice Timing\\nrest.nii\\narest.nii\\nRealignment\\narest.nii\\narest.nii\\nmeanarest.nii\\nrarest.nii\\nrp rest.txt\\nCoregistration\\nReference\\nimage\\n(meanarest.nii)\\nSource\\nimage\\nis\\nanatomical\\nimage\\n(anat.nii)\\nModified\\nanatomical\\nimage (ranat.nii)\\nSegmentation\\nModified anatomical im-\\nage (ranat.nii)\\nc1*.nii:image of the\\ngrey matter\\nc2*.nii:image of the\\nwhite matter\\nc3*.nii\\n:image\\nof\\ncerebrospinal fluid\\nc4*.nii\\n:image\\nof\\nbone tissue\\nc5*.nii :image of soft\\ntissue image\\nranat* seg8.mat\\n:Transformation\\nmatrix for MNI space\\ny ranat*.nii\\n:Field\\ndeformation image:\\nNormalization\\nDeformation\\nimage(y ranat*.nii)\\nimage to write(arrest*.nii)\\nwarest.nii normalized\\nSmoothing\\nwarest.nii normalized\\nswarest.nii smoothed\\nimages\\nTABLE I\\nINPUT AND OUTPUT DATA FOR PROCESSING.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 3}, page_content='Deformation\\nimage(y ranat*.nii)\\nimage to write(arrest*.nii)\\nwarest.nii normalized\\nSmoothing\\nwarest.nii normalized\\nswarest.nii smoothed\\nimages\\nTABLE I\\nINPUT AND OUTPUT DATA FOR PROCESSING.\\nTo evaluate the preprocessing a statistical analysis is\\napplied to the basis of the GLM. p the threshold value equal\\n0.01.\\nset-level\\ncluster-level\\nmm mm mm\\np\\nc\\npF W E−cor\\nqF DR−cor\\nkE\\npuncor\\n0.715\\n7\\n0.741\\n0.996\\n0.995\\n0.997\\n0.999\\n0.999\\n1.000\\n0.970\\n0.970\\n0.970\\n0.970\\n0.970\\n0.970\\n0.970\\n457\\n46\\n54\\n35\\n17\\n8\\n1\\n0.164\\n0.676\\n0.647\\n0.722\\n0.818\\n0.885\\n0.970\\n0\\n-2\\n12\\n-2\\n-4\\n-36\\n-44\\n8\\n-2\\n30\\n46\\n52\\n-2\\n46\\n12\\n-2\\n50\\n-18\\n24\\n12\\n16\\n10\\n32\\n14\\n32\\n30\\n14\\nTABLE II\\nSTATISTICAL REPORT 1.\\npeak-level\\nmm mm mm\\npF W E−corr\\nqF DR−corr\\nT\\n(Ze)\\npuncorr\\n0.449\\n0.633\\n0.993\\n0.838\\n0.995\\n0.997\\n0.999\\n0.999\\n1.000\\n0.731\\n0.731\\n0.910\\n0.885\\n0.910\\n0.910\\n0.910\\n0.910\\n0.977\\n3.53\\n3.34\\n2.64\\n3.10\\n2.60\\n2.56\\n2.48\\n2.45\\n2.36\\n3.48\\n3.30\\n2.62\\n3.07\\n2.58\\n2.54\\n2.46\\n2.44\\n2.34\\n0.000\\n0.000\\n0.004\\n0.001\\n0.005\\n0.006\\n0.007\\n0.007\\n0.010\\n0\\n-2\\n12\\n-2\\n-4\\n-36\\n-44\\n8\\n-2\\n30\\n46\\n52\\n-2\\n46'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 3}, page_content='0.910\\n0.977\\n3.53\\n3.34\\n2.64\\n3.10\\n2.60\\n2.56\\n2.48\\n2.45\\n2.36\\n3.48\\n3.30\\n2.62\\n3.07\\n2.58\\n2.54\\n2.46\\n2.44\\n2.34\\n0.000\\n0.000\\n0.004\\n0.001\\n0.005\\n0.006\\n0.007\\n0.007\\n0.010\\n0\\n-2\\n12\\n-2\\n-4\\n-36\\n-44\\n8\\n-2\\n30\\n46\\n52\\n-2\\n46\\n12\\n-2\\n50\\n-18\\n24\\n12\\n16\\n10\\n32\\n14\\n32\\n30\\n14\\nTABLE III\\nSTATISTICAL REPORT 2.\\nThe statistical results give the coordinates of the activated\\nareas in the MNI space. The information on each activated\\naggregate is the voxels of the statistical values exceeding the\\nthreshold specified during contrast estimation. Set-level: c is\\nthe number of activated aggregates above the threshold. The\\np-value is the probability of obtaining c or more clusters in\\nthe volume of interest. Cluster-level: this item gives statistical\\ninference results at the level of a cluster. kE gives the number\\nof voxels of each cluster or the selected cluster on the glass\\nbrain. In case the test performed is a T-test. p corrected is\\nthe probability of obtaining kE voxels or more in the volume'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 3}, page_content='of voxels of each cluster or the selected cluster on the glass\\nbrain. In case the test performed is a T-test. p corrected is\\nthe probability of obtaining kE voxels or more in the volume\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='of interest. p uncorrected is the probability of getting an\\naggregate of kE voxels or more.\\nIV. CONCLUSION\\nThere has been an explosive interest in the use of fMRI\\nin recent years. Working with FMRI data is complex, and\\npre-processing is essential before any analysis. The choice of\\npre-processing requires an application of several pipelines to\\nevaluate each part to arrive at the most efficient processing\\nin the end. In the course of this article, we have worked\\non FMRI data processing in SPM includes two types of\\nprocessing; Temporal Processing, which includes slice timing\\noperation, and also has Spatial Processing, which provides\\nfor Realign, Co-registration, Segmentation, Normalization,\\nand Smoothing. and we evaluate this processing by applying\\nan MLG.\\nREFERENCES\\n[1] Farah M., J. Emerging ethical issues in neuroscience.(2002). Nature\\nNeuroscience, 5, 1123-1129.\\n[2] Murphy, K., Birn, R. M., and Bandettini, P. A. (2013). Resting-'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='an MLG.\\nREFERENCES\\n[1] Farah M., J. Emerging ethical issues in neuroscience.(2002). Nature\\nNeuroscience, 5, 1123-1129.\\n[2] Murphy, K., Birn, R. M., and Bandettini, P. A. (2013). Resting-\\nstate fMRI confounds and cleanup. Neuroimage 80, 349–359. doi:\\n10.1016/j.neuroimage.2013.04.001.\\n[3] Bright,\\nM.\\nG.,\\nand\\nMurphy,\\nK.\\n(2015).\\nIs\\nfMRI\\n“noise”\\nreally\\nnoise?\\nResting\\nstate\\nnuisance\\nregressors\\nremove\\nvariance\\nwith\\nnetwork\\nstructure.\\nNeuroimage\\n114,\\n158–169.\\ndoi:\\n10.1016/j.neuroimage.2015.03.070.\\n[4] Caballero-Gaudes, C., and Reynolds, R. C. (2017). Methods for\\ncleaning the BOLD fMRI signal. Neuroimage 154, 128–149. doi:\\n10.1016/j.neuroimage.2016.12.018.\\n[5] Esteban, O., Markiewicz, C. J., Blair, R. W., Moodie, C. A., Isik, A. I.,\\nErramuzpe, A., et al. (2019). fMRIPrep: a robust preprocessing pipeline\\nfor functional MRI. Nat. Methods 16, 111–116. doi: 10.1038/s41592-\\n018-0235-4.\\n[6] Marcus, D. S., Harms, M. P., Snyder, A. Z., Jenkinson, M., Wilson, J. A.,'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='for functional MRI. Nat. Methods 16, 111–116. doi: 10.1038/s41592-\\n018-0235-4.\\n[6] Marcus, D. S., Harms, M. P., Snyder, A. Z., Jenkinson, M., Wilson, J. A.,\\nGlasser, M. F., et al. (2013). Human Connectome Project informatics:\\nquality control, database services, and data visualization. Neuroimage\\n80, 202–219. doi: 10.1016/j.neuroimage.2013.05.077.\\n[7] Fischl,\\nB.\\n(2012).\\nFreeSurfer.\\nNeuroimage\\n62,\\n774–781.\\ndoi:\\n10.1016/j.neuroimage.2012.01.021.\\n[8] Jenkinson, M., Beckmann, C. F., Behrens, T. E. J., Woolrich, M.\\nW., and Smith, S. M. (2012). Fsl. Neuroimage 62, 782–790. doi:\\n10.1016/j.neuroimage.2011.09.015.\\n[9] Cox, R. W. (1996). AFNI: software for analysis and visualization of\\nfunctional magnetic resonance neuroimages. Comput. Biomed. Res. 29,\\n162–173. doi: 10.1006/cbmr.1996.0014.\\n[10] FZ Benabdallah, A Drissi El Maliani, D Lotfi, R Jennane, M El\\nHassouni.An autism spectrum disorder adaptive identification based on\\nthe Elimination of brain connections: a proof of long-range undercon-'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='Hassouni.An autism spectrum disorder adaptive identification based on\\nthe Elimination of brain connections: a proof of long-range undercon-\\nnectivity,Soft Computing 26 (10), 4701-4711,26,2022.\\n[11] FZ Benabdallah, E Maliani, A Drissi, M El Hassouni:A CNN Based\\n3D Connectivity Matrices Features for Autism Detection: Application\\non ABIDE I,International Symposium on Ubiquitous Networking, 293-\\n302,2021.\\n[12] FZ Benabdallah, AD El Maliani, D Lotfi, M El Hassouni:Analysis of the\\nover-connectivity in autistic brains using the maximum spanning tree:\\napplication on the multi-site and heterogeneous ABIDE dataset,2020 8th\\nInternational Conference on Wireless Networks and Mobile,2020.\\n[13] K. J. Friston, A. P. Holmes, K. J. Worsley, J.-P. Poline, C. D. Frith, and\\nR. S. Frackowiak. Statistical parametric maps in functional imaging : a\\ngeneral linear approach. Human brain mapping, 2(4) :189–210, 1994a.\\n[14] Penny W. D., Friston K. J., Ashburner J. T., Kiebel S. J., Nichols T.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='general linear approach. Human brain mapping, 2(4) :189–210, 1994a.\\n[14] Penny W. D., Friston K. J., Ashburner J. T., Kiebel S. J., Nichols T.\\nE.(2006). Statistical Mapping: the analysis of functional brain imaging\\n(1st edition). Academic Press.\\n[15] Autism Brain Imaging Data Exchange, [http://fcon 1000.projects.nitrc.\\norg/indi/abide/].\\n[16] V. Calhoun, X. Golay, and G. Pearlson. Improved fmri slice timing\\ncorrection : interpolation errors and wrap around e!ects. In Proceedings,\\nISMRM, 9th annual meeting, Denver, page 810, 2000.\\n[17] R. Sladky, K. J. Friston, J. Tr¨ostl, R. Cunnington, E. Moser, and C.\\nWindischberger. Slice-timing effects and their correction in functional\\nmri. Neuroimage, 58(2) :588–594, 2011.\\n[18] F. Maes, A. Collignon, D. Vandermeulen, G. Marchal, and P. Suetens.\\nMultimodality image registration by maximization of mutual informa-\\ntion. IEEE transactions on medical imaging, 16(2) :187–198, 1997.'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='Multimodality image registration by maximization of mutual informa-\\ntion. IEEE transactions on medical imaging, 16(2) :187–198, 1997.\\n[19] J. Talairach and P. Tournoux. Co-planar stereotaxic atlas of the human\\nbrain. 3- dimensional proportional system : an approach to cerebral\\nimaging. 1988.\\n[20] A. Evans. 3-d multi-modality human brain mapping : Past, present and\\nfuture. ANNALS OF NUCLEAR MEDICINE, 7 :S13–S13, 1993.\\n[21] N. Tzourio-Mazoyer, B. Landeau, D. Papathanassiou, F. Crivello, O.\\nEtard, N. Delcroix, B. Mazoyer, and M. Joliot. Automated anatomical\\nlabeling of activations in spm using a macroscopic anatomical parcella-\\ntion of the mni mri single-subject brain. Neuroimage, 15(1) :273–289,\\n2002.\\n[22] A. Evans. 3-d multi-modality human brain mapping : Past, present and\\nfuture. ANNALS OF NUCLEAR MEDICINE, 7 :S13–S13, 1993.\\n[23] K. Friston, O. Josephs, E. Zarahn, A. Holmes, S. Rouquette, and J.-\\nB. Poline. To smooth or not to smooth ? : Bias and effciency in fmri'),\n",
       " Document(metadata={'producer': 'pdfTeX-1.40.23; modified using iText® 7.1.1 ©2000-2018 iText Group NV (AGPL-version)', 'creator': 'Certified by IEEE PDFeXpress at October 30, 2022 23:59:06', 'creationdate': '2022-10-30T14:27:37+00:00', 'source': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'file_path': '..\\\\data\\\\literature\\\\Preprocessing_FMRI_Data_In_SPM12 (1).pdf', 'total_pages': 5, 'format': 'PDF 1.5', 'title': 'Preprocessing FMRI Data In SPM12', 'author': '', 'subject': '2022 9th International Conference on Wireless Networks and Mobile Communications (WINCOM);2022; ; ;10.1109/WINCOM55661.2022.9966463', 'keywords': '', 'moddate': '2022-12-05T19:28:58-05:00', 'trapped': 'False', 'modDate': \"D:20221205192858-05'00'\", 'creationDate': 'D:20221030142737Z', 'page': 4}, page_content='future. ANNALS OF NUCLEAR MEDICINE, 7 :S13–S13, 1993.\\n[23] K. Friston, O. Josephs, E. Zarahn, A. Holmes, S. Rouquette, and J.-\\nB. Poline. To smooth or not to smooth ? : Bias and effciency in fmri\\ntime-series analysis. NeuroImage, 12 (2) :196–208, 2000.\\n[24] G. M. Boynton, S. A. Engel, G. H. Glover, and D. J. Heeger. Linear\\nsystems analysis of functional magnetic resonance imaging in human\\nv1. Journal of Neuroscience, 16(13) : 4207–4221, 1996.\\n[25] K. J. Friston, A. P. Holmes, K. J. Worsley, J.-P. Poline, C. D. Frith, and\\nR. S. Frackowiak. Statistical parametric maps in functional imaging : a\\ngeneral linear approach. Human brain mapping, 2(4) :189–210, 1994a.\\n[26] R. Goebel. fmri : From nuclear spins to brain functions. In Springer US,\\n2015.\\nAuthorized licensed use limited to: INDIAN INSTITUTE OF TECHNOLOGY KHARAGPUR. Downloaded on September 16,2025 at 06:30:16 UTC from IEEE Xplore.  Restrictions apply.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='Slice-timing effects and their correction in functional MRI\\nRonald Sladky a,b, Karl J. Friston c, Jasmin Tröstl a,b, Ross Cunnington a,d,\\nEwald Moser a,b, Christian Windischberger a,b,⁎\\na MR Centre of Excellence, Medical University of Vienna, Lazarettgasse 14, 1090 Vienna, Austria\\nb Center for Medical Physics and Biomedical Engineering, Medical University of Vienna, Währinger Gürtel 18-20, 1090 Vienna, Austria\\nc Wellcome Trust Centre for Neuroimaging, Institute of Neurology, University College London, 12 Queen Square, London WC1N 3BG, UK\\nd School of Psychology & Queensland Brain Institute, University of Queensland, Brisbane, Australia\\na b s t r a c t\\na r t i c l e\\ni n f o\\nArticle history:\\nReceived 7 March 2011\\nRevised 23 June 2011\\nAccepted 24 June 2011\\nAvailable online 2 July 2011\\nKeywords:\\nFunctional MRI\\nPre-processing\\nAnalysis\\nSlice-timing correction\\nExact timing is essential for functional MRI data analysis. Datasets are commonly measured using repeated 2D'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='Keywords:\\nFunctional MRI\\nPre-processing\\nAnalysis\\nSlice-timing correction\\nExact timing is essential for functional MRI data analysis. Datasets are commonly measured using repeated 2D\\nimaging methods, resulting in a temporal offset between slices. To compensate for this timing difference,\\nslice-timing correction (i.e. temporal data interpolation) has been used as an fMRI pre-processing step for\\nmore than ﬁfteen years. However, there has been an ongoing debate about the effectiveness and applicability\\nof this method. This paper presents the ﬁrst elaborated analysis of the impact of the slice-timing effect on\\nsimulated data for different fMRI paradigms and measurement parameters, taking into account data noise and\\nsmoothing effects. Here we show, depending on repetition time and paradigm design, slice-timing effects can\\nsigniﬁcantly impair fMRI results and slice-timing correction methods can successfully compensate for these'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='signiﬁcantly impair fMRI results and slice-timing correction methods can successfully compensate for these\\neffects and therefore increase the robustness of the data analysis. In addition, our results from simulated data\\nwere supported by empirical in vivo datasets. Our ﬁndings suggest that slice-timing correction should be\\nincluded in the fMRI pre-processing pipeline.\\n©\\nIntroduction\\nFunctional magnetic resonance imaging (fMRI) data sets are\\ngenerally acquired using sequential 2D imaging techniques like\\nsingle-shot echo planar imaging sequences (Stehling et al., 1991;\\nTurner et al., 1998). Since fMRI data analysis is essentially a time\\ncourse analysis, exact timing with respect to the stimulus presenta-\\ntion paradigm is crucial.\\nSingle-shot EPI sequences allow for single-slice acquisition times\\nin the range of 50–150 ms. Whole-brain coverage is achieved by\\nsequentially repeated image acquisition for a stack of individual slices.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='in the range of 50–150 ms. Whole-brain coverage is achieved by\\nsequentially repeated image acquisition for a stack of individual slices.\\nDepending on brain coverage and slice thickness, a whole volume can\\nbe acquired within typical repetition times (TRs) ranging from\\nhundreds of milliseconds to several seconds. This causes slice\\nacquisition delays between individual slices which may add up to\\nsigniﬁcant temporal shifts over the full 3D volume between the\\nexpected and actually measured hemodynamic response (Fig. 1). As a\\nconsequence, the reliability and power of time series analysis may be\\ncompromised, resulting in degraded sensitivity to detect activations.\\nTo compensate for slice acquisition delays slice-timing correction\\n(STC) has been proposed as a pre-processing step (Calhoun et al.,\\n2000; Henson et al., 1999) and is currently included in all major fMRI\\nsoftware packages (such as SPM, AFNI, BrainVoyager or FSL). In STC,'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='2000; Henson et al., 1999) and is currently included in all major fMRI\\nsoftware packages (such as SPM, AFNI, BrainVoyager or FSL). In STC,\\nthe individual slice is temporally realigned to a reference slice based\\non its relative timing using an appropriate resampling method.\\nDifferent data interpolation methods have been proposed for STC\\nincluding linear, sinc and cubic spline interpolation (Goebel, 2010).\\nA more elegant way would be to model known nuisance factors\\nduring analysis rather than changing the data. Accordingly, it has also\\nbeen suggested to account for slice-timing differences via adjust-\\nments in the analysis procedure, in particular the model setup, by\\nusing (a) additional regressors based on the temporal derivatives of\\nthe expected hemodynamic response function in the general linear\\nmodel (GLM), or (b) slice-dependent regressors time-shifted accord-\\ning to their acquisition lag (Henson et al., 1999).\\nAdding temporal derivatives might be advisable to compensate for'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='model (GLM), or (b) slice-dependent regressors time-shifted accord-\\ning to their acquisition lag (Henson et al., 1999).\\nAdding temporal derivatives might be advisable to compensate for\\nnon-linear neural and vascular effects resulting in time-shifted and\\ntime-dispersed BOLD responses in different regions, a crucial issue\\nparticularly for event-related fMRI (Calhoun et al., 2004; Friston et al.,\\n1998; Worsley and Taylor, 2006). However, it has not been shown yet,\\nwhether including temporal derivatives into the GLM can fully\\ncompensate for slice acquisition delays.\\nThe second strategy to counter slice-timing effects is based on\\nconstructing slice-speciﬁc time-shifted regressors (Henson et al.,\\n1999). This strategy, however, is problematic with spatially smoothed\\nNeuroImage 58 (2011) 588–594\\n⁎ Corresponding author at: MR Centre of Excellence, Medical University of Vienna,\\nLazarettgasse 14, 1090 Vienna, Austria. Fax: +43 1 40400 7631.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 0}, page_content='NeuroImage 58 (2011) 588–594\\n⁎ Corresponding author at: MR Centre of Excellence, Medical University of Vienna,\\nLazarettgasse 14, 1090 Vienna, Austria. Fax: +43 1 40400 7631.\\nE-mail addresses: ronald.sladky@meduniwien.ac.at (R. Sladky),\\nchristian.windischberger@meduniwien.ac.at (C. Windischberger).\\n1053-8119 © 2011 Elsevier Inc.\\ndoi:10.1016/j.neuroimage.2011.06.078\\nContents lists available at ScienceDirect\\nNeuroImage\\njournal homepage: www.elsevier.com/locate/ynimg\\n2011 Elsevier Inc. Open access under CC BY-NC-ND license.\\nOpen access under CC BY-NC-ND license.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='data sets. In many fMRI processing pipelines spatial smoothing is\\nconsidered a mandatory pre-processing step; it not only allows for\\nrandom-ﬁeld approximation based corrections for multiple compar-\\nisons, but also for group analyses in standard space (and also to\\nincrease image signal-to-noise ratios) (Hopﬁnger et al., 2000). Usually\\n6−10 mm3 isotropic Gaussian kernels are used for smoothing,\\nleading to considerable signal dispersion across adjacent slices.\\nWhereas this is not a serious concern for sequentially acquired\\nvolumes, it may cause signiﬁcant estimator bias for interleaved\\nacquisition, because neighboring slices are typically about a half TR\\napart in time.\\nSo far, no extensive study has been published on the effectiveness\\nand robustness of STC and potential unwanted effects on data quality.\\nHenson et al. (1999) showed a better detection rate of neural activity\\nfor slice-timing corrected data for a particular event-related design.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='Henson et al. (1999) showed a better detection rate of neural activity\\nfor slice-timing corrected data for a particular event-related design.\\nSince then, the applicability of STC, and whether to perform it or not,\\nhas been discussed vigorously (if informally) within the neuroimag-\\ning community. The main contention has been whether slice-timing\\ncorrection removes artefactual variance components when there is a\\nlong interval between scans (e.g. TRN2 s). This is the situation in\\nwhich STC would be most effective but it is also the situation, which\\npresents the greatest problems for STC interpolation schemes. This\\npaper is an attempt to treat slice-timing effects formally and assess\\nwhether their correction can improve fMRI data analyses.\\nHere we assess the inﬂuence of slice acquisition delays (slice-\\ntiming effects) using simulated data sets varying acquisition param-\\neters, paradigm designs and noise levels. Herein we compare the'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='Here we assess the inﬂuence of slice acquisition delays (slice-\\ntiming effects) using simulated data sets varying acquisition param-\\neters, paradigm designs and noise levels. Herein we compare the\\nresults from GLM analyses of uncorrected and slice-timing corrected\\ndata (as implemented in SPM8) in order to quantify the efﬁcacy of STC\\nmethods. To further assess the interaction between STC and the\\ninclusion of temporal derivatives in the GLM, we compared STC to no\\nSTC, with and without temporal derivatives. This resulted in a\\ncomparison of four schemes or methods corresponding to the cells\\nof a 2×2 factorial design. To validate the conclusion from simulated\\ndata, we also show the results of a study in human subjects using a\\nparadigm, which was speciﬁcally designed to investigate slice-timing\\neffects in real fMRI studies.\\nMethods\\nSlice-timing correction approaches\\nDifferent methods of temporal interpolation can be used for slice-'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='effects in real fMRI studies.\\nMethods\\nSlice-timing correction approaches\\nDifferent methods of temporal interpolation can be used for slice-\\ntiming correction. While linear interpolation (Eq. (1)) is easily\\nimplemented and fast, it may introduce undesired temporal smooth-\\ning and more sophisticated interpolation schemes are preferred, such\\nas sinc interpolation as used by default within SPM8. Assuming an\\nacquired time series y of slice number n at time point t(n), linear\\ninterpolation to the slice-time of a reference slice t(r) can formally be\\nexpressed as:\\ny rð Þ\\nn =\\nt rð Þ−t n−1\\nð\\nÞ\\nð\\nÞyn + t n\\nð Þ−t rð Þ\\nð\\nÞyn−1\\nt n\\nð Þ−t n−1\\nð\\nÞ\\nð1Þ\\nAn efﬁcient implementation of the sinc interpolation (Eq. (2))\\nentails applying a phase shift (i.e. adding a constant value) in the\\nfrequency domain of the signal, obtained by fast Fourier transforma-\\ntion. This method is used in SPM, using an implementation to\\ncompensate for interpolation and wrap-around effects (Calhoun et al.,\\n2000).\\ny rð Þ\\nn =\\n∑\\n∞'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='tion. This method is used in SPM, using an implementation to\\ncompensate for interpolation and wrap-around effects (Calhoun et al.,\\n2000).\\ny rð Þ\\nn =\\n∑\\n∞\\ni=−∞\\nxisinc\\nπ\\nTR r−iTR\\nð\\nÞ\\n\\x01\\n\\x03\\nð2Þ\\nCubic spline interpolation is the default interpolation method in\\nBrainVoyager (Goebel, 2010).\\nThe effects of slice-timing correction were studied based on\\nextensive simulations covering a wide range of blocked and event-\\nrelated design paradigms, varying inter-stimulus intervals, contrast-\\nto-noise ratios (CNRs), slice acquisition orders and TRs. In addition,\\nfMRI data from a speciﬁcally designed group study were analyzed in\\norder to validate the ﬁndings in simulated data. Slice-timing\\ncorrections were performed using the spm_slice_timing.m function\\nas implemented in SPM8 (Wellcome Trust Centre for Neuroimaging,\\nUCL, UK) and results were analyzed using MathWorks MATLAB\\nR2009a (Natick, MA, USA).\\nSimulated data\\nA set of 4-dimensional matrices were created to simulate a series'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='UCL, UK) and results were analyzed using MathWorks MATLAB\\nR2009a (Natick, MA, USA).\\nSimulated data\\nA set of 4-dimensional matrices were created to simulate a series\\nof MRI scans with 20 slices, a resolution of 32×32 px2 and 300\\nseconds run duration. Voxel size was set to 1.5×1.5×3 mm3. The time\\ncourses of each set were based on different blocked and event-related\\ndesigns convolved with the canonical hemodynamic response\\nfunction included in SPM8.\\nThe length of on and off blocks in blocked designs was set to 10, 15,\\n20, 25, and 30 s. Event-related designs were single events with an\\naverage stimulus onset asynchrony (SOA, i.e. the time between the\\nonset of one stimulus and the consecutive stimulus) of 4, 6, 8, 10, and\\n12 s; SOAs were jittered according to a uniform distribution within\\n\\x02500=0 SOA. Although very short SOAs are known to furnish low\\ndetection sensitivity, they were included in our test set as an extreme\\nexample.\\nThis raw data with high temporal resolution (f=20Hz) were'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='detection sensitivity, they were included in our test set as an extreme\\nexample.\\nThis raw data with high temporal resolution (f=20Hz) were\\nsampled to temporal resolutions of TR=1, 2, 3, and 4 seconds,\\nincluding slice-dependent acquisition time differences by shifting the\\nresponse function by TR−\\nTR\\nnumber of slices between the ﬁrst and the\\nlast scanned slice. Sequential and interleaved scanning orders were\\nsimulated.\\nDifferent levels of white noise were added to time courses of each\\nvoxel yielding simulated contrast-to-noise ratios (CNR) of 1.7, 2.5, 5,\\nand inﬁnite (i.e. noise-free), respectively.\\n3 \\n4 \\n5 \\n6 \\n7 \\n8 \\n9 \\n10\\n11\\n12\\n1\\n2\\n3\\n4\\n5\\n1\\n2\\n3\\n4\\n5\\nTime [TRs]\\nSlices\\nFig. 1. Illustration of the slice-timing problem. The hemodynamic responses of the\\nindividual slices are acquired at different points in time (top), yielding an aberration in\\nthe scanned data (bottom). The observed time courses of a hemodynamic response'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 1}, page_content='individual slices are acquired at different points in time (top), yielding an aberration in\\nthe scanned data (bottom). The observed time courses of a hemodynamic response\\nreach their maximum amplitude earlier in slices acquired later. Without adequate\\ncompensation this leads to biased estimators in fMRI analysis.\\n589\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 2}, page_content='590\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='After 3-dimensional spatial smoothing of 6 mm FWHM (isotropic\\nGaussian kernel), GLM analysis was performed with stimulus onsets\\naligned to the acquisition time of the middle slice. This caused a TR-\\ndependent slice-timing difference of approximately up to ±0.5,1,1.5,\\nand 2s, respectively, between slices. In separate analyses, the\\ntemporal derivative of the response function was also modeled in\\nthe GLM to allow some ﬂexibility in the signal onsets and to evaluate\\nits capabilities in compensating slice acquisition delays (Friston et al.,\\n1998).\\nFinally, four datasets were evaluated: uncorrected (original) and\\nSTC (STC) data, each analyzed with and without temporal derivatives\\n(original+TD, STC+TD) in the GLM.\\nHuman fMRI data\\nSubjects\\n13 healthy subjects (8 male, 5 female; age: 25.4±3.3 years)\\nparticipated in this study, which was approved by the ethics committee\\nof the Medical University of Vienna, Austria, and gave written informed'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='participated in this study, which was approved by the ethics committee\\nof the Medical University of Vienna, Austria, and gave written informed\\nconsent prior to the experiment. Subjects were clinically assessed and\\nfree of current or past psychiatric or neurological conditions. Negative\\ndrug screening was mandatory. Nicotine or caffeine intake was not\\nallowed on the day of the measurement.\\nData acquisition and fMRI paradigm\\nThe participants were instructed to do bimanual ﬁnger tapping\\nduring the display of an animated checkerboard, changing black and\\nwhite tiles with a frequency of 8Hz, in a blocked paradigm, consisting\\nof six blocks of 10s resting and 10s ﬁnger-tapping periods (130s total).\\nThis design is known to robustly trigger neural activation in visual\\nareas (occipital lobe) and motor areas (precentral gyrus, supplemen-\\ntary motor area, basal ganglia). Finger movements were self-paced\\nand none of the subjects reported difﬁculties with this task.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='areas (occipital lobe) and motor areas (precentral gyrus, supplemen-\\ntary motor area, basal ganglia). Finger movements were self-paced\\nand none of the subjects reported difﬁculties with this task.\\n75 volumes were acquired on a TIM TRIO 3T full-body MR scanner\\n(SIEMENS Medical, Germany) using single-shot 2D EPI with the\\nfollowing parameters: 20 axial slices of 3mm thickness (2.1mm slice\\ngap) aligned to the connecting line between anterior and posterior\\ncommissure with a matrix size of 128 × 128px2, TE = 40 ms,\\nTR=1800ms and a ﬁeld of view of 192×192mm2 (1.5×1.5×3mm3\\nvoxel size). Slices were acquired interleaved along the z-axis in feet-\\nto-head direction starting with even-numbered slices (i.e. 2, 4,…, 20,\\n1, 3,…, 19).\\nPre-processing and analysis\\nSlice-timing correction as implemented in SPM8 was applied to all\\n13 empirical datasets. All other standard pre-processing steps were\\nsimilarly performed on both the uncorrected original data and the STC'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='13 empirical datasets. All other standard pre-processing steps were\\nsimilarly performed on both the uncorrected original data and the STC\\ndata, including realignment to compensate for bulk head movements,\\nnormalization to MNI space and spatial smoothing with a Gaussian\\nkernel of 8×8×8mm3 FWHM. Each data set was masked with the\\nSPM8 brainmask template.\\nGLM analysis was performed using regressors generated by\\nconvolving the time course of the visual stimulation onsets and\\nduration with the SPM8 canonical hemodynamic response function.\\nMotion parameters obtained from realignment were included in the\\nGLM. All other analysis steps were identical to those described for\\nsimulated data, yielding four datasets of uncorrected and slice-timing\\ncorrected data, each with and without temporal derivatives. The\\nsingle-subject parameter estimate maps from the canonical regressors\\nwere used for second level group analyses.\\nWe assessed the effect of slice-timing correction and temporal'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='single-subject parameter estimate maps from the canonical regressors\\nwere used for second level group analyses.\\nWe assessed the effect of slice-timing correction and temporal\\nderivative modeling in two ways. First, we quantiﬁed the bias in\\nparameter estimation, associated with temporal correction and\\nmodeling, by examining the parameter estimates relative to the\\nestimates from noiseless data. To assess sensitivity, we analyzed all\\ncombinations of treated data to test for activations at the group level\\nin the empirical study, using statistical parametric maps based upon\\nT-statistics.\\nResults\\nSimulated data\\nOur results clearly showed a reduction of parameter estimates (β-\\nvalues) with increased estimator bias, depending on the measure-\\nment and paradigm parameters. In general, this bias was reduced by\\nthe STC procedure.\\nAs expected, the reduction of parameter estimates due to slice-\\ndelay effects were more pronounced for long TRs, event-related'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='the STC procedure.\\nAs expected, the reduction of parameter estimates due to slice-\\ndelay effects were more pronounced for long TRs, event-related\\ndesigns and designs with shorter SOA. We found a maximum decrease\\nof parameter estimates of 63% (event-related design, average\\nSOA=4s, TR=4s). In blocked designs we observed a reduction of\\n12% (block length 10s, TR=4 s). Note that in our simulations the\\nparadigm regressors and slice-timing correction are temporally\\naligned to the middle slice of the 3D volume. Therefore, a TR of e.g.\\n4 s in this simulation is comparable to a dataset with a TR of 2s when\\nthe ﬁrst (or last) slice acquired are used as a reference slice. This can\\nbe a reasonable practice, if the region of interest and putative\\nactivations are located near the ﬁrst (or last) slice because it\\nsuppresses temporal interpolation effects in these areas.\\nCompared to uncorrected data an increase of parameter estimates\\nwas found for all tested paradigms, TRs and CNRs in slice-timing'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='suppresses temporal interpolation effects in these areas.\\nCompared to uncorrected data an increase of parameter estimates\\nwas found for all tested paradigms, TRs and CNRs in slice-timing\\ncorrected data. Importantly, STC did not cause a decrease in average or\\nminimum parameter estimates in any of our datasets (Fig. 2B). One\\nsimulation set (event-related design, average SOA=4s, TR=4s;\\nFig. 2B, lower panel, 1st box, 4th column) showed a decrease by 4%\\nin maximum parameter estimates. This, however, is attributable to\\naliasing artifacts caused by noncritical sampling, which can occur\\nwhen the average SOA is shorter or equal to the TR.\\nOmitting STC and compensating for slice-timing effects by\\nincluding the temporal derivatives into the model yielded no increase\\nin mean parameter estimates (Fig. 2A).\\nWhile higher levels of Gaussian noise caused higher residual\\nvalues in the result of the GLM analysis, it had little impact on the\\nparameter estimates. We found no noise-related changes in the'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='While higher levels of Gaussian noise caused higher residual\\nvalues in the result of the GLM analysis, it had little impact on the\\nparameter estimates. We found no noise-related changes in the\\nperformance efﬁciency of slice-timing correction.\\nTo better assess the consequences of slice-timing correction we\\nfurther analyzed its effect on the parameter estimates in individual\\nslices. Fig. 3 shows a comparison of typical event-related designs. It\\ncan be seen that in all presented cases slice-timing correction led to an\\nincrease of parameter estimates and a reduction of residuals. We\\nfound this for all tested designs and image parameters, with the\\nexception of designs with subcritical sampling. It should be noted,\\nhowever, that this decrease in parameter estimates was not due to a\\nfailure of the slice-timing correction algorithm but due to aliasing\\nartifacts because of noncritical sampling of the data.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='however, that this decrease in parameter estimates was not due to a\\nfailure of the slice-timing correction algorithm but due to aliasing\\nartifacts because of noncritical sampling of the data.\\nFig. 2. Effects of including temporal derivatives (TD) [A] or performing slice-timing correction (STC) [B] on simulated data. The ﬁgure shows a comparison of parameter estimates for\\nuncorrected (red) and corrected (green) data for blocked and event-related designs (upper and lower boxes respectively), varying TRs (columns) and CNRs (sub-columns within TR\\ncolumns). Values are shown relative to the maximum parameter estimate of the uncorrected, noise-free data. Bold lines indicate the mean across all slices. Shaded areas show the\\nrange from highest to lowest estimate. It can be seen that parameter estimates of the slice-timing corrected datasets [B] are, in general (higher mean and smaller range), higher than'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 3}, page_content='range from highest to lowest estimate. It can be seen that parameter estimates of the slice-timing corrected datasets [B] are, in general (higher mean and smaller range), higher than\\nthe uncorrected datasets and therefore closer to the unbiased estimates with optimal temporal alignment (i.e. 100%). Also for lower contrast-to-noise ratios (CNRs) higher parameter\\nestimates were consistently gained for slice-timing corrected data sets.\\n591\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 4}, page_content='Three-dimensional smoothing may conceal the effects of STC on\\nthe individual slices. Therefore we also analyzed simulated images\\nthat were only smoothed in-plane. For our tested datasets slice-timing\\ncorrection yielded robust and profound improvements of parameter\\nestimates for all individual slices. There was a signiﬁcant decrease of\\nparameter estimates and increase of residuals in the uncorrected\\ndatasets in slices with larger temporal offset (Fig. 3).\\nHuman fMRI data\\nAs expected, the paradigm stimulated brain activation changes\\nin the motor areas (M1, SMA, z=56mm [MNI]), the visual cortex (V1,\\nz=−4mm [MNI]) and the cerebellum (culmen, V1, z=−24mm\\n[MNI]) (Fig. 4, left table, original).\\nComparing slice-timing corrected and uncorrected data using\\npaired T-tests (Fig. 4, right table, orig. vs STC) revealed signiﬁcant\\n(pb0.001) activation differences. This particularly affected dorsal\\nregions, i.e. regions with higher temporal offset due to the acquisition'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 4}, page_content='(pb0.001) activation differences. This particularly affected dorsal\\nregions, i.e. regions with higher temporal offset due to the acquisition\\ndirection, where the slice-timing effect was expected to be most\\ndominant. Importantly, robust SMA activation was detected in the STC\\ndata sets only. Differences were less prominent in ventral brain\\nregions, where slice-timing effects were expected to be lower.\\nFor uncorrected data, adding temporal derivative regressors in the\\ndesign matrix caused a reduction in sensitivity, particularly in the\\nmotor network (M1 and SMA) (Fig. 4, left table, TD, and right table,\\nxtitoriginal vs TD).\\nIncluding the temporal derivatives in the GLM analysis of the slice-\\ntiming corrected data yielded higher parameter estimates in the focal\\nFig. 3. Slice-dependent changes in a selection of typical event-related designs. The top part of each graph shows a comparison of parameter estimates from uncorrected (red) and'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 4}, page_content='Fig. 3. Slice-dependent changes in a selection of typical event-related designs. The top part of each graph shows a comparison of parameter estimates from uncorrected (red) and\\nslice-timing corrected (green) data sets. Dashed lines represent the parameter estimates of models where temporal derivatives were included. Residuals of the models are plotted at\\nthe bottom part of the graphs. To visualize the impact of spatial smoothing on the temporal alignment the same data is presented 3D (panel A) and 2D (panel B) smoothed.\\n592\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content='point of M1 activation (Fig. 4, left table, STC+TD, and right table,\\noriginal vs STC+TD). However, we found no activation in the SMA.\\nDiscussion\\nOur results clearly show a beneﬁt of slice-timing correction for\\nparameter estimation on single-subject level, which, in turn, could\\nsigniﬁcantly improve sensitivity in group statistical analysis. This is\\nparticularly true for event-related designs and blocked designs with\\nshort block length. Analysis of in vivo fMRI images supported our\\nﬁndings from simulated data.\\nNo adverse effects of STC with respect to parameter estimate bias\\nwere found for realistic paradigm designs. Higher noise levels had no\\nimpact on estimates or the efﬁciency of the applied STC algorithm.\\nFurther, no adverse effects were found in in vivo datasets with non-\\nGaussian noise due to physiological and motion artifacts.\\nWhile beneﬁcial effects were found for STC, adding temporal\\nderivatives to the model did not suppress bias (shrinkage) in'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content='Gaussian noise due to physiological and motion artifacts.\\nWhile beneﬁcial effects were found for STC, adding temporal\\nderivatives to the model did not suppress bias (shrinkage) in\\nparameter estimates to the same extent. Clearly, the shrinkage of\\nparameter estimates does not necessarily decrease sensitivity,\\nprovided the between subject variability in parameter estimates\\ndecreases in proportion. However, our empirical example suggests\\nthat slice-timing correction increased sensitivity to group effects in a\\nway that was subverted by including temporal derivatives. It has been\\nshown previously that using TD in whole-brain analyses can diminish\\npower depending on the response latency (Della-Maggiore et al.,\\n2002). This ﬁnding is consistent with our ﬁndings.\\nFurthermore, we know that temporal derivatives can explain\\nvariance due to temporal delays of about ±1s (Henson et al., 2002).\\nThis upper bound on the delays that temporal derivatives can handle'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content='Furthermore, we know that temporal derivatives can explain\\nvariance due to temporal delays of about ±1s (Henson et al., 2002).\\nThis upper bound on the delays that temporal derivatives can handle\\nreﬂects the failure of the ﬁrst-order Taylor expansion implicit in their\\nuse, which is determined quantitatively by the characteristic time\\nconstants of the HRF. This bound means that, even under the\\nassumption of an optimal model (i.e. regressors time-locked to the\\nmiddle of one TR), derivatives will not model adequately slice-timing\\neffects caused by TRs greater than 2s. Our simulations corroborate this\\nfact and additionally showed no consistent improvements even for\\nshorter TRs. This is expected as the canonical HRF and its derivative\\ncan lose their orthogonality when convolved with a stimulus function.\\nTherefore, a part of the variance explained by the canonical HRF\\nregressor may be modeled by its derivative. In general, including the'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content='Therefore, a part of the variance explained by the canonical HRF\\nregressor may be modeled by its derivative. In general, including the\\ntemporal derivatives will thus lead to a decrease in residuals, possibly\\nincreasing statistical signiﬁcance on the single-subject level, but it will\\nalso reduce the parameter estimates of the canonical regressor. This\\nmay reduce sensitivity, as only the latter usually enter ordinary least\\nsquares random-effects group analysis. In principle, the problem of\\nnon-orthogonality can be addressed by specifying (F) contrasts at the\\nsecond level over both the canonical and temporal derivative\\nparameter estimates. However, this depends upon their being a\\nconsistent group effect on both, which may not be guaranteed if slice-\\ntiming effects are artefactual (Calhoun et al., 2004; Henson et al.,\\n1999; Steffener et al., 2010).\\nThe question of whether to include temporal derivatives or not is\\nreally an issue of model selection. Generally speaking, their inclusion'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content=\"1999; Steffener et al., 2010).\\nThe question of whether to include temporal derivatives or not is\\nreally an issue of model selection. Generally speaking, their inclusion\\nwill explain away early or late hemodynamic responses, which could\\nreduce sensitivity. On the other hand, it is at least theoretically\\npossible that modeling real or artifactual latency differences will\\nincrease the precision of parameter estimates. The overall impact on\\nsensitivity to canonical hemodynamic responses is clearly difﬁcult to\\ngeneralize about. However, the procedure described in this paper\\nprovides a simple protocol for assessing the impact of temporal\\nderivatives, with or without slice-timing correction (see Fig. 4).\\nThe results in this paper are based on SPM8's implementation of\\nthe STC algorithm. Nevertheless, our ﬁndings should be transferable\\nto the implementations in other fMRI applications, as cubic spline\\ninterpolation and sinc interpolation have been shown to provide\"),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content='the STC algorithm. Nevertheless, our ﬁndings should be transferable\\nto the implementations in other fMRI applications, as cubic spline\\ninterpolation and sinc interpolation have been shown to provide\\nalmost identical results (Goebel, 2010).\\nA sometimes-neglected issue is the interaction of the slice-timing\\neffect and motion. Subject movement along the z-axis or head tilts\\naround the x-axis will introduce not only spatial but also temporal\\nartifacts. This is a concern particularly for clinical applications where\\npatients might not be able to fully suppress their movements while in\\nFig. 4. Comparison of in vivo fMRI data. Group SPMs of a visual-motor task versus rest condition (left table) and paired t-test comparisons of compensation strategies for slice-timing\\neffects. Threshold was set to pb0.001. The positive effects of slice-timing correction on the parameter estimates (STC, left table, third column) are present particularly in the motor'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 5}, page_content='effects. Threshold was set to pb0.001. The positive effects of slice-timing correction on the parameter estimates (STC, left table, third column) are present particularly in the motor\\nnetwork (ﬁrst row). Due to our measurement parameters (feet-to-head acquisition order) slice-timing effects were expected to be most prominent in this dorsal section.\\nComparison of the uncorrected and STC data revealed signiﬁcant increase of t-values in group statistics.\\n593\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 6}, page_content='the scanner. Given the nature of typical serial fMRI pre-processing\\npipelines one could consider performing slice-timing correction\\nbefore realignment if subject movement is moderate. In the presence\\nof pronounced inter-slice movements it might be preferable to realign\\nbefore STC. Clearly, serial correction approaches like these are\\nsuboptimal and cannot compensate for the interactions between\\ntime shifts and movement (Bannister et al., 2007), and a combined\\nmethod of motion and timing correction might thus be indicated\\n(Bannister et al., 2004). However, this is not yet implemented in the\\nmajor fMRI analysis packages.\\nA non-beneﬁcial side effect of all resampling methods is the\\npossible introduction of aliasing effects for signals at frequencies\\nabove the Nyquist sampling limit of f =\\n1\\n2 × TR. Given a typical TR of\\n2 s (f=0.25Hz) a minimal inter-stimulus interval (ISI) of more than\\n4s is recommended. When using the slice acquired in the middle of'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 6}, page_content='1\\n2 × TR. Given a typical TR of\\n2 s (f=0.25Hz) a minimal inter-stimulus interval (ISI) of more than\\n4s is recommended. When using the slice acquired in the middle of\\nthe acquisition period (TR) as a reference slice, the signal needs to be\\nshifted by TR / 2, therefore reducing the suggested minimal ISI to 2s.\\nWhen choosing the mid-slice as the target less temporal interpolation\\nis needed. This might decrease the risk of interpolation artifacts. Note\\nthat all regressors in the GLM also need to be adjusted for this shift in\\ntime by TR / 2.\\nFinally, Holmes et al. (1997) had shown that no physiologically\\ninduced signal change was present at frequencies over 0.25Hz. As a\\nconsequence fMRI data can be assumed band-limited, which means\\nthat potential aliasing effects by STC only affect signal components\\noriginating from noise.\\nIn conclusion, slice-timing correction has been shown to suppress\\nbias in quantitative parameter estimates at the single-subject level'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 6}, page_content='originating from noise.\\nIn conclusion, slice-timing correction has been shown to suppress\\nbias in quantitative parameter estimates at the single-subject level\\nand, empirically, to increase sensitivity in group-level statistical\\ninferences.\\nAcknowledgments\\nThe authors would like to thank Guillaume Flandin and Elmar\\nLaistler for helpful comments and the OeNB (Austrian National Bank,\\nP12982) and the FWF (Austrian Science Fund, P23021) for ﬁnancial\\nsupport.\\nReferences\\nBannister, P.R., Brady, J.M., Jenkinson, M., 2004. TIGER - A new model for spatio-\\ntemporal realignment of fMRI data. Lecture Notes in Computer Science 3117,\\n292–303.\\nBannister, P.R., Brady, J.M., Jenkinson, M., 2007. Integrating temporal information with a\\nnon-rigid method of motion correction for functional magnetic resonance images.\\nImage and Vision Computing 25, 311–320.\\nCalhoun, V., Golay, X., Pearlson, G., 2000. Improved fMRI slice timing correction:'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 6}, page_content='Image and Vision Computing 25, 311–320.\\nCalhoun, V., Golay, X., Pearlson, G., 2000. Improved fMRI slice timing correction:\\ninterpolation errors and wrap around effects. Proceedings, ISMRM, 9th Annual\\nMeeting, Denver, p. 810.\\nCalhoun, V., Stevens, M., Pearlson, G., Kiehl, K., 2004. fMRI analysis with the general\\nlinear model: removal of latency-induced amplitude bias by incorporation of\\nhemodynamic derivative terms. Neuroimage 22, 252–257.\\nDella-Maggiore, V., Wilkin, C., Peres-Neto, P.R., McIntosh, A.R., 2002. An empirical\\ncomparison of SPM preprocessing parameters to the analysis of fMRI data.\\nNeuroImage 17, 19–28.\\nFriston, K., Fletcher, P., Josephs, O., Holmes, A., Rugg, M., 1998. Event-related fMRI:\\ncharacterizing differential responses. Neuroimage 7, 77–83.\\nGoebel, R., 2010. Brain Voyager—Users Guide.\\nHenson, R., Buechel, C., Josephs, O., Friston, K., 1999. The slice-timing problem in event-\\nrelated fMRI. NeuroImage 9, 125-125.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 6}, page_content='Goebel, R., 2010. Brain Voyager—Users Guide.\\nHenson, R., Buechel, C., Josephs, O., Friston, K., 1999. The slice-timing problem in event-\\nrelated fMRI. NeuroImage 9, 125-125.\\nHenson, R., Price, C., Rugg, M., Turner, R., Friston, K., 2002. Detecting latency differences\\nin event-related BOLD responses: application to words versus nonwords and initial\\nversus repeated face presentations. NeuroImage 15, 83–97.\\nHolmes, A.P., Josephs, O., Büchel, C., Friston, K.J., 1997. Statistical modelling of low-\\nfrequency confounds in fMRI. Proceedings, HBM, 3rd Annual Meeting, Copenhagen,\\np. 480.\\nHopﬁnger, J.B., Büchel, C., Holmes, A.P., Friston, K.J., 2000. A study of analysis\\nparameters that inﬂuence the sensitivity of event-related fMRI analyses. Neuro-\\nImage 11, 326–333.\\nSteffener, J., Tabert, M., Reuben, A., Stern, Y., 2010. Investigating hemodynamic\\nresponse variability at the group level using basis functions. NeuroImage 49,\\n2113–2122.'),\n",
       " Document(metadata={'producer': 'Acrobat Distiller 8.1.0 (Windows)', 'creator': 'Elsevier', 'creationdate': '2011-08-02T07:13:02+08:00', 'source': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'file_path': '..\\\\data\\\\literature\\\\Slice-timing effects and their correction in functional MRI.pdf', 'total_pages': 7, 'format': 'PDF 1.7', 'title': 'Slice-timing effects and their correction in functional MRI', 'author': 'Ronald Sladky', 'subject': 'NeuroImage, 58 (2011) 588-594. doi:10.1016/j.neuroimage.2011.06.078', 'keywords': 'Functional MRI; Pre-processing; Analysis; Slice-timing correction', 'moddate': '2014-09-04T21:26:08+05:30', 'trapped': '', 'modDate': \"D:20140904212608+05'30'\", 'creationDate': \"D:20110802071302+08'00'\", 'page': 6}, page_content='Image 11, 326–333.\\nSteffener, J., Tabert, M., Reuben, A., Stern, Y., 2010. Investigating hemodynamic\\nresponse variability at the group level using basis functions. NeuroImage 49,\\n2113–2122.\\nStehling, M., Turner, R., Mansﬁeld, P., 1991. Echo-planar imaging: magnetic resonance\\nimaging in a fraction of a second. Science 254, 43.\\nTurner, R., Howseman, A., Rees, G., Josephs, O., Friston, K., 1998. Functional magnetic\\nresonance imaging of the human brain: data acquisition and analysis. Experimental\\nBrain Research 123, 5–12.\\nWorsley, K., Taylor, J., 2006. Detecting fMRI activation allowing for unknown latency of\\nthe hemodynamic response. NeuroImage 29, 649–654.\\n594\\nR. Sladky et al. / NeuroImage 58 (2011) 588–594')]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chunks = split_documents(pdf_documents)\n",
    "chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "76843b32",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generate embeddings for 146 texts...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches:   0%|          | 0/5 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 5/5 [00:18<00:00,  3.73s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generate embeddings with shape: (146, 384)\n",
      "Adding 146 documents to vector store...\n",
      "✅ Successfully added 146 documents\n",
      "📊 Total documents in collection: 292\n"
     ]
    }
   ],
   "source": [
    "### Convert the text to embeddings\n",
    "texts=[doc.page_content for doc in chunks]\n",
    "\n",
    "## Generate the Embeddings\n",
    "\n",
    "embeddings=embedding_manager.generate_embeddings(texts)\n",
    "\n",
    "##store int he vector dtaabase\n",
    "vectorstore.add_documents(chunks,embeddings)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86162b5e",
   "metadata": {},
   "source": [
    "Retriever Pipeline From VectorStore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "edb92726",
   "metadata": {},
   "outputs": [],
   "source": [
    "class RAGRetriever:\n",
    "    \"\"\"Handles query-based retrieval from the vector store\"\"\"\n",
    "    \n",
    "    def __init__(self, vector_store: VectorStore, embedding_manager: EmbeddingManager):\n",
    "        \"\"\"\n",
    "        Initialize the retriever\n",
    "        \n",
    "        Args:\n",
    "            vector_store: Vector store containing document embeddings\n",
    "            embedding_manager: Manager for generating query embeddings\n",
    "        \"\"\"\n",
    "        self.vector_store = vector_store\n",
    "        self.embedding_manager = embedding_manager\n",
    "\n",
    "    def retrieve(self, query: str, top_k: int = 5, score_threshold: float = 0.0) -> List[Dict[str, Any]]:\n",
    "        \"\"\"\n",
    "        Retrieve relevant documents for a query\n",
    "        \n",
    "        Args:\n",
    "            query: The search query\n",
    "            top_k: Number of top results to return\n",
    "            score_threshold: Minimum similarity score threshold\n",
    "            \n",
    "        Returns:\n",
    "            List of dictionaries containing retrieved documents and metadata\n",
    "        \"\"\"\n",
    "        print(f\"Retrieving documents for query: '{query}'\")\n",
    "        print(f\"Top K: {top_k}, Score threshold: {score_threshold}\")\n",
    "        \n",
    "        # Generate query embedding\n",
    "        query_embedding = self.embedding_manager.generate_embeddings([query])[0]\n",
    "        \n",
    "        # Search in vector store\n",
    "        try:\n",
    "            results = self.vector_store.collection.query(\n",
    "                query_embeddings=[query_embedding.tolist()],\n",
    "                n_results=top_k\n",
    "            )\n",
    "        # Process results\n",
    "            retrieved_docs = []\n",
    "            \n",
    "            if results['documents'] and results['documents'][0]:\n",
    "                documents = results['documents'][0]\n",
    "                metadatas = results['metadatas'][0]\n",
    "                distances = results['distances'][0]\n",
    "                ids = results['ids'][0]\n",
    "                \n",
    "                for i, (doc_id, document, metadata, distance) in enumerate(zip(ids, documents, metadatas, distances)):\n",
    "                    # Convert distance to similarity score (ChromaDB uses cosine distance)\n",
    "                    similarity_score = 1 - distance\n",
    "                    \n",
    "                    if similarity_score >= score_threshold:\n",
    "                        retrieved_docs.append({\n",
    "                            'id': doc_id,\n",
    "                            'content': document,\n",
    "                            'metadata': metadata,\n",
    "                            'similarity_score': similarity_score,\n",
    "                            'distance': distance,\n",
    "                            'rank': i + 1\n",
    "                        })\n",
    "                \n",
    "                print(f\"Retrieved {len(retrieved_docs)} documents (after filtering)\")\n",
    "            else:\n",
    "                print(\"No documents found\")\n",
    "            \n",
    "            return retrieved_docs\n",
    "            \n",
    "        except Exception as e:\n",
    "            print(f\"Error during retrieval: {e}\")\n",
    "            return []\n",
    "\n",
    "rag_retriever=RAGRetriever(vectorstore,embedding_manager)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "dbf5e5d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<__main__.RAGRetriever at 0x1df9a657230>"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rag_retriever"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "edd5dc76",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving documents for query: 'What is attention is all you need'\n",
      "Top K: 5, Score threshold: 0.0\n",
      "Generate embeddings for 1 texts...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:00<00:00, 24.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generate embeddings with shape: (1, 384)\n",
      "Retrieved 0 documents (after filtering)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rag_retriever.retrieve(\"What is attention is all you need\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "dc51e8e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving documents for query: 'Unified Multi-task Learning Framework'\n",
      "Top K: 5, Score threshold: 0.0\n",
      "Generate embeddings for 1 texts...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:00<00:00, 24.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generate embeddings with shape: (1, 384)\n",
      "Retrieved 0 documents (after filtering)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rag_retriever.retrieve(\"Unified Multi-task Learning Framework\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a37f7c5c",
   "metadata": {},
   "source": [
    "RAG Pipeline- VectorDB To LLM Output Generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "f9ea3d1b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "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\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "print(os.getenv(\"GROQ_API_KEY\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c3c4fd48",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain_groq import ChatGroq\n",
    "from langchain_core.prompts import PromptTemplate\n",
    "from langchain_core.messages import HumanMessage, SystemMessage\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "1cd0fcf0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "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\n"
     ]
    }
   ],
   "source": [
    "from dotenv import load_dotenv\n",
    "import os\n",
    "\n",
    "load_dotenv()   # loads .env file\n",
    "\n",
    "api_key = os.getenv(\"GROQ_API_KEY\")\n",
    "print(api_key)  # should NOT be None\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "78e0b85e",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Simple RAG pipeline with Groq LLM\n",
    "from langchain_groq import ChatGroq\n",
    "import os\n",
    "from dotenv import load_dotenv\n",
    "load_dotenv()\n",
    "\n",
    "### Initialize the Groq LLM (set your GROQ_API_KEY in environment)\n",
    "groq_api_key = os.getenv(\"GROQ_API_KEY\")\n",
    "\n",
    "llm=ChatGroq(groq_api_key=groq_api_key,model_name=\"gemma2-9b-it\",temperature=0.1,max_tokens=1024)\n",
    "\n",
    "## 2. Simple RAG function: retrieve context + generate response\n",
    "def rag_simple(query,retriever,llm,top_k=3):\n",
    "    ## retriever the context\n",
    "    results=retriever.retrieve(query,top_k=top_k)\n",
    "    context=\"\\n\\n\".join([doc['content'] for doc in results]) if results else \"\"\n",
    "    if not context:\n",
    "        return \"No relevant context found to answer the question.\"\n",
    "    \n",
    "    ## generate the answwer using GROQ LLM\n",
    "    prompt=f\"\"\"Use the following context to answer the question concisely.\n",
    "        Context:\n",
    "        {context}\n",
    "\n",
    "        Question: {query}\n",
    "\n",
    "        Answer:\"\"\"\n",
    "    \n",
    "    response=llm.invoke([prompt.format(context=context,query=query)])\n",
    "    return response.content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "42011ac7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving documents for query: 'What is attention mechanism?'\n",
      "Top K: 3, Score threshold: 0.0\n",
      "Generate embeddings for 1 texts...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:00<00:00, 28.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generate embeddings with shape: (1, 384)\n",
      "Retrieved 0 documents (after filtering)\n",
      "No relevant context found to answer the question.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "answer=rag_simple(\"What is attention mechanism?\",rag_retriever,llm)\n",
    "print(answer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "9c487538",
   "metadata": {},
   "outputs": [],
   "source": [
    "class GroqLLM:\n",
    "    def __init__(self, model_name: str = \"gemma2-9b-it\", api_key: str =None):\n",
    "        \"\"\"\n",
    "        Initialize Groq LLM\n",
    "        \n",
    "        Args:\n",
    "            model_name: Groq model name (qwen2-72b-instruct, llama3-70b-8192, etc.)\n",
    "            api_key: Groq API key (or set GROQ_API_KEY environment variable)\n",
    "        \"\"\"\n",
    "        self.model_name = model_name\n",
    "        self.api_key = api_key or os.environ.get(\"GROQ_API_KEY\")\n",
    "        \n",
    "        if not self.api_key:\n",
    "            raise ValueError(\"Groq API key is required. Set GROQ_API_KEY environment variable or pass api_key parameter.\")\n",
    "        \n",
    "        self.llm = ChatGroq(\n",
    "            groq_api_key=self.api_key,\n",
    "            model_name=self.model_name,\n",
    "            temperature=0.1,\n",
    "            max_tokens=1024\n",
    "        )\n",
    "        \n",
    "        print(f\"Initialized Groq LLM with model: {self.model_name}\")\n",
    "\n",
    "    def generate_response(self, query: str, context: str, max_length: int = 500) -> str:\n",
    "        \"\"\"\n",
    "        Generate response using retrieved context\n",
    "        \n",
    "        Args:\n",
    "            query: User question\n",
    "            context: Retrieved document context\n",
    "            max_length: Maximum response length\n",
    "            \n",
    "        Returns:\n",
    "            Generated response string\n",
    "        \"\"\"\n",
    "        \n",
    "        # Create prompt template\n",
    "        prompt_template = PromptTemplate(\n",
    "            input_variables=[\"context\", \"question\"],\n",
    "            template=\"\"\"You are a helpful AI assistant. Use the following context to answer the question accurately and concisely.\n",
    "\n",
    "Context:\n",
    "{context}\n",
    "\n",
    "Question: {question}\n",
    "\n",
    "Answer: Provide a clear and informative answer based on the context above. If the context doesn't contain enough information to answer the question, say so.\"\"\"\n",
    "        )\n",
    "        \n",
    "        # Format the prompt\n",
    "        formatted_prompt = prompt_template.format(context=context, question=query)\n",
    "        \n",
    "        try:\n",
    "            # Generate response\n",
    "            messages = [HumanMessage(content=formatted_prompt)]\n",
    "            response = self.llm.invoke(messages)\n",
    "            return response.content\n",
    "            \n",
    "        except Exception as e:\n",
    "            return f\"Error generating response: {str(e)}\"\n",
    "        \n",
    "    def generate_response_simple(self, query: str, context: str) -> str:\n",
    "        \"\"\"\n",
    "        Simple response generation without complex prompting\n",
    "        \n",
    "        Args:\n",
    "            query: User question\n",
    "            context: Retrieved context\n",
    "            \n",
    "        Returns:\n",
    "            Generated response\n",
    "        \"\"\"\n",
    "        simple_prompt = f\"\"\"Based on this context: {context}\n",
    "\n",
    "Question: {query}\n",
    "\n",
    "Answer:\"\"\"\n",
    "        \n",
    "        try:\n",
    "            messages = [HumanMessage(content=simple_prompt)]\n",
    "            response = self.llm.invoke(messages)\n",
    "            return response.content\n",
    "        except Exception as e:\n",
    "            return f\"Error: {str(e)}\"\n",
    "  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "6f196001",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initialized Groq LLM with model: gemma2-9b-it\n",
      "Groq LLM initialized successfully!\n"
     ]
    }
   ],
   "source": [
    "# Initialize Groq LLM (you'll need to set GROQ_API_KEY environment variable)\n",
    "try:\n",
    "    groq_llm = GroqLLM(api_key=os.getenv(\"GROQ_API_KEY\"))\n",
    "    print(\"Groq LLM initialized successfully!\")\n",
    "except ValueError as e:\n",
    "    print(f\"Warning: {e}\")\n",
    "    print(\"Please set your GROQ_API_KEY environment variable to use the LLM.\")\n",
    "    groq_llm = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "327d9c42",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving documents for query: 'Unified Multi-task Learning Framework'\n",
      "Top K: 5, Score threshold: 0.0\n",
      "Generate embeddings for 1 texts...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:00<00:00, 25.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generate embeddings with shape: (1, 384)\n",
      "Retrieved 0 documents (after filtering)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### get the context from the retriever and pass it to the LLM\n",
    "\n",
    "rag_retriever.retrieve(\"Unified Multi-task Learning Framework\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "673bfef6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40b175c4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
